<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.4.555">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Ian R. Sturgill">

<title>Figure code</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="figure_code_files/libs/clipboard/clipboard.min.js"></script>
<script src="figure_code_files/libs/quarto-html/quarto.js"></script>
<script src="figure_code_files/libs/quarto-html/popper.min.js"></script>
<script src="figure_code_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="figure_code_files/libs/quarto-html/anchor.min.js"></script>
<link href="figure_code_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="figure_code_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="figure_code_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="figure_code_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="figure_code_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">


</head>

<body class="fullcontent">

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">

<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Figure code</h1>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Ian R. Sturgill </p>
          </div>
  </div>
    
  
    
  </div>
  


</header>


<section id="purpose" class="level1">
<h1>Purpose</h1>
<p>The purpose of this document is to both walk through the text and demonstrate reconstruction of the figures from our paper “Expanded detection and impact of <em>BAP1</em> alterations in cancer”, available on bioRxiv at: <a href="https://www.biorxiv.org/content/10.1101/2023.11.21.568094v2" class="uri">https://www.biorxiv.org/content/10.1101/2023.11.21.568094v2</a></p>
<p>For more detailed intermediate analyses and data that can’t be hosted on Github, please see the associated Zenodo submission at: <a href="https://zenodo.org/doi/10.5281/zenodo.10175692" class="uri">https://zenodo.org/doi/10.5281/zenodo.10175692</a></p>
</section>
<section id="figures" class="level1">
<h1>Figures</h1>
<section id="note" class="level2">
<h2 class="anchored" data-anchor-id="note">Note</h2>
<p>Please note that some plots won’t display properly from the Quarto markdown – especially <code>ComplexHeatmap</code> plots. Generating the plots outside of the Quarto markdown environment should solve most visual issues. Plots were generated in R and finalized in Adobe Illustrator 2021.</p>
<p>This project uses an R environment from the <code>renv</code> package to capture package versions used in the analyses.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a>renv<span class="sc">::</span><span class="fu">activate</span>() <span class="co"># Activate R environment if necessary</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="figure-1---pan-cancer-bap1-variants-in-tcga" class="level2">
<h2 class="anchored" data-anchor-id="figure-1---pan-cancer-bap1-variants-in-tcga">Figure 1 - Pan-cancer <em>BAP1</em> variants in TCGA</h2>
<p>Load the variant and per-sample annotation data. <strong>Please note that germline mutations are dbGaP controlled data and must be downloaded independently from the PanCanAtlas Publications page. If you don’t download these data separately, you will need to make a few changes to the code for it to run.</strong></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(data.table)</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a>variants_germline <span class="ot">&lt;-</span> <span class="fu">fread</span>(<span class="st">"../data/PanCanAtlas_Publications/PCA_pathVar_integrated_filtered_adjusted.tsv"</span>)</span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a>variants_germline <span class="ot">&lt;-</span> variants_germline[variants_germline<span class="sc">$</span>HUGO_Symbol <span class="sc">==</span> <span class="st">"BAP1"</span>, ]</span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a>variants_germline<span class="sc">$</span>length <span class="ot">&lt;-</span> variants_germline<span class="sc">$</span>Stop <span class="sc">-</span> variants_germline<span class="sc">$</span>Start <span class="sc">+</span> <span class="dv">1</span></span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a>variants <span class="ot">&lt;-</span> <span class="fu">read.delim</span>(<span class="st">"tables/table_s2a.txt"</span>)</span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a>variants_simple <span class="ot">&lt;-</span> <span class="fu">read.delim</span>(<span class="st">"tables/table_s2c.txt"</span>)</span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a>mc3 <span class="ot">&lt;-</span> <span class="fu">read.delim</span>(<span class="st">"tables/table_s2b.txt"</span>)</span>
<span id="cb2-11"><a href="#cb2-11" aria-hidden="true" tabindex="-1"></a>mc3_simple <span class="ot">&lt;-</span> <span class="fu">read.delim</span>(<span class="st">"tables/table_s2d.txt"</span>)</span>
<span id="cb2-12"><a href="#cb2-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-13"><a href="#cb2-13" aria-hidden="true" tabindex="-1"></a>cohort_anno <span class="ot">&lt;-</span> <span class="fu">read.delim</span>(<span class="st">"tables/table_s3a.txt"</span>)</span>
<span id="cb2-14"><a href="#cb2-14" aria-hidden="true" tabindex="-1"></a>sample_anno <span class="ot">&lt;-</span> <span class="fu">read.delim</span>(<span class="st">"tables/table_s3b.txt"</span>)</span>
<span id="cb2-15"><a href="#cb2-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-16"><a href="#cb2-16" aria-hidden="true" tabindex="-1"></a><span class="co"># TCGA long and short cancer type IDs</span></span>
<span id="cb2-17"><a href="#cb2-17" aria-hidden="true" tabindex="-1"></a>tcga_projects <span class="ot">&lt;-</span> <span class="fu">unique</span>(sample_anno<span class="sc">$</span>cohort)</span>
<span id="cb2-18"><a href="#cb2-18" aria-hidden="true" tabindex="-1"></a>tcga_names <span class="ot">&lt;-</span> <span class="fu">c</span>()</span>
<span id="cb2-19"><a href="#cb2-19" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> tcga_projects) {</span>
<span id="cb2-20"><a href="#cb2-20" aria-hidden="true" tabindex="-1"></a>  tcga_names <span class="ot">&lt;-</span> <span class="fu">c</span>(tcga_names, <span class="fu">strsplit</span>(i, <span class="st">"-"</span>)[[<span class="dv">1</span>]][<span class="dv">2</span>])</span>
<span id="cb2-21"><a href="#cb2-21" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb2-22"><a href="#cb2-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-23"><a href="#cb2-23" aria-hidden="true" tabindex="-1"></a><span class="co"># Standard cancer type colors for plots</span></span>
<span id="cb2-24"><a href="#cb2-24" aria-hidden="true" tabindex="-1"></a>tcga_colors <span class="ot">&lt;-</span> <span class="fu">read.delim</span>(<span class="st">"../data/tcga_colors.txt"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>The first numbers referenced in the results text are:<br>
</p>
<blockquote class="blockquote">
<p>Out of 10,414 samples and 33 cancer types, a total of 1337 non-synonymous, non-intronic variants (1329 somatic and 8 germline) were detected in 988 individuals across the two updated hg38-aligned approaches, including 14 indels &gt;=40bp in length (Supplemental Table S2A).</p>
</blockquote>
<p>All variants here are nonsynonymous, non-intronic variants which have passed individual variant caller quality checks (FILTER flag == “PASS” or GDC_FILTER == ““).</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>num_germline_variants <span class="ot">&lt;-</span> <span class="fu">nrow</span>(variants_germline)</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a><span class="co"># If you don't have germline data access, there are 8 germline BAP1 variants</span></span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a><span class="co"># num_germline_variants &lt;- 8</span></span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>num_variants <span class="ot">&lt;-</span> <span class="fu">nrow</span>(variants) <span class="sc">+</span> num_germline_variants</span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a>num_germline_mutants <span class="ot">&lt;-</span> <span class="fu">length</span>(<span class="fu">unique</span>(variants_germline<span class="sc">$</span>bcr_patient_barcode)) <span class="sc">-</span> </span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">sum</span>(variants_germline<span class="sc">$</span>bcr_patient_barcode <span class="sc">%in%</span> variants<span class="sc">$</span>SAMPLE)</span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a><span class="co"># If you don't have germline data, there is a 1-sample overlap = 7 mutant samples</span></span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a><span class="co"># num_germline_mutants &lt;- 7 </span></span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a>num_mutants <span class="ot">&lt;-</span> <span class="fu">length</span>(<span class="fu">unique</span>(variants<span class="sc">$</span>SAMPLE)) <span class="sc">+</span> num_germline_mutants</span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-12"><a href="#cb3-12" aria-hidden="true" tabindex="-1"></a>num_germline_40bpindels <span class="ot">&lt;-</span> <span class="fu">sum</span>(variants_germline<span class="sc">$</span>length <span class="sc">&gt;=</span> <span class="dv">40</span>)</span>
<span id="cb3-13"><a href="#cb3-13" aria-hidden="true" tabindex="-1"></a><span class="co"># If you don't have germline data, there are 0 &gt;=40bp indels</span></span>
<span id="cb3-14"><a href="#cb3-14" aria-hidden="true" tabindex="-1"></a><span class="co"># num_germline_40bpindels &lt;- 0</span></span>
<span id="cb3-15"><a href="#cb3-15" aria-hidden="true" tabindex="-1"></a>num_40bpindels <span class="ot">&lt;-</span> <span class="fu">sum</span>(variants<span class="sc">$</span>length <span class="sc">&gt;=</span> <span class="dv">40</span>) <span class="sc">+</span> num_germline_40bpindels</span>
<span id="cb3-16"><a href="#cb3-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-17"><a href="#cb3-17" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="fu">sprintf</span>(<span class="st">"# nonsynonymous, non-intronic variants passing simple filtering: %s"</span>,</span>
<span id="cb3-18"><a href="#cb3-18" aria-hidden="true" tabindex="-1"></a>        num_variants), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># nonsynonymous, non-intronic variants passing simple filtering: 1337 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="fu">sprintf</span>(<span class="st">"# mutant samples passing simple filtering: %s"</span>,</span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>        num_mutants), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># mutant samples passing simple filtering: 988 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="fu">sprintf</span>(<span class="st">"# indels &gt;=40bp passing simple filtering: %s"</span>,</span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>        num_40bpindels), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># indels &gt;=40bp passing simple filtering: 14 </code></pre>
</div>
</div>
<p>Next we performed additional filtering to identify variant calls with greater support:</p>
<blockquote class="blockquote">
<p>We conducted further review of variants to select those with greater support (tumor variant allele frequency ≥0.2, tumor alternate allele read count ≥2, and manual review of read alignment in IGV), resulting in a reduction from 1337 to 265 variants in 233 individuals across TCGA cancer types, including 11 larger indels ≥40bp that were not previously detected (Figure 1A-B and Supplemental Table S2C).</p>
</blockquote>
<div class="cell">
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="co"># All germline variants pass simple filtering</span></span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a><span class="co"># all(variants_germline$tumorVAF &gt;= 0.2)</span></span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a><span class="co"># [1] TRUE</span></span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a><span class="co"># all(variants_germline$tumorAltCnt &gt;= 2)</span></span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a><span class="co"># [1] TRUE</span></span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true" tabindex="-1"></a>num_simple_variants <span class="ot">&lt;-</span> <span class="fu">nrow</span>(variants_simple) <span class="sc">+</span> num_germline_variants</span>
<span id="cb9-8"><a href="#cb9-8" aria-hidden="true" tabindex="-1"></a>num_simple_mutants <span class="ot">&lt;-</span> <span class="fu">length</span>(<span class="fu">unique</span>(variants_simple<span class="sc">$</span>SAMPLE)) <span class="sc">+</span> num_germline_mutants</span>
<span id="cb9-9"><a href="#cb9-9" aria-hidden="true" tabindex="-1"></a>num_simple_40bpindels <span class="ot">&lt;-</span> <span class="fu">sum</span>(variants_simple<span class="sc">$</span>length <span class="sc">&gt;=</span> <span class="dv">40</span>) <span class="sc">+</span> num_germline_40bpindels</span>
<span id="cb9-10"><a href="#cb9-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-11"><a href="#cb9-11" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="fu">sprintf</span>(<span class="st">"# nonsynonymous, non-intronic variants passing simple filtering: %s"</span>,</span>
<span id="cb9-12"><a href="#cb9-12" aria-hidden="true" tabindex="-1"></a>        num_simple_variants), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># nonsynonymous, non-intronic variants passing simple filtering: 265 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="fu">sprintf</span>(<span class="st">"# mutant samples passing simple filtering: %s"</span>,</span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>        num_simple_mutants), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># mutant samples passing simple filtering: 233 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="fu">sprintf</span>(<span class="st">"# indels &gt;=40bp passing simple filtering: %s"</span>,</span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a>        num_simple_40bpindels), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># indels &gt;=40bp passing simple filtering: 11 </code></pre>
</div>
</div>
<section id="figure-1a" class="level3">
<h3 class="anchored" data-anchor-id="figure-1a">Figure 1A</h3>
<p>This figure was made with screen captures from Integrative Genomics Viewer (IGV) views of the BAP1 gene locus near exon 5 for the TCGA-FV-A3I0 sample. Image was finalized in Adobe Illustrator 2021.</p>
</section>
<section id="figure-1b" class="level3">
<h3 class="anchored" data-anchor-id="figure-1b">Figure 1B</h3>
<p>This is a summarization of the number of variants in a schematic created by BioRender.com (<a href="https://www.biorender.com/" class="uri">https://www.biorender.com/</a>).</p>
<p>We then compare to the historical MC3 variant calls with simple filtering:</p>
<blockquote class="blockquote">
<p>This earlier pipeline resulted in the detection of 251 variants from 224 individuals, of which 190 variants from 170 individuals similarly passed simple call quality filtering, none of which were ≥40bp (Supplemental Figure S1 and Supplemental Table S2B and S2D).</p>
</blockquote>
<div class="cell">
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a>num_mc3 <span class="ot">&lt;-</span> <span class="fu">nrow</span>(mc3) <span class="sc">+</span> num_germline_variants</span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a>num_mc3_mutants <span class="ot">&lt;-</span> <span class="fu">length</span>(<span class="fu">unique</span>(mc3<span class="sc">$</span>SAMPLE)) <span class="sc">+</span> num_germline_mutants</span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a>num_mc3_40bpindels <span class="ot">&lt;-</span> <span class="fu">sum</span>(mc3<span class="sc">$</span>length <span class="sc">&gt;=</span> <span class="dv">40</span>) <span class="sc">+</span> num_germline_40bpindels</span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="fu">sprintf</span>(<span class="st">"# MC3 nonsynonymous, non-intronic variants passing simple filtering: %s"</span>, num_mc3), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># MC3 nonsynonymous, non-intronic variants passing simple filtering: 251 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="fu">sprintf</span>(<span class="st">"# MC3 mutant samples passing simple filtering: %s"</span>,</span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a>        num_mc3_mutants), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># MC3 mutant samples passing simple filtering: 224 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="fu">sprintf</span>(<span class="st">"# MC3 indels &gt;=40bp passing simple filtering: %s"</span>,</span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a>        num_mc3_40bpindels), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># MC3 indels &gt;=40bp passing simple filtering: 0 </code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a>num_mc3_simple <span class="ot">&lt;-</span> <span class="fu">nrow</span>(mc3_simple) <span class="sc">+</span> num_germline_variants</span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a>num_mc3_simple_mutants <span class="ot">&lt;-</span> <span class="fu">length</span>(<span class="fu">unique</span>(mc3_simple<span class="sc">$</span>SAMPLE)) <span class="sc">+</span> num_germline_mutants</span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a>num_mc3_simple_40bpindels <span class="ot">&lt;-</span> <span class="fu">sum</span>(mc3_simple<span class="sc">$</span>length <span class="sc">&gt;=</span> <span class="dv">40</span>) <span class="sc">+</span> num_germline_40bpindels</span>
<span id="cb21-4"><a href="#cb21-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-5"><a href="#cb21-5" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="fu">sprintf</span>(<span class="st">"# MC3 nonsynonymous, non-intronic variants passing simple filtering: %s"</span>, num_mc3_simple), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># MC3 nonsynonymous, non-intronic variants passing simple filtering: 190 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="fu">sprintf</span>(<span class="st">"# MC3 mutant samples passing simple filtering: %s"</span>,</span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a>        num_mc3_simple_mutants), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># MC3 mutant samples passing simple filtering: 170 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="fu">sprintf</span>(<span class="st">"# MC3 indels &gt;=40bp passing simple filtering: %s"</span>,</span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a>        num_mc3_simple_40bpindels), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># MC3 indels &gt;=40bp passing simple filtering: 0 </code></pre>
</div>
</div>
</section>
<section id="supplemental-figure-s1" class="level3">
<h3 class="anchored" data-anchor-id="supplemental-figure-s1">Supplemental Figure S1</h3>
<p>This is a supplemental supporting figure for figure 1. Image was finalized in Adobe Illustrator 2021.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="fu">boxplot</span>(mc3_simple<span class="sc">$</span>length, </span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a>        variants_simple<span class="sc">$</span>length, </span>
<span id="cb27-3"><a href="#cb27-3" aria-hidden="true" tabindex="-1"></a>        <span class="at">outline =</span> <span class="cn">FALSE</span>, </span>
<span id="cb27-4"><a href="#cb27-4" aria-hidden="true" tabindex="-1"></a>        <span class="at">ylim =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="fu">max</span>(variants_simple<span class="sc">$</span>length)), </span>
<span id="cb27-5"><a href="#cb27-5" aria-hidden="true" tabindex="-1"></a>        <span class="at">names =</span> <span class="fu">c</span>(<span class="st">"TCGA MC3"</span>, <span class="st">"New"</span>), </span>
<span id="cb27-6"><a href="#cb27-6" aria-hidden="true" tabindex="-1"></a>        <span class="at">ylab =</span> <span class="st">"Variant length (bp)"</span>)</span>
<span id="cb27-7"><a href="#cb27-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-8"><a href="#cb27-8" aria-hidden="true" tabindex="-1"></a><span class="fu">points</span>(<span class="fu">jitter</span>(<span class="fu">rep</span>(<span class="dv">1</span>, <span class="fu">nrow</span>(mc3_simple))),</span>
<span id="cb27-9"><a href="#cb27-9" aria-hidden="true" tabindex="-1"></a>       mc3_simple<span class="sc">$</span>length,</span>
<span id="cb27-10"><a href="#cb27-10" aria-hidden="true" tabindex="-1"></a>       <span class="at">pch =</span> <span class="dv">16</span>)</span>
<span id="cb27-11"><a href="#cb27-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-12"><a href="#cb27-12" aria-hidden="true" tabindex="-1"></a><span class="fu">points</span>(<span class="fu">jitter</span>(<span class="fu">rep</span>(<span class="dv">2</span>, <span class="fu">nrow</span>(variants_simple))),</span>
<span id="cb27-13"><a href="#cb27-13" aria-hidden="true" tabindex="-1"></a>       variants_simple<span class="sc">$</span>length,</span>
<span id="cb27-14"><a href="#cb27-14" aria-hidden="true" tabindex="-1"></a>       <span class="at">pch =</span> <span class="dv">16</span>)</span>
<span id="cb27-15"><a href="#cb27-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-16"><a href="#cb27-16" aria-hidden="true" tabindex="-1"></a><span class="fu">abline</span>(<span class="at">h =</span> <span class="dv">40</span>, <span class="at">lty =</span> <span class="st">"dotted"</span>, <span class="at">col =</span> <span class="st">"darkgrey"</span>) <span class="co">#40bp threshold</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="figure_code_files/figure-html/unnamed-chunk-7-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>We want to have a reasonable idea of whether there are shared variant calls across the new and historical data. Although there is a complication from hg19 vs.&nbsp;hg38 alignment, we can say that a variant is shared if both the variant allele and the sample ID are identical.</p>
<blockquote class="blockquote">
<p>Of the 182 somatic variants in MC3, 77 (30%) were new calls and the remaining 180 were concordant (Figure 1C).</p>
</blockquote>
<div class="cell">
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a>shared <span class="ot">&lt;-</span> <span class="fu">c</span>()</span>
<span id="cb28-2"><a href="#cb28-2" aria-hidden="true" tabindex="-1"></a>shared_rownum <span class="ot">&lt;-</span> <span class="fu">c</span>()</span>
<span id="cb28-3"><a href="#cb28-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-4"><a href="#cb28-4" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(variants_simple)) {</span>
<span id="cb28-5"><a href="#cb28-5" aria-hidden="true" tabindex="-1"></a>  compare_id <span class="ot">&lt;-</span> variants_simple<span class="sc">$</span>SAMPLE[i]</span>
<span id="cb28-6"><a href="#cb28-6" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (compare_id <span class="sc">%in%</span> mc3_simple<span class="sc">$</span>SAMPLE) {</span>
<span id="cb28-7"><a href="#cb28-7" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (variants_simple<span class="sc">$</span>variant_allele[i] <span class="sc">%in%</span> </span>
<span id="cb28-8"><a href="#cb28-8" aria-hidden="true" tabindex="-1"></a>        mc3_simple<span class="sc">$</span>variant_allele[mc3_simple<span class="sc">$</span>SAMPLE <span class="sc">==</span> compare_id]) {</span>
<span id="cb28-9"><a href="#cb28-9" aria-hidden="true" tabindex="-1"></a>      shared <span class="ot">&lt;-</span> <span class="fu">c</span>(shared, compare_id)</span>
<span id="cb28-10"><a href="#cb28-10" aria-hidden="true" tabindex="-1"></a>      shared_rownum <span class="ot">&lt;-</span> <span class="fu">c</span>(shared_rownum, i)</span>
<span id="cb28-11"><a href="#cb28-11" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb28-12"><a href="#cb28-12" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb28-13"><a href="#cb28-13" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb28-14"><a href="#cb28-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-15"><a href="#cb28-15" aria-hidden="true" tabindex="-1"></a>num_shared_variants <span class="ot">&lt;-</span> <span class="fu">length</span>(shared)</span>
<span id="cb28-16"><a href="#cb28-16" aria-hidden="true" tabindex="-1"></a>num_new_unique <span class="ot">&lt;-</span> <span class="fu">nrow</span>(variants_simple) <span class="sc">-</span> num_shared_variants</span>
<span id="cb28-17"><a href="#cb28-17" aria-hidden="true" tabindex="-1"></a>percent_new_unique <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="fu">round</span>((num_new_unique <span class="sc">/</span> <span class="fu">nrow</span>(variants_simple)) <span class="sc">*</span> <span class="dv">100</span>, </span>
<span id="cb28-18"><a href="#cb28-18" aria-hidden="true" tabindex="-1"></a>                                   <span class="dv">2</span>), <span class="st">"%"</span>)</span>
<span id="cb28-19"><a href="#cb28-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-20"><a href="#cb28-20" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="fu">sprintf</span>(<span class="st">"# new variant calls: %s (%s)"</span>,</span>
<span id="cb28-21"><a href="#cb28-21" aria-hidden="true" tabindex="-1"></a>        num_new_unique, percent_new_unique), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># new variant calls: 77 (29.96%) </code></pre>
</div>
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="fu">sprintf</span>(<span class="st">"# shared variant calls between new and MC3: %s"</span>,</span>
<span id="cb30-2"><a href="#cb30-2" aria-hidden="true" tabindex="-1"></a>        num_shared_variants), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># shared variant calls between new and MC3: 180 </code></pre>
</div>
</div>
<p>We can also pull up the details for the 1 variant in MC3 that was not detected (or detected differently) in the new variant call dataset:</p>
<blockquote class="blockquote">
<p>All MC3 variants and mutant samples were captured in the new approach, except one variant, a 5bp deletion in KIRP sample TCGA-2Z-A9JD (Supplemental Tables S2C and S2D). However, our realignment approach detected two single-nucleotide variants (deletion and missense) in the similar region[…]</p>
</blockquote>
<div class="cell">
<div class="sourceCode cell-code" id="cb32"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a><span class="fu">data.frame</span>(<span class="fu">rbind</span>(mc3_simple[mc3_simple<span class="sc">$</span>SAMPLE <span class="sc">==</span> <span class="st">"TCGA-2Z-A9JD"</span>, </span>
<span id="cb32-2"><a href="#cb32-2" aria-hidden="true" tabindex="-1"></a>                           <span class="fu">c</span>(<span class="st">"Start_Position"</span>, <span class="st">"End_Position"</span>, </span>
<span id="cb32-3"><a href="#cb32-3" aria-hidden="true" tabindex="-1"></a>                             <span class="st">"variant_allele"</span>, <span class="st">"Variant_Classification"</span>)],</span>
<span id="cb32-4"><a href="#cb32-4" aria-hidden="true" tabindex="-1"></a>                variants_simple[variants_simple<span class="sc">$</span>SAMPLE <span class="sc">==</span> <span class="st">"TCGA-2Z-A9JD"</span>, </span>
<span id="cb32-5"><a href="#cb32-5" aria-hidden="true" tabindex="-1"></a>                               <span class="fu">c</span>(<span class="st">"Start_Position"</span>, <span class="st">"End_Position"</span>, </span>
<span id="cb32-6"><a href="#cb32-6" aria-hidden="true" tabindex="-1"></a>                                 <span class="st">"variant_allele"</span>, <span class="st">"Variant_Classification"</span>)]),</span>
<span id="cb32-7"><a href="#cb32-7" aria-hidden="true" tabindex="-1"></a>           <span class="at">row.names =</span> <span class="fu">c</span>(<span class="st">"MC3_variant"</span>, <span class="st">"New_variant1"</span>, <span class="st">"New_variant2"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>             Start_Position End_Position variant_allele Variant_Classification
MC3_variant        52437652     52437657          AAAGG        Frame_Shift_Del
New_variant1       52403636     52403636              -        Frame_Shift_Del
New_variant2       52403641     52403641              G      Missense_Mutation</code></pre>
</div>
</div>
</section>
<section id="figure-1c" class="level3">
<h3 class="anchored" data-anchor-id="figure-1c">Figure 1C</h3>
<p>Here we plot % representation of variant classifications among mutatations ordered from highest to lowest, with numbers above their respective bars. If there were no variants of a particular classification, the classification was not included. Image below was finalized in Adobe Illustrator 2021.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb34"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a>new_ids <span class="ot">&lt;-</span> variants_simple<span class="sc">$</span>SAMPLE[<span class="sc">-</span>shared_rownum]</span>
<span id="cb34-2"><a href="#cb34-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-3"><a href="#cb34-3" aria-hidden="true" tabindex="-1"></a>variant_df <span class="ot">&lt;-</span> <span class="fu">table</span>(variants_simple<span class="sc">$</span>Variant_Classification, variants_simple<span class="sc">$</span>SAMPLE <span class="sc">%in%</span> shared)</span>
<span id="cb34-4"><a href="#cb34-4" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(variant_df) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"New calls"</span>, <span class="st">"TCGA MC3 calls"</span>)</span>
<span id="cb34-5"><a href="#cb34-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-6"><a href="#cb34-6" aria-hidden="true" tabindex="-1"></a>variant_df <span class="ot">&lt;-</span> variant_df[<span class="fu">order</span>(<span class="fu">rowSums</span>(variant_df), <span class="at">decreasing =</span> <span class="cn">TRUE</span>),</span>
<span id="cb34-7"><a href="#cb34-7" aria-hidden="true" tabindex="-1"></a>                         <span class="fu">c</span>(<span class="st">"TCGA MC3 calls"</span>, <span class="st">"New calls"</span>)]</span>
<span id="cb34-8"><a href="#cb34-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-9"><a href="#cb34-9" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(variant_df) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"_"</span>, <span class="st">" "</span>, <span class="fu">rownames</span>(variant_df))</span>
<span id="cb34-10"><a href="#cb34-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-11"><a href="#cb34-11" aria-hidden="true" tabindex="-1"></a>num_variants <span class="ot">&lt;-</span> <span class="fu">length</span>(new_ids) <span class="sc">+</span> <span class="fu">length</span>(shared)</span>
<span id="cb34-12"><a href="#cb34-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-13"><a href="#cb34-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Plotting % representation of variants split by old/shared vs new-unique calls</span></span>
<span id="cb34-14"><a href="#cb34-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Numbers indicated on top</span></span>
<span id="cb34-15"><a href="#cb34-15" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mar =</span> <span class="fu">c</span>(<span class="dv">8</span>, <span class="dv">4</span>, <span class="dv">4</span>, <span class="dv">2</span>))</span>
<span id="cb34-16"><a href="#cb34-16" aria-hidden="true" tabindex="-1"></a>class_barplot <span class="ot">&lt;-</span> <span class="fu">barplot</span>(<span class="fu">t</span>(variant_df) <span class="sc">*</span> (<span class="dv">100</span><span class="sc">/</span>num_variants), </span>
<span id="cb34-17"><a href="#cb34-17" aria-hidden="true" tabindex="-1"></a>                         <span class="at">las =</span> <span class="dv">2</span>, </span>
<span id="cb34-18"><a href="#cb34-18" aria-hidden="true" tabindex="-1"></a>                         <span class="at">cex.names =</span> <span class="fl">0.7</span>,</span>
<span id="cb34-19"><a href="#cb34-19" aria-hidden="true" tabindex="-1"></a>                         <span class="at">ylab =</span> <span class="st">"% of total somatic variants"</span>,</span>
<span id="cb34-20"><a href="#cb34-20" aria-hidden="true" tabindex="-1"></a>                         <span class="at">col =</span> <span class="fu">c</span>(<span class="st">"grey"</span>, <span class="st">"steelblue"</span>),</span>
<span id="cb34-21"><a href="#cb34-21" aria-hidden="true" tabindex="-1"></a>                         <span class="at">xlim =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">10</span>), <span class="at">ylim =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">50</span>),</span>
<span id="cb34-22"><a href="#cb34-22" aria-hidden="true" tabindex="-1"></a>                         <span class="at">width =</span> <span class="fu">rep</span>(<span class="fl">0.5</span>, <span class="dv">10</span>))</span>
<span id="cb34-23"><a href="#cb34-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-24"><a href="#cb34-24" aria-hidden="true" tabindex="-1"></a>plot_labs <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"n="</span>, <span class="fu">colSums</span>(<span class="fu">t</span>(variant_df)))</span>
<span id="cb34-25"><a href="#cb34-25" aria-hidden="true" tabindex="-1"></a><span class="fu">text</span>(<span class="at">x =</span> class_barplot, <span class="at">y =</span> <span class="fu">sort</span>(<span class="fu">colSums</span>(<span class="fu">t</span>(variant_df) <span class="sc">*</span> (<span class="dv">100</span><span class="sc">/</span>num_variants)), </span>
<span id="cb34-26"><a href="#cb34-26" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">decreasing =</span> <span class="cn">TRUE</span>),</span>
<span id="cb34-27"><a href="#cb34-27" aria-hidden="true" tabindex="-1"></a>     <span class="at">label =</span> plot_labs,</span>
<span id="cb34-28"><a href="#cb34-28" aria-hidden="true" tabindex="-1"></a>     <span class="at">pos =</span> <span class="dv">3</span>, <span class="at">cex =</span> <span class="fl">0.8</span>, <span class="at">col =</span> <span class="st">"black"</span>, <span class="at">offset =</span> <span class="fl">0.7</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="figure_code_files/figure-html/unnamed-chunk-10-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>From the text:</p>
<blockquote class="blockquote">
<p>The majority of <em>BAP1</em> somatic variants were missense mutations (110/257, 42.8%), followed by deleterious nonsense and frameshift events (94/257, 36.6%, Figure 1C and Supplemental Table S2C).</p>
</blockquote>
<div class="cell">
<div class="sourceCode cell-code" id="cb35"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a>num_nonsense_frameshift <span class="ot">&lt;-</span> <span class="fu">sum</span>(variants_simple<span class="sc">$</span>Variant_Classification <span class="sc">%in%</span> </span>
<span id="cb35-2"><a href="#cb35-2" aria-hidden="true" tabindex="-1"></a>                                 <span class="fu">c</span>(<span class="st">"Nonsense_Mutation"</span>, <span class="st">"Frame_Shift_Del"</span>,</span>
<span id="cb35-3"><a href="#cb35-3" aria-hidden="true" tabindex="-1"></a>                                   <span class="st">"Frame_Shift_Ins"</span>))</span>
<span id="cb35-4"><a href="#cb35-4" aria-hidden="true" tabindex="-1"></a>percent_nonsense_frameshift <span class="ot">&lt;-</span> <span class="fu">round</span>((num_nonsense_frameshift <span class="sc">/</span> <span class="fu">nrow</span>(variants_simple)) <span class="sc">*</span> <span class="dv">100</span>, <span class="dv">1</span>)</span>
<span id="cb35-5"><a href="#cb35-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-6"><a href="#cb35-6" aria-hidden="true" tabindex="-1"></a>num_missense <span class="ot">&lt;-</span> <span class="fu">sum</span>(variants_simple<span class="sc">$</span>Variant_Classification <span class="sc">==</span> <span class="st">"Missense_Mutation"</span>)</span>
<span id="cb35-7"><a href="#cb35-7" aria-hidden="true" tabindex="-1"></a>percent_missense <span class="ot">&lt;-</span> <span class="fu">round</span>((num_missense <span class="sc">/</span> <span class="fu">nrow</span>(variants_simple)) <span class="sc">*</span> <span class="dv">100</span>, <span class="dv">1</span>)</span>
<span id="cb35-8"><a href="#cb35-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-9"><a href="#cb35-9" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="fu">sprintf</span>(<span class="st">"# missense variants: %s (%s)"</span>,</span>
<span id="cb35-10"><a href="#cb35-10" aria-hidden="true" tabindex="-1"></a>            num_missense, <span class="fu">paste0</span>(percent_missense, <span class="st">"%"</span>)), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># missense variants: 110 (42.8%) </code></pre>
</div>
<div class="sourceCode cell-code" id="cb37"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="fu">sprintf</span>(<span class="st">"# nonsense and frameshift variants: %s (%s)"</span>, </span>
<span id="cb37-2"><a href="#cb37-2" aria-hidden="true" tabindex="-1"></a>            num_nonsense_frameshift, <span class="fu">paste0</span>(percent_nonsense_frameshift, <span class="st">"%"</span>)), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># nonsense and frameshift variants: 94 (36.6%) </code></pre>
</div>
</div>
<blockquote class="blockquote">
<p>Of the cancer types represented in TCGA, 21/33 (63.6%) have at least one BAP1 mutant sample (Figure 1C). Only three tumor types had &gt;5% mutation frequency: uveal melanoma (UVM), malignant mesothelioma (MESO), and cholangiocarcinoma (CHOL) (Supplemental Table S3A).</p>
</blockquote>
<div class="cell">
<div class="sourceCode cell-code" id="cb39"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" aria-hidden="true" tabindex="-1"></a>num_mutant_cohorts <span class="ot">&lt;-</span> <span class="fu">length</span>(<span class="fu">unique</span>(variants_simple<span class="sc">$</span>Cancer_Type))</span>
<span id="cb39-2"><a href="#cb39-2" aria-hidden="true" tabindex="-1"></a>percent_mutant_cohorts <span class="ot">&lt;-</span> <span class="fu">round</span>(num_mutant_cohorts <span class="sc">/</span> <span class="fu">length</span>(tcga_projects) <span class="sc">*</span> <span class="dv">100</span>, <span class="dv">1</span>)</span>
<span id="cb39-3"><a href="#cb39-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-4"><a href="#cb39-4" aria-hidden="true" tabindex="-1"></a>types_5freq <span class="ot">&lt;-</span> cohort_anno<span class="sc">$</span>Cohort[(cohort_anno<span class="sc">$</span>Mutation.only <span class="sc">+</span> cohort_anno<span class="sc">$</span>Mutation.Copynumber) <span class="sc">/</span></span>
<span id="cb39-5"><a href="#cb39-5" aria-hidden="true" tabindex="-1"></a>                         cohort_anno<span class="sc">$</span>Cohort.size <span class="sc">&gt;</span> <span class="fl">0.05</span>]</span>
<span id="cb39-6"><a href="#cb39-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-7"><a href="#cb39-7" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="fu">sprintf</span>(<span class="st">"# cancer types with mutant samples: %s (%s)"</span>, </span>
<span id="cb39-8"><a href="#cb39-8" aria-hidden="true" tabindex="-1"></a>            num_mutant_cohorts, <span class="fu">paste0</span>(percent_mutant_cohorts, <span class="st">"%"</span>)), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># cancer types with mutant samples: 21 (63.6%) </code></pre>
</div>
<div class="sourceCode cell-code" id="cb41"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="fu">sprintf</span>(<span class="st">"Cancer types with &gt;5 percent mutation frequency: %s"</span>,</span>
<span id="cb41-2"><a href="#cb41-2" aria-hidden="true" tabindex="-1"></a>            <span class="fu">paste</span>(types_5freq, <span class="at">collapse =</span> <span class="st">", "</span>)), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Cancer types with &gt;5 percent mutation frequency: KIRC, CHOL, MESO, UVM, UCEC, LIHC </code></pre>
</div>
</div>
</section>
<section id="figure-1d" class="level3">
<h3 class="anchored" data-anchor-id="figure-1d">Figure 1D</h3>
<p>Here we plot % mutant samples per cohort ordered from highest to lowest, with numbers above their respective bars. If there were no mutations, the cohort was not included. Image below was finalized in Adobe Illustrator 2021.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb43"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1" aria-hidden="true" tabindex="-1"></a>num_mutant <span class="ot">&lt;-</span> <span class="fu">c</span>()</span>
<span id="cb43-2"><a href="#cb43-2" aria-hidden="true" tabindex="-1"></a>percent_mutant <span class="ot">&lt;-</span> <span class="fu">c</span>()</span>
<span id="cb43-3"><a href="#cb43-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-4"><a href="#cb43-4" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> tcga_projects) {</span>
<span id="cb43-5"><a href="#cb43-5" aria-hidden="true" tabindex="-1"></a>  n_mut <span class="ot">&lt;-</span> <span class="fu">sum</span>(<span class="fu">grepl</span>(<span class="st">"mutation"</span>, sample_anno<span class="sc">$</span>alteration_type[sample_anno<span class="sc">$</span>cohort <span class="sc">==</span> i]),</span>
<span id="cb43-6"><a href="#cb43-6" aria-hidden="true" tabindex="-1"></a>               <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb43-7"><a href="#cb43-7" aria-hidden="true" tabindex="-1"></a>  n_cohort <span class="ot">&lt;-</span> <span class="fu">sum</span>(sample_anno<span class="sc">$</span>cohort <span class="sc">==</span> i)</span>
<span id="cb43-8"><a href="#cb43-8" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb43-9"><a href="#cb43-9" aria-hidden="true" tabindex="-1"></a>  num_mutant <span class="ot">&lt;-</span> <span class="fu">c</span>(num_mutant, n_mut)</span>
<span id="cb43-10"><a href="#cb43-10" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb43-11"><a href="#cb43-11" aria-hidden="true" tabindex="-1"></a>  percent_mutant <span class="ot">&lt;-</span> <span class="fu">c</span>(percent_mutant, (n_mut<span class="sc">/</span>n_cohort)<span class="sc">*</span><span class="dv">100</span>)</span>
<span id="cb43-12"><a href="#cb43-12" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb43-13"><a href="#cb43-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-14"><a href="#cb43-14" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(num_mutant) <span class="ot">&lt;-</span> tcga_names</span>
<span id="cb43-15"><a href="#cb43-15" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(percent_mutant) <span class="ot">&lt;-</span> tcga_names</span>
<span id="cb43-16"><a href="#cb43-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-17"><a href="#cb43-17" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove cancer types which have 0 mutant samples</span></span>
<span id="cb43-18"><a href="#cb43-18" aria-hidden="true" tabindex="-1"></a>num_mutant <span class="ot">&lt;-</span> num_mutant[num_mutant <span class="sc">!=</span> <span class="dv">0</span>]</span>
<span id="cb43-19"><a href="#cb43-19" aria-hidden="true" tabindex="-1"></a>percent_mutant <span class="ot">&lt;-</span> percent_mutant[percent_mutant <span class="sc">!=</span> <span class="dv">0</span>]</span>
<span id="cb43-20"><a href="#cb43-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-21"><a href="#cb43-21" aria-hidden="true" tabindex="-1"></a>tcga_col <span class="ot">&lt;-</span> tcga_colors<span class="sc">$</span>Hex.Colors</span>
<span id="cb43-22"><a href="#cb43-22" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(tcga_col) <span class="ot">&lt;-</span> tcga_colors<span class="sc">$</span>Study.Abbreviation</span>
<span id="cb43-23"><a href="#cb43-23" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mar =</span> <span class="fu">c</span>(<span class="dv">8</span>, <span class="dv">4</span>, <span class="dv">4</span>, <span class="dv">2</span>))</span>
<span id="cb43-24"><a href="#cb43-24" aria-hidden="true" tabindex="-1"></a>mut_barplot <span class="ot">&lt;-</span> <span class="fu">barplot</span>(<span class="fu">sort</span>(percent_mutant, <span class="at">decreasing =</span> <span class="cn">TRUE</span>),</span>
<span id="cb43-25"><a href="#cb43-25" aria-hidden="true" tabindex="-1"></a>                       <span class="at">las =</span> <span class="dv">2</span>, </span>
<span id="cb43-26"><a href="#cb43-26" aria-hidden="true" tabindex="-1"></a>                       <span class="at">cex.names =</span> <span class="fl">0.7</span>, </span>
<span id="cb43-27"><a href="#cb43-27" aria-hidden="true" tabindex="-1"></a>                       <span class="at">col =</span> tcga_col[<span class="fu">names</span>(<span class="fu">sort</span>(percent_mutant, <span class="at">decreasing =</span> <span class="cn">TRUE</span>))], </span>
<span id="cb43-28"><a href="#cb43-28" aria-hidden="true" tabindex="-1"></a>                       <span class="at">names.arg =</span> <span class="fu">names</span>(<span class="fu">sort</span>(percent_mutant, <span class="at">decreasing =</span> <span class="cn">TRUE</span>)),</span>
<span id="cb43-29"><a href="#cb43-29" aria-hidden="true" tabindex="-1"></a>                       <span class="at">ylab =</span> <span class="st">"% cohort with BAP1 mutation"</span>, </span>
<span id="cb43-30"><a href="#cb43-30" aria-hidden="true" tabindex="-1"></a>                       <span class="at">ylim =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">50</span>))</span>
<span id="cb43-31"><a href="#cb43-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-32"><a href="#cb43-32" aria-hidden="true" tabindex="-1"></a>plot_labs <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"n="</span>, num_mutant[<span class="fu">order</span>(percent_mutant, <span class="at">decreasing =</span> <span class="cn">TRUE</span>)])</span>
<span id="cb43-33"><a href="#cb43-33" aria-hidden="true" tabindex="-1"></a><span class="fu">text</span>(<span class="at">x =</span> mut_barplot, </span>
<span id="cb43-34"><a href="#cb43-34" aria-hidden="true" tabindex="-1"></a>     <span class="at">y =</span> <span class="fu">sort</span>(percent_mutant, <span class="at">decreasing =</span> <span class="cn">TRUE</span>), </span>
<span id="cb43-35"><a href="#cb43-35" aria-hidden="true" tabindex="-1"></a>     <span class="at">label =</span> plot_labs, </span>
<span id="cb43-36"><a href="#cb43-36" aria-hidden="true" tabindex="-1"></a>     <span class="at">pos =</span> <span class="dv">3</span>, </span>
<span id="cb43-37"><a href="#cb43-37" aria-hidden="true" tabindex="-1"></a>     <span class="at">cex =</span> <span class="fl">0.8</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="figure_code_files/figure-html/unnamed-chunk-13-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>Next from the text:</p>
<blockquote class="blockquote">
<p>Somatic variants were broadly distributed across the length of the gene with no significant difference in protein-level domain representation for missense versus other mutation types (two-sided Fisher’s exact test p=0.1) (Figure 1E and Supplemental Table S2C).</p>
</blockquote>
<div class="cell">
<div class="sourceCode cell-code" id="cb44"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb44-1"><a href="#cb44-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Annotated BAP1 domains from Haugh et al. (JAMA 2017), Figure 4</span></span>
<span id="cb44-2"><a href="#cb44-2" aria-hidden="true" tabindex="-1"></a>uch <span class="ot">&lt;-</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">240</span></span>
<span id="cb44-3"><a href="#cb44-3" aria-hidden="true" tabindex="-1"></a>bard1 <span class="ot">&lt;-</span> <span class="dv">182</span><span class="sc">:</span><span class="dv">365</span></span>
<span id="cb44-4"><a href="#cb44-4" aria-hidden="true" tabindex="-1"></a>hbm <span class="ot">&lt;-</span> <span class="dv">363</span><span class="sc">:</span><span class="dv">366</span></span>
<span id="cb44-5"><a href="#cb44-5" aria-hidden="true" tabindex="-1"></a>brca1 <span class="ot">&lt;-</span> <span class="dv">594</span><span class="sc">:</span><span class="dv">657</span></span>
<span id="cb44-6"><a href="#cb44-6" aria-hidden="true" tabindex="-1"></a>nls <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">656</span><span class="sc">:</span><span class="dv">661</span>, <span class="dv">717</span><span class="sc">:</span><span class="dv">722</span>)</span>
<span id="cb44-7"><a href="#cb44-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-8"><a href="#cb44-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Find overlap of protein positions and annotated domains</span></span>
<span id="cb44-9"><a href="#cb44-9" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(variants_simple)) {</span>
<span id="cb44-10"><a href="#cb44-10" aria-hidden="true" tabindex="-1"></a>  protein_position <span class="ot">&lt;-</span> variants_simple<span class="sc">$</span>Protein_position[i]</span>
<span id="cb44-11"><a href="#cb44-11" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (protein_position <span class="sc">==</span> <span class="st">""</span>) {</span>
<span id="cb44-12"><a href="#cb44-12" aria-hidden="true" tabindex="-1"></a>    variants_simple<span class="sc">$</span>domain[i] <span class="ot">&lt;-</span> <span class="cn">NA</span></span>
<span id="cb44-13"><a href="#cb44-13" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb44-14"><a href="#cb44-14" aria-hidden="true" tabindex="-1"></a>  <span class="cf">else</span> {</span>
<span id="cb44-15"><a href="#cb44-15" aria-hidden="true" tabindex="-1"></a>    protein_position <span class="ot">&lt;-</span> <span class="fu">strsplit</span>(protein_position, <span class="st">"/"</span>)[[<span class="dv">1</span>]][<span class="dv">1</span>]</span>
<span id="cb44-16"><a href="#cb44-16" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="fu">grepl</span>(<span class="st">"-"</span>, protein_position)) {</span>
<span id="cb44-17"><a href="#cb44-17" aria-hidden="true" tabindex="-1"></a>      propos1 <span class="ot">&lt;-</span> <span class="fu">strsplit</span>(protein_position, <span class="st">"-"</span>)[[<span class="dv">1</span>]][<span class="dv">1</span>]</span>
<span id="cb44-18"><a href="#cb44-18" aria-hidden="true" tabindex="-1"></a>      propos2 <span class="ot">&lt;-</span> <span class="fu">strsplit</span>(protein_position, <span class="st">"-"</span>)[[<span class="dv">1</span>]][<span class="dv">2</span>]</span>
<span id="cb44-19"><a href="#cb44-19" aria-hidden="true" tabindex="-1"></a>      propos1 <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(propos1 <span class="sc">==</span> <span class="st">"?"</span>, propos2, propos1)</span>
<span id="cb44-20"><a href="#cb44-20" aria-hidden="true" tabindex="-1"></a>      propos2 <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(propos2 <span class="sc">==</span> <span class="st">"?"</span>, propos1, propos2)</span>
<span id="cb44-21"><a href="#cb44-21" aria-hidden="true" tabindex="-1"></a>      protein_position <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(propos1<span class="sc">:</span>propos2)</span>
<span id="cb44-22"><a href="#cb44-22" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb44-23"><a href="#cb44-23" aria-hidden="true" tabindex="-1"></a>    variants_simple<span class="sc">$</span>domain[i] <span class="ot">&lt;-</span> </span>
<span id="cb44-24"><a href="#cb44-24" aria-hidden="true" tabindex="-1"></a>      <span class="fu">ifelse</span>(<span class="fu">any</span>(protein_position <span class="sc">%in%</span> uch) <span class="sc">&amp;</span> </span>
<span id="cb44-25"><a href="#cb44-25" aria-hidden="true" tabindex="-1"></a>               <span class="fu">any</span>(protein_position <span class="sc">%in%</span> bard1), <span class="st">"UCH+BARD1"</span>,</span>
<span id="cb44-26"><a href="#cb44-26" aria-hidden="true" tabindex="-1"></a>             <span class="fu">ifelse</span>(<span class="fu">any</span>(protein_position <span class="sc">%in%</span> uch), <span class="st">"UCH"</span>,</span>
<span id="cb44-27"><a href="#cb44-27" aria-hidden="true" tabindex="-1"></a>                    <span class="fu">ifelse</span>(<span class="fu">any</span>(protein_position <span class="sc">%in%</span> bard1) <span class="sc">&amp;</span></span>
<span id="cb44-28"><a href="#cb44-28" aria-hidden="true" tabindex="-1"></a>                             <span class="fu">any</span>(protein_position <span class="sc">%in%</span> hbm), <span class="st">"BARD1+HBM"</span>,</span>
<span id="cb44-29"><a href="#cb44-29" aria-hidden="true" tabindex="-1"></a>                           <span class="fu">ifelse</span>(<span class="fu">any</span>(protein_position <span class="sc">%in%</span> bard1), <span class="st">"BARD1"</span>,</span>
<span id="cb44-30"><a href="#cb44-30" aria-hidden="true" tabindex="-1"></a>                                  <span class="fu">ifelse</span>(<span class="fu">any</span>(protein_position <span class="sc">%in%</span> hbm), <span class="st">"HBM"</span>,</span>
<span id="cb44-31"><a href="#cb44-31" aria-hidden="true" tabindex="-1"></a>                                         <span class="fu">ifelse</span>(<span class="fu">any</span>(protein_position <span class="sc">%in%</span> brca1) <span class="sc">&amp;</span></span>
<span id="cb44-32"><a href="#cb44-32" aria-hidden="true" tabindex="-1"></a>                                               <span class="fu">any</span>(protein_position <span class="sc">%in%</span> nls), <span class="st">"BRCA1+NLS"</span>, </span>
<span id="cb44-33"><a href="#cb44-33" aria-hidden="true" tabindex="-1"></a>                                               <span class="fu">ifelse</span>(<span class="fu">any</span>(protein_position <span class="sc">%in%</span> brca1), <span class="st">"BRCA1"</span>,</span>
<span id="cb44-34"><a href="#cb44-34" aria-hidden="true" tabindex="-1"></a>                                                          <span class="fu">ifelse</span>(<span class="fu">any</span>(protein_position <span class="sc">%in%</span> nls), <span class="st">"NLS"</span>, <span class="st">"Other"</span>))))))))</span>
<span id="cb44-35"><a href="#cb44-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-36"><a href="#cb44-36" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb44-37"><a href="#cb44-37" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb44-38"><a href="#cb44-38" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(variants_simple<span class="sc">$</span>domain, variants_simple<span class="sc">$</span>Mutation_Type)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>           
            Missense_Mutation Other_Mutation
  BARD1                    12             20
  BARD1+HBM                 1              0
  BRCA1                     4              9
  BRCA1+NLS                 1              0
  NLS                       2              2
  Other                    28             45
  UCH                      46             37
  UCH+BARD1                16             11</code></pre>
</div>
<div class="sourceCode cell-code" id="cb46"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb46-1"><a href="#cb46-1" aria-hidden="true" tabindex="-1"></a>variants_fisher <span class="ot">&lt;-</span> <span class="fu">fisher.test</span>(<span class="fu">table</span>(variants_simple<span class="sc">$</span>domain, variants_simple<span class="sc">$</span>Mutation_Type))</span>
<span id="cb46-2"><a href="#cb46-2" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="fu">sprintf</span>(<span class="st">"Two-sided Fisher's exact test p=%s"</span>,</span>
<span id="cb46-3"><a href="#cb46-3" aria-hidden="true" tabindex="-1"></a>            <span class="fu">round</span>(variants_fisher<span class="sc">$</span>p.value, <span class="dv">2</span>)))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Two-sided Fisher's exact test p=0.1</code></pre>
</div>
</div>
<blockquote class="blockquote">
<p>We observed 161/257 (62.6%) variants occurred at an annotated functional domain with 110/257 (42.8%) occurring at the catalytic UCH domain (Figure 1E and Supplemental Table S2C).</p>
</blockquote>
<div class="cell">
<div class="sourceCode cell-code" id="cb48"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb48-1"><a href="#cb48-1" aria-hidden="true" tabindex="-1"></a>num_functional <span class="ot">&lt;-</span> <span class="fu">sum</span>(variants_simple<span class="sc">$</span>domain <span class="sc">!=</span> <span class="st">"Other"</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb48-2"><a href="#cb48-2" aria-hidden="true" tabindex="-1"></a>percent_functional <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="fu">round</span>((num_functional <span class="sc">/</span> <span class="fu">nrow</span>(variants_simple)) <span class="sc">*</span> <span class="dv">100</span>, <span class="dv">1</span>), <span class="st">"%"</span>)</span>
<span id="cb48-3"><a href="#cb48-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-4"><a href="#cb48-4" aria-hidden="true" tabindex="-1"></a>num_uch <span class="ot">&lt;-</span> <span class="fu">sum</span>(variants_simple<span class="sc">$</span>domain <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"UCH"</span>, <span class="st">"UCH+BARD1"</span>))</span>
<span id="cb48-5"><a href="#cb48-5" aria-hidden="true" tabindex="-1"></a>percent_uch <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="fu">round</span>((num_uch <span class="sc">/</span> <span class="fu">nrow</span>(variants_simple)) <span class="sc">*</span> <span class="dv">100</span>, <span class="dv">1</span>), <span class="st">"%"</span>)</span>
<span id="cb48-6"><a href="#cb48-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-7"><a href="#cb48-7" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="fu">sprintf</span>(<span class="st">"# variants occurring at functional domain: %s (%s)"</span>, </span>
<span id="cb48-8"><a href="#cb48-8" aria-hidden="true" tabindex="-1"></a>            num_functional, percent_functional), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># variants occurring at functional domain: 161 (62.6%) </code></pre>
</div>
<div class="sourceCode cell-code" id="cb50"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb50-1"><a href="#cb50-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="fu">sprintf</span>(<span class="st">"# variants occurring at UCH domain: %s (%s)"</span>, </span>
<span id="cb50-2"><a href="#cb50-2" aria-hidden="true" tabindex="-1"></a>            num_uch, percent_uch), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># variants occurring at UCH domain: 110 (42.8%) </code></pre>
</div>
</div>
</section>
<section id="figure-1e" class="level3">
<h3 class="anchored" data-anchor-id="figure-1e">Figure 1E</h3>
<p>This figure was made outside of R by exporting the 265 variant calls into a readable format for the cBioPortal MutationMapper (<a href="https://www.cbioportal.org/mutation_mapper)." class="uri">https://www.cbioportal.org/mutation_mapper).</a> Please note that GRCh38 is the reference genome to use for this dataset. Image was adapted from cBioPortal and finalized in Adobe Illustrator 2021, coloring to just show missense vs.&nbsp;other mutations.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb52"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb52-1"><a href="#cb52-1" aria-hidden="true" tabindex="-1"></a>cbio <span class="ot">&lt;-</span> variants_simple[, <span class="fu">c</span>(<span class="st">"SAMPLE"</span>, <span class="st">"Chromosome"</span>, </span>
<span id="cb52-2"><a href="#cb52-2" aria-hidden="true" tabindex="-1"></a>                            <span class="st">"Start_Position"</span>, <span class="st">"End_Position"</span>, <span class="st">"Hugo_Symbol"</span>,</span>
<span id="cb52-3"><a href="#cb52-3" aria-hidden="true" tabindex="-1"></a>                            <span class="st">"Mutation_Type"</span>, <span class="st">"Reference_Allele"</span>, <span class="st">"variant_allele"</span>,</span>
<span id="cb52-4"><a href="#cb52-4" aria-hidden="true" tabindex="-1"></a>                            <span class="st">"Cancer_Type"</span>)]</span>
<span id="cb52-5"><a href="#cb52-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-6"><a href="#cb52-6" aria-hidden="true" tabindex="-1"></a>cbio_germline <span class="ot">&lt;-</span> variants_germline[, <span class="fu">c</span>(<span class="st">"bcr_patient_barcode"</span>, <span class="st">"Chromosome"</span>,</span>
<span id="cb52-7"><a href="#cb52-7" aria-hidden="true" tabindex="-1"></a>                                       <span class="st">"Start"</span>, <span class="st">"Stop"</span>, <span class="st">"HUGO_Symbol"</span>,</span>
<span id="cb52-8"><a href="#cb52-8" aria-hidden="true" tabindex="-1"></a>                                       <span class="st">"Variant_Classification"</span>,</span>
<span id="cb52-9"><a href="#cb52-9" aria-hidden="true" tabindex="-1"></a>                                       <span class="st">"Reference"</span>,<span class="st">"Alternate"</span>, <span class="st">"cancer"</span>)]</span>
<span id="cb52-10"><a href="#cb52-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-11"><a href="#cb52-11" aria-hidden="true" tabindex="-1"></a>cbio_germline<span class="sc">$</span>Chromosome <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">"chr"</span>, cbio_germline<span class="sc">$</span>Chromosome)</span>
<span id="cb52-12"><a href="#cb52-12" aria-hidden="true" tabindex="-1"></a>cbio_germline<span class="sc">$</span>Variant_Classification <span class="ot">&lt;-</span> <span class="st">"Other_Mutation"</span></span>
<span id="cb52-13"><a href="#cb52-13" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(cbio_germline) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"SAMPLE"</span>, <span class="st">"Chromosome"</span>, <span class="st">"Start_Position"</span>,</span>
<span id="cb52-14"><a href="#cb52-14" aria-hidden="true" tabindex="-1"></a>                             <span class="st">"End_Position"</span>,<span class="st">"Hugo_Symbol"</span>, <span class="st">"Mutation_Type"</span>,</span>
<span id="cb52-15"><a href="#cb52-15" aria-hidden="true" tabindex="-1"></a>                             <span class="st">"Reference_Allele"</span>,<span class="st">"variant_allele"</span>, <span class="st">"Cancer_Type"</span>)</span>
<span id="cb52-16"><a href="#cb52-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-17"><a href="#cb52-17" aria-hidden="true" tabindex="-1"></a>cbio <span class="ot">&lt;-</span> <span class="fu">rbind</span>(cbio, cbio_germline)</span>
<span id="cb52-18"><a href="#cb52-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-19"><a href="#cb52-19" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(cbio) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Sample_ID"</span>, <span class="st">"Chromosome"</span>, <span class="st">"Start_Position"</span>,</span>
<span id="cb52-20"><a href="#cb52-20" aria-hidden="true" tabindex="-1"></a>                    <span class="st">"End_Position"</span>, <span class="st">"Hugo_Symbol"</span>, <span class="st">"Mutation_Type"</span>,</span>
<span id="cb52-21"><a href="#cb52-21" aria-hidden="true" tabindex="-1"></a>                    <span class="st">"Reference_Allele"</span>, <span class="st">"Variant_Allele"</span>, <span class="st">"Cancer_Type"</span>)</span>
<span id="cb52-22"><a href="#cb52-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-23"><a href="#cb52-23" aria-hidden="true" tabindex="-1"></a><span class="co"># write.table(cbio, file = "cbio_import.txt", sep = "\t", row.names = FALSE, quote = FALSE)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="figure-2---pan-cancer-bap1-copy-number-loss-and-alterations" class="level2">
<h2 class="anchored" data-anchor-id="figure-2---pan-cancer-bap1-copy-number-loss-and-alterations">Figure 2 - Pan-cancer <em>BAP1</em> copy number loss and alterations</h2>
<p>Here we turn to looking at the contribution of gene-level BAP1 copy number loss across TCGA.</p>
<blockquote class="blockquote">
<p>BAP1 CN loss was observed in 1547 samples and estimated at single copy loss in 99% of all samples with CN loss (Table S3A-B)</p>
</blockquote>
<div class="cell">
<div class="sourceCode cell-code" id="cb53"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb53-1"><a href="#cb53-1" aria-hidden="true" tabindex="-1"></a>num_copynumber <span class="ot">&lt;-</span> <span class="fu">sum</span>(<span class="fu">grepl</span>(<span class="st">"copynumber"</span>, sample_anno<span class="sc">$</span>alteration_type))</span>
<span id="cb53-2"><a href="#cb53-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb53-3"><a href="#cb53-3" aria-hidden="true" tabindex="-1"></a>percent_heterozygous_loss <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="fu">round</span>(<span class="fu">sum</span>(sample_anno<span class="sc">$</span>copynumber_absolute_bap1 <span class="sc">==</span> <span class="dv">1</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>) <span class="sc">/</span> num_copynumber <span class="sc">*</span> <span class="dv">100</span>, <span class="dv">1</span>), <span class="st">"%"</span>)</span>
<span id="cb53-4"><a href="#cb53-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb53-5"><a href="#cb53-5" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="fu">sprintf</span>(<span class="st">"# of samples with copy number loss: %s"</span>, num_copynumber), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># of samples with copy number loss: 1547 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb55"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb55-1"><a href="#cb55-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="fu">sprintf</span>(<span class="st">"Percent heterozygous loss of all CN loss: %s"</span>, percent_heterozygous_loss), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Percent heterozygous loss of all CN loss: 99% </code></pre>
</div>
</div>
<blockquote class="blockquote">
<p>Gene-level CN loss occurs frequently (&gt;30% of samples) in KIRC, UVM, CHOL, MESO, LIHC, lung squamous cell carcinoma (LUSC), and head and neck squamous cancers (HNSC) (Figure 2A and Table S3A).</p>
</blockquote>
<div class="cell">
<div class="sourceCode cell-code" id="cb57"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb57-1"><a href="#cb57-1" aria-hidden="true" tabindex="-1"></a>types_30freq <span class="ot">&lt;-</span> cohort_anno<span class="sc">$</span>Cohort[(cohort_anno<span class="sc">$</span>Copynumber.only <span class="sc">+</span> </span>
<span id="cb57-2"><a href="#cb57-2" aria-hidden="true" tabindex="-1"></a>   cohort_anno<span class="sc">$</span>Mutation.Copynumber) <span class="sc">/</span> cohort_anno<span class="sc">$</span>Cohort.size <span class="sc">&gt;</span> <span class="fl">0.3</span>]</span>
<span id="cb57-3"><a href="#cb57-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb57-4"><a href="#cb57-4" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="fu">sprintf</span>(<span class="st">"Cancer types with &gt;30 percent gene-level CN loss: %s"</span>,</span>
<span id="cb57-5"><a href="#cb57-5" aria-hidden="true" tabindex="-1"></a>            <span class="fu">paste</span>(types_30freq, <span class="at">collapse =</span> <span class="st">", "</span>)), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Cancer types with &gt;30 percent gene-level CN loss: KIRC, CHOL, MESO, UVM, LUSC, HNSC </code></pre>
</div>
</div>
<section id="figure-2a-right" class="level3">
<h3 class="anchored" data-anchor-id="figure-2a-right">Figure 2A, right</h3>
<p><code>Plotgardener</code> is a highly customizable R package for generating figures, especially those featuring genomic data. The goal for this panel was to show average segment widths associated with BAP1 copy number loss in order to understand spatially how much of the chromosome 3p arm was lost in cases of loss. We also annotate other important tumor suppressors co-located on chromosome 3p. This may not display properly in the Quarto markdown preview. Image was finalized in Adobe Illustrator 2021.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb59"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb59-1"><a href="#cb59-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(plotgardener)</span>
<span id="cb59-2"><a href="#cb59-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(org.Hs.eg.db)</span>
<span id="cb59-3"><a href="#cb59-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(TxDb.Hsapiens.UCSC.hg38.knownGene)</span>
<span id="cb59-4"><a href="#cb59-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(AnnotationHub)</span>
<span id="cb59-5"><a href="#cb59-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(annotables)</span>
<span id="cb59-6"><a href="#cb59-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-7"><a href="#cb59-7" aria-hidden="true" tabindex="-1"></a>chr3_length <span class="ot">&lt;-</span> <span class="dv">198295559</span></span>
<span id="cb59-8"><a href="#cb59-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-9"><a href="#cb59-9" aria-hidden="true" tabindex="-1"></a>bap1_start <span class="ot">&lt;-</span> grch38<span class="sc">$</span>start[grch38<span class="sc">$</span>symbol <span class="sc">==</span> <span class="st">"BAP1"</span>]</span>
<span id="cb59-10"><a href="#cb59-10" aria-hidden="true" tabindex="-1"></a>bap1_end <span class="ot">&lt;-</span> grch38<span class="sc">$</span>end[grch38<span class="sc">$</span>symbol <span class="sc">==</span> <span class="st">"BAP1"</span>]</span>
<span id="cb59-11"><a href="#cb59-11" aria-hidden="true" tabindex="-1"></a>bap1_length <span class="ot">&lt;-</span> bap1_end <span class="sc">-</span> bap1_start <span class="sc">+</span> <span class="dv">1</span></span>
<span id="cb59-12"><a href="#cb59-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-13"><a href="#cb59-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-14"><a href="#cb59-14" aria-hidden="true" tabindex="-1"></a>pbrm1_start <span class="ot">&lt;-</span> grch38<span class="sc">$</span>start[grch38<span class="sc">$</span>symbol <span class="sc">==</span> <span class="st">"PBRM1"</span>]</span>
<span id="cb59-15"><a href="#cb59-15" aria-hidden="true" tabindex="-1"></a>pbrm1_end <span class="ot">&lt;-</span> grch38<span class="sc">$</span>end[grch38<span class="sc">$</span>symbol <span class="sc">==</span> <span class="st">"PBRM1"</span>]</span>
<span id="cb59-16"><a href="#cb59-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-17"><a href="#cb59-17" aria-hidden="true" tabindex="-1"></a>setd2_start <span class="ot">&lt;-</span> grch38<span class="sc">$</span>start[grch38<span class="sc">$</span>symbol <span class="sc">==</span> <span class="st">"SETD2"</span>]</span>
<span id="cb59-18"><a href="#cb59-18" aria-hidden="true" tabindex="-1"></a>setd2_end <span class="ot">&lt;-</span> grch38<span class="sc">$</span>end[grch38<span class="sc">$</span>symbol <span class="sc">==</span> <span class="st">"SETD2"</span>]</span>
<span id="cb59-19"><a href="#cb59-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-20"><a href="#cb59-20" aria-hidden="true" tabindex="-1"></a>vhl_start <span class="ot">&lt;-</span> grch38<span class="sc">$</span>start[grch38<span class="sc">$</span>symbol <span class="sc">==</span> <span class="st">"VHL"</span>]</span>
<span id="cb59-21"><a href="#cb59-21" aria-hidden="true" tabindex="-1"></a>vhl_end <span class="ot">&lt;-</span> grch38<span class="sc">$</span>end[grch38<span class="sc">$</span>symbol <span class="sc">==</span> <span class="st">"VHL"</span>]</span>
<span id="cb59-22"><a href="#cb59-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-23"><a href="#cb59-23" aria-hidden="true" tabindex="-1"></a><span class="fu">pageCreate</span>(<span class="at">width =</span> <span class="dv">6</span>, <span class="at">height =</span> <span class="dv">8</span>, <span class="at">default.units =</span> <span class="st">"inches"</span>)</span>
<span id="cb59-24"><a href="#cb59-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-25"><a href="#cb59-25" aria-hidden="true" tabindex="-1"></a><span class="fu">plotIdeogram</span>(<span class="at">chrom =</span> <span class="st">"chr3"</span>, <span class="at">assembly =</span> <span class="st">"hg38"</span>,</span>
<span id="cb59-26"><a href="#cb59-26" aria-hidden="true" tabindex="-1"></a>             <span class="at">x =</span> <span class="dv">1</span>, <span class="at">y =</span> <span class="fl">0.5</span>, <span class="at">width =</span> <span class="fl">3.25</span>, <span class="at">height =</span> <span class="fl">0.15</span>, </span>
<span id="cb59-27"><a href="#cb59-27" aria-hidden="true" tabindex="-1"></a>             <span class="at">just =</span> <span class="fu">c</span>(<span class="st">"left"</span>, <span class="st">"top"</span>), <span class="at">default.units =</span> <span class="st">"inches"</span>)</span>
<span id="cb59-28"><a href="#cb59-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-29"><a href="#cb59-29" aria-hidden="true" tabindex="-1"></a><span class="fu">plotText</span>(<span class="at">label =</span> <span class="st">"Chromosome 3"</span>,</span>
<span id="cb59-30"><a href="#cb59-30" aria-hidden="true" tabindex="-1"></a>         <span class="at">fontsize =</span> <span class="dv">8</span>, <span class="at">fontcolor =</span> <span class="st">"black"</span>, <span class="at">x =</span> <span class="fl">4.25</span>, <span class="at">y =</span> <span class="fl">0.4</span>, </span>
<span id="cb59-31"><a href="#cb59-31" aria-hidden="true" tabindex="-1"></a>         <span class="at">just =</span> <span class="st">"right"</span>, <span class="at">default.units =</span> <span class="st">"inches"</span>)</span>
<span id="cb59-32"><a href="#cb59-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-33"><a href="#cb59-33" aria-hidden="true" tabindex="-1"></a><span class="fu">plotText</span>(<span class="at">label =</span> <span class="st">"BAP1 locus"</span>,</span>
<span id="cb59-34"><a href="#cb59-34" aria-hidden="true" tabindex="-1"></a>         <span class="at">fontsize =</span> <span class="dv">8</span>, <span class="at">fontcolor =</span> <span class="st">"black"</span>, <span class="at">x =</span> <span class="dv">1</span> <span class="sc">+</span> (bap1_start<span class="sc">/</span>chr3_length)<span class="sc">*</span><span class="fl">3.25</span>, <span class="at">y =</span> <span class="fl">0.1</span>, </span>
<span id="cb59-35"><a href="#cb59-35" aria-hidden="true" tabindex="-1"></a>         <span class="at">just =</span> <span class="st">"center"</span>, <span class="at">default.units =</span> <span class="st">"inches"</span>)</span>
<span id="cb59-36"><a href="#cb59-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-37"><a href="#cb59-37" aria-hidden="true" tabindex="-1"></a><span class="fu">plotSegments</span>(<span class="at">x0 =</span> <span class="dv">1</span> <span class="sc">+</span> (bap1_start<span class="sc">/</span>chr3_length)<span class="sc">*</span><span class="fl">3.25</span>, <span class="at">x1 =</span> <span class="dv">1</span> <span class="sc">+</span> (bap1_end<span class="sc">/</span>chr3_length)<span class="sc">*</span><span class="fl">3.25</span>, </span>
<span id="cb59-38"><a href="#cb59-38" aria-hidden="true" tabindex="-1"></a>             <span class="at">y0 =</span> <span class="fl">0.25</span>, <span class="at">y1 =</span> <span class="fl">0.7</span>, <span class="at">lwd =</span> <span class="dv">1</span>, <span class="at">linecolor =</span> <span class="st">"red"</span>)</span>
<span id="cb59-39"><a href="#cb59-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-40"><a href="#cb59-40" aria-hidden="true" tabindex="-1"></a><span class="fu">plotText</span>(<span class="at">label =</span> <span class="st">"PBRM1 locus"</span>,</span>
<span id="cb59-41"><a href="#cb59-41" aria-hidden="true" tabindex="-1"></a>         <span class="at">fontsize =</span> <span class="dv">8</span>, <span class="at">fontcolor =</span> <span class="st">"black"</span>, <span class="at">x =</span> <span class="dv">1</span> <span class="sc">+</span> (pbrm1_start<span class="sc">/</span>chr3_length)<span class="sc">*</span><span class="fl">3.25</span>, <span class="at">y =</span> <span class="fl">0.1</span>, </span>
<span id="cb59-42"><a href="#cb59-42" aria-hidden="true" tabindex="-1"></a>         <span class="at">just =</span> <span class="st">"center"</span>, <span class="at">default.units =</span> <span class="st">"inches"</span>)</span>
<span id="cb59-43"><a href="#cb59-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-44"><a href="#cb59-44" aria-hidden="true" tabindex="-1"></a><span class="fu">plotSegments</span>(<span class="at">x0 =</span> <span class="dv">1</span> <span class="sc">+</span> (pbrm1_start<span class="sc">/</span>chr3_length)<span class="sc">*</span><span class="fl">3.25</span>, <span class="at">x1 =</span> <span class="dv">1</span> <span class="sc">+</span> (pbrm1_end<span class="sc">/</span>chr3_length)<span class="sc">*</span><span class="fl">3.25</span>, </span>
<span id="cb59-45"><a href="#cb59-45" aria-hidden="true" tabindex="-1"></a>             <span class="at">y0 =</span> <span class="fl">0.25</span>, <span class="at">y1 =</span> <span class="fl">0.7</span>, <span class="at">lwd =</span> <span class="dv">1</span>, <span class="at">linecolor =</span> <span class="st">"grey"</span>)</span>
<span id="cb59-46"><a href="#cb59-46" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-47"><a href="#cb59-47" aria-hidden="true" tabindex="-1"></a><span class="fu">plotText</span>(<span class="at">label =</span> <span class="st">"SETD2 locus"</span>,</span>
<span id="cb59-48"><a href="#cb59-48" aria-hidden="true" tabindex="-1"></a>         <span class="at">fontsize =</span> <span class="dv">8</span>, <span class="at">fontcolor =</span> <span class="st">"black"</span>, <span class="at">x =</span> <span class="dv">1</span> <span class="sc">+</span> (setd2_start<span class="sc">/</span>chr3_length)<span class="sc">*</span><span class="fl">3.25</span>, <span class="at">y =</span> <span class="fl">0.1</span>, </span>
<span id="cb59-49"><a href="#cb59-49" aria-hidden="true" tabindex="-1"></a>         <span class="at">just =</span> <span class="st">"center"</span>, <span class="at">default.units =</span> <span class="st">"inches"</span>)</span>
<span id="cb59-50"><a href="#cb59-50" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-51"><a href="#cb59-51" aria-hidden="true" tabindex="-1"></a><span class="fu">plotSegments</span>(<span class="at">x0 =</span> <span class="dv">1</span> <span class="sc">+</span> (setd2_start<span class="sc">/</span>chr3_length)<span class="sc">*</span><span class="fl">3.25</span>, <span class="at">x1 =</span> <span class="dv">1</span> <span class="sc">+</span> (setd2_end<span class="sc">/</span>chr3_length)<span class="sc">*</span><span class="fl">3.25</span>, </span>
<span id="cb59-52"><a href="#cb59-52" aria-hidden="true" tabindex="-1"></a>             <span class="at">y0 =</span> <span class="fl">0.25</span>, <span class="at">y1 =</span> <span class="fl">0.7</span>, <span class="at">lwd =</span> <span class="dv">1</span>, <span class="at">linecolor =</span> <span class="st">"grey"</span>)</span>
<span id="cb59-53"><a href="#cb59-53" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-54"><a href="#cb59-54" aria-hidden="true" tabindex="-1"></a><span class="fu">plotText</span>(<span class="at">label =</span> <span class="st">"VHL locus"</span>,</span>
<span id="cb59-55"><a href="#cb59-55" aria-hidden="true" tabindex="-1"></a>         <span class="at">fontsize =</span> <span class="dv">8</span>, <span class="at">fontcolor =</span> <span class="st">"black"</span>, <span class="at">x =</span> <span class="dv">1</span> <span class="sc">+</span> (vhl_start<span class="sc">/</span>chr3_length)<span class="sc">*</span><span class="fl">3.25</span>, <span class="at">y =</span> <span class="fl">0.1</span>, </span>
<span id="cb59-56"><a href="#cb59-56" aria-hidden="true" tabindex="-1"></a>         <span class="at">just =</span> <span class="st">"center"</span>, <span class="at">default.units =</span> <span class="st">"inches"</span>)</span>
<span id="cb59-57"><a href="#cb59-57" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-58"><a href="#cb59-58" aria-hidden="true" tabindex="-1"></a><span class="fu">plotSegments</span>(<span class="at">x0 =</span> <span class="dv">1</span> <span class="sc">+</span> (vhl_start<span class="sc">/</span>chr3_length)<span class="sc">*</span><span class="fl">3.25</span>, <span class="at">x1 =</span> <span class="dv">1</span> <span class="sc">+</span> (vhl_end<span class="sc">/</span>chr3_length)<span class="sc">*</span><span class="fl">3.25</span>, </span>
<span id="cb59-59"><a href="#cb59-59" aria-hidden="true" tabindex="-1"></a>             <span class="at">y0 =</span> <span class="fl">0.25</span>, <span class="at">y1 =</span> <span class="fl">0.7</span>, <span class="at">lwd =</span> <span class="dv">1</span>, <span class="at">linecolor =</span> <span class="st">"grey"</span>)</span>
<span id="cb59-60"><a href="#cb59-60" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-61"><a href="#cb59-61" aria-hidden="true" tabindex="-1"></a>j <span class="ot">&lt;-</span> <span class="fl">0.75</span> <span class="co"># this is an iterator variable for positioning plot elements</span></span>
<span id="cb59-62"><a href="#cb59-62" aria-hidden="true" tabindex="-1"></a>mean_seg_start_list <span class="ot">&lt;-</span> <span class="fu">c</span>()</span>
<span id="cb59-63"><a href="#cb59-63" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-64"><a href="#cb59-64" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> tcga_projects[<span class="sc">!</span>tcga_projects <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"TCGA-THCA"</span>, <span class="st">"TCGA-TGCT"</span>)]) {</span>
<span id="cb59-65"><a href="#cb59-65" aria-hidden="true" tabindex="-1"></a>  loc <span class="ot">&lt;-</span> <span class="fu">as.character</span>(<span class="fu">na.omit</span>(sample_anno<span class="sc">$</span>bap1_segment_location[sample_anno<span class="sc">$</span>cohort <span class="sc">==</span> i <span class="sc">&amp;</span></span>
<span id="cb59-66"><a href="#cb59-66" aria-hidden="true" tabindex="-1"></a>                                                            sample_anno<span class="sc">$</span>copynumber_absolute_bap1 <span class="sc">&lt;</span> <span class="dv">2</span>]))</span>
<span id="cb59-67"><a href="#cb59-67" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">length</span>(loc) <span class="sc">&gt;</span> <span class="dv">0</span>) {</span>
<span id="cb59-68"><a href="#cb59-68" aria-hidden="true" tabindex="-1"></a>    mean_seg_start <span class="ot">&lt;-</span> <span class="fu">c</span>()</span>
<span id="cb59-69"><a href="#cb59-69" aria-hidden="true" tabindex="-1"></a>    mean_seg_end <span class="ot">&lt;-</span> <span class="fu">c</span>()</span>
<span id="cb59-70"><a href="#cb59-70" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> (k <span class="cf">in</span> loc) {</span>
<span id="cb59-71"><a href="#cb59-71" aria-hidden="true" tabindex="-1"></a>      mean_seg_start <span class="ot">&lt;-</span> <span class="fu">c</span>(mean_seg_start, </span>
<span id="cb59-72"><a href="#cb59-72" aria-hidden="true" tabindex="-1"></a>                          <span class="fu">as.numeric</span>(<span class="fu">strsplit</span>(<span class="fu">strsplit</span>(k, <span class="st">":"</span>)[[<span class="dv">1</span>]][<span class="dv">2</span>], <span class="st">"-"</span>)[[<span class="dv">1</span>]][<span class="dv">1</span>]))</span>
<span id="cb59-73"><a href="#cb59-73" aria-hidden="true" tabindex="-1"></a>      mean_seg_end <span class="ot">&lt;-</span> <span class="fu">c</span>(mean_seg_end, </span>
<span id="cb59-74"><a href="#cb59-74" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">as.numeric</span>(<span class="fu">strsplit</span>(<span class="fu">strsplit</span>(k, <span class="st">":"</span>)[[<span class="dv">1</span>]][<span class="dv">2</span>], <span class="st">"-"</span>)[[<span class="dv">1</span>]][<span class="dv">2</span>]))</span>
<span id="cb59-75"><a href="#cb59-75" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb59-76"><a href="#cb59-76" aria-hidden="true" tabindex="-1"></a>    min_seg <span class="ot">&lt;-</span> <span class="fu">quantile</span>(mean_seg_start, <span class="fl">0.25</span>)</span>
<span id="cb59-77"><a href="#cb59-77" aria-hidden="true" tabindex="-1"></a>    max_seg <span class="ot">&lt;-</span> <span class="fu">quantile</span>(mean_seg_end, <span class="fl">0.75</span>)</span>
<span id="cb59-78"><a href="#cb59-78" aria-hidden="true" tabindex="-1"></a>    mean_seg_start_keep <span class="ot">&lt;-</span> mean_seg_start</span>
<span id="cb59-79"><a href="#cb59-79" aria-hidden="true" tabindex="-1"></a>    mean_seg_end_keep <span class="ot">&lt;-</span> mean_seg_end</span>
<span id="cb59-80"><a href="#cb59-80" aria-hidden="true" tabindex="-1"></a>    mean_seg_start <span class="ot">&lt;-</span> <span class="fu">mean</span>(mean_seg_start)</span>
<span id="cb59-81"><a href="#cb59-81" aria-hidden="true" tabindex="-1"></a>    mean_seg_start_list <span class="ot">&lt;-</span> <span class="fu">c</span>(mean_seg_start_list, mean_seg_start)</span>
<span id="cb59-82"><a href="#cb59-82" aria-hidden="true" tabindex="-1"></a>    mean_seg_end <span class="ot">&lt;-</span> <span class="fu">mean</span>(mean_seg_end)</span>
<span id="cb59-83"><a href="#cb59-83" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb59-84"><a href="#cb59-84" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb59-85"><a href="#cb59-85" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-86"><a href="#cb59-86" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> tcga_projects[<span class="sc">!</span>tcga_projects <span class="sc">%in%</span> </span>
<span id="cb59-87"><a href="#cb59-87" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">c</span>(<span class="st">"TCGA-THCA"</span>, <span class="st">"TCGA-TGCT"</span>)][<span class="fu">order</span>(mean_seg_start_list)]) {</span>
<span id="cb59-88"><a href="#cb59-88" aria-hidden="true" tabindex="-1"></a>  cohort_short <span class="ot">&lt;-</span> <span class="fu">strsplit</span>(i, <span class="st">"-"</span>)[[<span class="dv">1</span>]][<span class="dv">2</span>]</span>
<span id="cb59-89"><a href="#cb59-89" aria-hidden="true" tabindex="-1"></a>  plot_name <span class="ot">&lt;-</span> <span class="fu">strsplit</span>(i, <span class="st">"-"</span>)[[<span class="dv">1</span>]][<span class="dv">2</span>]</span>
<span id="cb59-90"><a href="#cb59-90" aria-hidden="true" tabindex="-1"></a>  loc <span class="ot">&lt;-</span> <span class="fu">as.character</span>(<span class="fu">na.omit</span>(sample_anno<span class="sc">$</span>bap1_segment_location[sample_anno<span class="sc">$</span>cohort <span class="sc">==</span> i <span class="sc">&amp;</span> sample_anno<span class="sc">$</span>copynumber_absolute_bap1 <span class="sc">&lt;</span> <span class="dv">2</span>]))</span>
<span id="cb59-91"><a href="#cb59-91" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">length</span>(loc) <span class="sc">&gt;</span> <span class="dv">0</span>) {</span>
<span id="cb59-92"><a href="#cb59-92" aria-hidden="true" tabindex="-1"></a>    mean_seg_start <span class="ot">&lt;-</span> <span class="fu">c</span>()</span>
<span id="cb59-93"><a href="#cb59-93" aria-hidden="true" tabindex="-1"></a>    mean_seg_end <span class="ot">&lt;-</span> <span class="fu">c</span>()</span>
<span id="cb59-94"><a href="#cb59-94" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> (k <span class="cf">in</span> loc) {</span>
<span id="cb59-95"><a href="#cb59-95" aria-hidden="true" tabindex="-1"></a>      mean_seg_start <span class="ot">&lt;-</span> <span class="fu">c</span>(mean_seg_start, </span>
<span id="cb59-96"><a href="#cb59-96" aria-hidden="true" tabindex="-1"></a>                          <span class="fu">as.numeric</span>(<span class="fu">strsplit</span>(<span class="fu">strsplit</span>(k, <span class="st">":"</span>)[[<span class="dv">1</span>]][<span class="dv">2</span>], <span class="st">"-"</span>)[[<span class="dv">1</span>]][<span class="dv">1</span>]))</span>
<span id="cb59-97"><a href="#cb59-97" aria-hidden="true" tabindex="-1"></a>      mean_seg_end <span class="ot">&lt;-</span> <span class="fu">c</span>(mean_seg_end, </span>
<span id="cb59-98"><a href="#cb59-98" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">as.numeric</span>(<span class="fu">strsplit</span>(<span class="fu">strsplit</span>(k, <span class="st">":"</span>)[[<span class="dv">1</span>]][<span class="dv">2</span>], <span class="st">"-"</span>)[[<span class="dv">1</span>]][<span class="dv">2</span>]))</span>
<span id="cb59-99"><a href="#cb59-99" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb59-100"><a href="#cb59-100" aria-hidden="true" tabindex="-1"></a>    min_seg <span class="ot">&lt;-</span> <span class="fu">quantile</span>(mean_seg_start, <span class="fl">0.25</span>)</span>
<span id="cb59-101"><a href="#cb59-101" aria-hidden="true" tabindex="-1"></a>    max_seg <span class="ot">&lt;-</span> <span class="fu">quantile</span>(mean_seg_end, <span class="fl">0.75</span>)</span>
<span id="cb59-102"><a href="#cb59-102" aria-hidden="true" tabindex="-1"></a>    mean_seg_start_keep <span class="ot">&lt;-</span> mean_seg_start</span>
<span id="cb59-103"><a href="#cb59-103" aria-hidden="true" tabindex="-1"></a>    mean_seg_end_keep <span class="ot">&lt;-</span> mean_seg_end</span>
<span id="cb59-104"><a href="#cb59-104" aria-hidden="true" tabindex="-1"></a>    mean_seg_start <span class="ot">&lt;-</span> <span class="fu">mean</span>(mean_seg_start)</span>
<span id="cb59-105"><a href="#cb59-105" aria-hidden="true" tabindex="-1"></a>    mean_seg_end <span class="ot">&lt;-</span> <span class="fu">mean</span>(mean_seg_end)</span>
<span id="cb59-106"><a href="#cb59-106" aria-hidden="true" tabindex="-1"></a>    <span class="fu">plotRect</span>(<span class="at">x =</span> <span class="dv">1</span> <span class="sc">+</span> (mean_seg_start<span class="sc">/</span>chr3_length)<span class="sc">*</span><span class="fl">3.25</span>,</span>
<span id="cb59-107"><a href="#cb59-107" aria-hidden="true" tabindex="-1"></a>             <span class="at">y =</span> j,</span>
<span id="cb59-108"><a href="#cb59-108" aria-hidden="true" tabindex="-1"></a>             <span class="at">width =</span> ((mean_seg_end <span class="sc">-</span> mean_seg_start)<span class="sc">/</span>chr3_length)<span class="sc">*</span><span class="fl">3.25</span>,</span>
<span id="cb59-109"><a href="#cb59-109" aria-hidden="true" tabindex="-1"></a>             <span class="at">height =</span> <span class="fl">0.15</span>,</span>
<span id="cb59-110"><a href="#cb59-110" aria-hidden="true" tabindex="-1"></a>             <span class="at">fill =</span> tcga_col[cohort_short],</span>
<span id="cb59-111"><a href="#cb59-111" aria-hidden="true" tabindex="-1"></a>             <span class="at">just =</span> <span class="st">"left"</span>)</span>
<span id="cb59-112"><a href="#cb59-112" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb59-113"><a href="#cb59-113" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plotText</span>(<span class="at">label =</span> cohort_short,</span>
<span id="cb59-114"><a href="#cb59-114" aria-hidden="true" tabindex="-1"></a>           <span class="at">fontsize =</span> <span class="dv">7</span>, <span class="at">fontcolor =</span> <span class="st">"black"</span>, <span class="at">x =</span> <span class="dv">5</span>, <span class="at">y =</span> j, <span class="at">just =</span> <span class="st">"left"</span>, <span class="at">default.units =</span> <span class="st">"inches"</span>)</span>
<span id="cb59-115"><a href="#cb59-115" aria-hidden="true" tabindex="-1"></a>  j <span class="ot">&lt;-</span> j <span class="sc">+</span> <span class="fl">0.2</span></span>
<span id="cb59-116"><a href="#cb59-116" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="figure_code_files/figure-html/unnamed-chunk-19-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb60"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb60-1"><a href="#cb60-1" aria-hidden="true" tabindex="-1"></a><span class="fu">pageGuideHide</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="figure_code_files/figure-html/unnamed-chunk-19-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="figure-2a-left" class="level3">
<h3 class="anchored" data-anchor-id="figure-2a-left">Figure 2A, left</h3>
<p>This is a companion barplot of frequency and number of gene-level BAP1 copy number loss. Image was finalized in Adobe Illustrator 2021.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb61"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb61-1"><a href="#cb61-1" aria-hidden="true" tabindex="-1"></a>percent_cnv <span class="ot">&lt;-</span> <span class="fu">c</span>()</span>
<span id="cb61-2"><a href="#cb61-2" aria-hidden="true" tabindex="-1"></a>number_cnv <span class="ot">&lt;-</span> <span class="fu">c</span>()</span>
<span id="cb61-3"><a href="#cb61-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb61-4"><a href="#cb61-4" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> tcga_projects[<span class="sc">!</span>tcga_projects <span class="sc">%in%</span> </span>
<span id="cb61-5"><a href="#cb61-5" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">c</span>(<span class="st">"TCGA-THCA"</span>, <span class="st">"TCGA-TGCT"</span>)]) {</span>
<span id="cb61-6"><a href="#cb61-6" aria-hidden="true" tabindex="-1"></a>  num_cnv <span class="ot">&lt;-</span> <span class="fu">sum</span>(<span class="fu">grepl</span>(<span class="st">"copynumber"</span>, sample_anno<span class="sc">$</span>alteration_type[sample_anno<span class="sc">$</span>cohort <span class="sc">==</span> i]),</span>
<span id="cb61-7"><a href="#cb61-7" aria-hidden="true" tabindex="-1"></a>                 <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb61-8"><a href="#cb61-8" aria-hidden="true" tabindex="-1"></a>  num_cohort <span class="ot">&lt;-</span> <span class="fu">sum</span>(sample_anno<span class="sc">$</span>cohort <span class="sc">==</span> i)</span>
<span id="cb61-9"><a href="#cb61-9" aria-hidden="true" tabindex="-1"></a>  number_cnv <span class="ot">&lt;-</span> <span class="fu">c</span>(number_cnv, num_cnv)</span>
<span id="cb61-10"><a href="#cb61-10" aria-hidden="true" tabindex="-1"></a>  percent_cnv <span class="ot">&lt;-</span> <span class="fu">c</span>(percent_cnv, (num_cnv <span class="sc">/</span> num_cohort) <span class="sc">*</span> <span class="dv">100</span>)</span>
<span id="cb61-11"><a href="#cb61-11" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb61-12"><a href="#cb61-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb61-13"><a href="#cb61-13" aria-hidden="true" tabindex="-1"></a>cnv_names <span class="ot">&lt;-</span> <span class="fu">c</span>()</span>
<span id="cb61-14"><a href="#cb61-14" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> tcga_projects[<span class="sc">!</span>tcga_projects <span class="sc">%in%</span> </span>
<span id="cb61-15"><a href="#cb61-15" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">c</span>(<span class="st">"TCGA-THCA"</span>, <span class="st">"TCGA-TGCT"</span>)]) {</span>
<span id="cb61-16"><a href="#cb61-16" aria-hidden="true" tabindex="-1"></a>  cnv_names <span class="ot">&lt;-</span> <span class="fu">c</span>(cnv_names, <span class="fu">strsplit</span>(i, <span class="st">"-"</span>)[[<span class="dv">1</span>]][<span class="dv">2</span>])</span>
<span id="cb61-17"><a href="#cb61-17" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb61-18"><a href="#cb61-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb61-19"><a href="#cb61-19" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(percent_cnv) <span class="ot">&lt;-</span> cnv_names</span>
<span id="cb61-20"><a href="#cb61-20" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(number_cnv) <span class="ot">&lt;-</span> cnv_names</span>
<span id="cb61-21"><a href="#cb61-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb61-22"><a href="#cb61-22" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot order to correspond with figure 2A, right</span></span>
<span id="cb61-23"><a href="#cb61-23" aria-hidden="true" tabindex="-1"></a>plot_order <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"KIRC"</span>, <span class="st">"CHOL"</span>, <span class="st">"PAAD"</span>, <span class="st">"LUAD"</span>, <span class="st">"HNSC"</span>, <span class="st">"GBM"</span>,</span>
<span id="cb61-24"><a href="#cb61-24" aria-hidden="true" tabindex="-1"></a>                <span class="st">"LUSC"</span>, <span class="st">"KIRP"</span>, <span class="st">"ACC"</span>, <span class="st">"UVM"</span>, <span class="st">"CESC"</span>, <span class="st">"SKCM"</span>,</span>
<span id="cb61-25"><a href="#cb61-25" aria-hidden="true" tabindex="-1"></a>                <span class="st">"ESCA"</span>, <span class="st">"LAML"</span>, <span class="st">"KICH"</span>, <span class="st">"LGG"</span>, <span class="st">"STAD"</span>, <span class="st">"PCPG"</span>,</span>
<span id="cb61-26"><a href="#cb61-26" aria-hidden="true" tabindex="-1"></a>                <span class="st">"SARC"</span>, <span class="st">"READ"</span>, <span class="st">"BLCA"</span>, <span class="st">"BRCA"</span>, <span class="st">"LIHC"</span>, <span class="st">"COAD"</span>,</span>
<span id="cb61-27"><a href="#cb61-27" aria-hidden="true" tabindex="-1"></a>                <span class="st">"PRAD"</span>, <span class="st">"THYM"</span>, <span class="st">"DLBC"</span>, <span class="st">"MESO"</span>, <span class="st">"OV"</span>, <span class="st">"UCEC"</span>, <span class="st">"UCS"</span>)</span>
<span id="cb61-28"><a href="#cb61-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb61-29"><a href="#cb61-29" aria-hidden="true" tabindex="-1"></a>cnv_barplot <span class="ot">&lt;-</span> <span class="fu">barplot</span>(percent_cnv[plot_order], <span class="at">las =</span> <span class="dv">2</span>, <span class="at">cex.names =</span> <span class="fl">0.7</span>,</span>
<span id="cb61-30"><a href="#cb61-30" aria-hidden="true" tabindex="-1"></a>                       <span class="at">col =</span> tcga_col[plot_order], <span class="at">ylab =</span> <span class="st">"% BAP1 copy number loss"</span>,</span>
<span id="cb61-31"><a href="#cb61-31" aria-hidden="true" tabindex="-1"></a>                       <span class="at">ylim =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="fu">max</span>(percent_cnv)<span class="sc">+</span><span class="dv">40</span>))</span>
<span id="cb61-32"><a href="#cb61-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb61-33"><a href="#cb61-33" aria-hidden="true" tabindex="-1"></a><span class="fu">text</span>(<span class="at">x =</span> cnv_barplot, <span class="at">y =</span> percent_cnv[plot_order], </span>
<span id="cb61-34"><a href="#cb61-34" aria-hidden="true" tabindex="-1"></a>     <span class="at">label =</span> <span class="fu">paste0</span>(<span class="st">"n="</span>,number_cnv[plot_order]), </span>
<span id="cb61-35"><a href="#cb61-35" aria-hidden="true" tabindex="-1"></a>     <span class="at">pos =</span> <span class="dv">3</span>, <span class="at">cex =</span> <span class="dv">1</span>, <span class="at">col =</span> <span class="st">"black"</span>, <span class="at">srt =</span> <span class="dv">90</span>, <span class="at">offset =</span> <span class="fl">1.3</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="figure_code_files/figure-html/unnamed-chunk-20-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>From the text:</p>
<blockquote class="blockquote">
<p>For example, KIRC (median 60.7Mb, interquartile range IQR 33.6-82.3Mb), LUAD (median 63.0Mb, IQR 34.6-87.8Mb), and PCPG (median 40.8Mb, IQR 12.3-89.2Mb) have very large segment widths of loss (Supplemental Table S3B). In contrast, MESO (median 8.3Mb, IQR 0.2-29.2Mb), OV (median 9.5Mb, IQR 4.7-19.4Mb), and UCEC (median 7.2Mb, IQR 4.6-19.4Mb) have relatively more focal <em>BAP1</em> CN […]</p>
</blockquote>
<div class="cell">
<div class="sourceCode cell-code" id="cb62"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb62-1"><a href="#cb62-1" aria-hidden="true" tabindex="-1"></a>median_seg_width <span class="ot">&lt;-</span> <span class="cf">function</span>(cohort) {</span>
<span id="cb62-2"><a href="#cb62-2" aria-hidden="true" tabindex="-1"></a>  cohort_widths <span class="ot">&lt;-</span> sample_anno<span class="sc">$</span>bap1_segment_width[sample_anno<span class="sc">$</span>cohort <span class="sc">==</span> cohort <span class="sc">&amp;</span></span>
<span id="cb62-3"><a href="#cb62-3" aria-hidden="true" tabindex="-1"></a>                   <span class="fu">grepl</span>(<span class="st">"copynumber"</span>,sample_anno<span class="sc">$</span>alteration_type)]</span>
<span id="cb62-4"><a href="#cb62-4" aria-hidden="true" tabindex="-1"></a>  median_width <span class="ot">&lt;-</span> <span class="fu">median</span>(cohort_widths, <span class="at">na.rm =</span> <span class="cn">TRUE</span>) <span class="sc">/</span> <span class="fl">1e6</span></span>
<span id="cb62-5"><a href="#cb62-5" aria-hidden="true" tabindex="-1"></a>  width_25ile <span class="ot">&lt;-</span> <span class="fu">quantile</span>(cohort_widths, <span class="fl">0.25</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>) <span class="sc">/</span> <span class="fl">1e6</span></span>
<span id="cb62-6"><a href="#cb62-6" aria-hidden="true" tabindex="-1"></a>  width_75ile <span class="ot">&lt;-</span> <span class="fu">quantile</span>(cohort_widths, <span class="fl">0.75</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>) <span class="sc">/</span> <span class="fl">1e6</span></span>
<span id="cb62-7"><a href="#cb62-7" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb62-8"><a href="#cb62-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="fu">sprintf</span>(<span class="st">"%s median copy number width: %sMb (%sMb-%sMb)"</span>,</span>
<span id="cb62-9"><a href="#cb62-9" aria-hidden="true" tabindex="-1"></a>              cohort, <span class="fu">round</span>(median_width, <span class="dv">1</span>), <span class="fu">round</span>(width_25ile, <span class="dv">1</span>),</span>
<span id="cb62-10"><a href="#cb62-10" aria-hidden="true" tabindex="-1"></a>              <span class="fu">round</span>(width_75ile, <span class="dv">1</span>)), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span>
<span id="cb62-11"><a href="#cb62-11" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb62-12"><a href="#cb62-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb62-13"><a href="#cb62-13" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="fu">c</span>(<span class="st">"TCGA-KIRC"</span>, <span class="st">"TCGA-LUAD"</span>, <span class="st">"TCGA-PCPG"</span>, <span class="st">"TCGA-MESO"</span>, </span>
<span id="cb62-14"><a href="#cb62-14" aria-hidden="true" tabindex="-1"></a>            <span class="st">"TCGA-OV"</span>, <span class="st">"TCGA-UCEC"</span>)) {</span>
<span id="cb62-15"><a href="#cb62-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">median_seg_width</span>(i)</span>
<span id="cb62-16"><a href="#cb62-16" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>TCGA-KIRC median copy number width: 60.7Mb (33.6Mb-82.3Mb) 
TCGA-LUAD median copy number width: 63Mb (34.6Mb-87.8Mb) 
TCGA-PCPG median copy number width: 40.8Mb (12.3Mb-89.2Mb) 
TCGA-MESO median copy number width: 8.3Mb (0.2Mb-29.2Mb) 
TCGA-OV median copy number width: 9.5Mb (4.7Mb-19.4Mb) 
TCGA-UCEC median copy number width: 7.2Mb (4.6Mb-15.6Mb) </code></pre>
</div>
</div>
<blockquote class="blockquote">
<p>Combining CN and mutation data, 1679 samples from 32 tumor types were considered altered (Figure 2B and Table S3).</p>
</blockquote>
<div class="cell">
<div class="sourceCode cell-code" id="cb64"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb64-1"><a href="#cb64-1" aria-hidden="true" tabindex="-1"></a>num_altered <span class="ot">&lt;-</span> <span class="fu">sum</span>(sample_anno<span class="sc">$</span>altered_status <span class="sc">==</span> <span class="st">"altered"</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb64-2"><a href="#cb64-2" aria-hidden="true" tabindex="-1"></a>num_altered_cohorts <span class="ot">&lt;-</span> <span class="fu">length</span>(<span class="fu">unique</span>(<span class="fu">na.omit</span>(sample_anno<span class="sc">$</span>cohort[sample_anno<span class="sc">$</span>altered_status <span class="sc">==</span> <span class="st">"altered"</span>])))</span>
<span id="cb64-3"><a href="#cb64-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-4"><a href="#cb64-4" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="fu">sprintf</span>(<span class="st">"# pan-cancer BAP1 altered samples: %s"</span>,</span>
<span id="cb64-5"><a href="#cb64-5" aria-hidden="true" tabindex="-1"></a>            num_altered), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># pan-cancer BAP1 altered samples: 1679 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb66"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb66-1"><a href="#cb66-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="fu">sprintf</span>(<span class="st">"# BAP1 altered cancer types: %s"</span>,</span>
<span id="cb66-2"><a href="#cb66-2" aria-hidden="true" tabindex="-1"></a>            num_altered_cohorts), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># BAP1 altered cancer types: 32 </code></pre>
</div>
</div>
<blockquote class="blockquote">
<p>Alterations were predominantly copy number-driven (1547/1679, 92.1%). Mutations were less frequent; they occurred alone in 7.9% (132/1679) of altered samples and co-occurred with estimated single-copy CN loss in 6% (101/1679).</p>
</blockquote>
<div class="cell">
<div class="sourceCode cell-code" id="cb68"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb68-1"><a href="#cb68-1" aria-hidden="true" tabindex="-1"></a>percent_altered_copynumber <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="fu">round</span>(num_copynumber <span class="sc">/</span> num_altered <span class="sc">*</span> <span class="dv">100</span>, <span class="dv">1</span>), <span class="st">"%"</span>)</span>
<span id="cb68-2"><a href="#cb68-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb68-3"><a href="#cb68-3" aria-hidden="true" tabindex="-1"></a>num_mut_cn <span class="ot">&lt;-</span> <span class="fu">sum</span>(sample_anno<span class="sc">$</span>alteration_type <span class="sc">==</span> <span class="st">"mutation+copynumber"</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb68-4"><a href="#cb68-4" aria-hidden="true" tabindex="-1"></a>percent_mut_cn <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="fu">round</span>(num_mut_cn <span class="sc">/</span> num_altered <span class="sc">*</span> <span class="dv">100</span>, <span class="dv">1</span>), <span class="st">"%"</span>)</span>
<span id="cb68-5"><a href="#cb68-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb68-6"><a href="#cb68-6" aria-hidden="true" tabindex="-1"></a>num_mut <span class="ot">&lt;-</span> <span class="fu">sum</span>(sample_anno<span class="sc">$</span>alteration_type <span class="sc">==</span> <span class="st">"mutation"</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb68-7"><a href="#cb68-7" aria-hidden="true" tabindex="-1"></a>percent_mut <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="fu">round</span>(num_mut <span class="sc">/</span> num_altered <span class="sc">*</span> <span class="dv">100</span>, <span class="dv">1</span>), <span class="st">"%"</span>)</span>
<span id="cb68-8"><a href="#cb68-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb68-9"><a href="#cb68-9" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="fu">sprintf</span>(<span class="st">"Copy number-driven alterations: %s/%s (%s)"</span>,</span>
<span id="cb68-10"><a href="#cb68-10" aria-hidden="true" tabindex="-1"></a>            num_copynumber, num_altered, percent_altered_copynumber), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Copy number-driven alterations: 1547/1679 (92.1%) </code></pre>
</div>
<div class="sourceCode cell-code" id="cb70"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb70-1"><a href="#cb70-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="fu">sprintf</span>(<span class="st">"Mutation-driven alterations: %s/%s (%s)"</span>,</span>
<span id="cb70-2"><a href="#cb70-2" aria-hidden="true" tabindex="-1"></a>            num_mut, num_altered, percent_mut), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Mutation-driven alterations: 132/1679 (7.9%) </code></pre>
</div>
<div class="sourceCode cell-code" id="cb72"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb72-1"><a href="#cb72-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="fu">sprintf</span>(<span class="st">"Mutation+copynumber alterations: %s/%s (%s)"</span>,</span>
<span id="cb72-2"><a href="#cb72-2" aria-hidden="true" tabindex="-1"></a>            num_mut_cn, num_altered, percent_mut_cn), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Mutation+copynumber alterations: 101/1679 (6%) </code></pre>
</div>
</div>
<p>Comparisons of VAF across mutation types (mutation only or mutation+copynumber) and VAF versus tumor purity. Images were finalized in Adobe Illustrator 2021.</p>
<blockquote class="blockquote">
<p>Regardless of tumor purity, mean variant allele frequency is 0.46 ± 0.21 (standard deviation). Variant allele frequency is higher in samples with both mutations and copy number loss compared to mutations alone (two-sided Mann-Whitney U test with continuity correction p=0.002, Figure S2A). Variant allele frequency and tumor purity have a moderate positive correlation (linear regression Wherry adjusted r<sup>2</sup>=0.35, Figure S2B), suggesting that these mutations are less likely to be subclonal.</p>
</blockquote>
<div class="cell">
<div class="sourceCode cell-code" id="cb74"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb74-1"><a href="#cb74-1" aria-hidden="true" tabindex="-1"></a>mean_vaf <span class="ot">&lt;-</span> <span class="fu">round</span>(<span class="fu">median</span>(variants_simple<span class="sc">$</span>VAF), <span class="dv">2</span>)</span>
<span id="cb74-2"><a href="#cb74-2" aria-hidden="true" tabindex="-1"></a>sd_vaf <span class="ot">&lt;-</span> <span class="fu">round</span>(<span class="fu">sd</span>(variants_simple<span class="sc">$</span>VAF), <span class="dv">2</span>)</span>
<span id="cb74-3"><a href="#cb74-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb74-4"><a href="#cb74-4" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="fu">sprintf</span>(<span class="st">"Mean VAF: %s±%s"</span>,</span>
<span id="cb74-5"><a href="#cb74-5" aria-hidden="true" tabindex="-1"></a>            mean_vaf, sd_vaf), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Mean VAF: 0.46±0.21 </code></pre>
</div>
</div>
</section>
<section id="supplemental-figure-s2" class="level3">
<h3 class="anchored" data-anchor-id="supplemental-figure-s2">Supplemental Figure S2</h3>
<p>S2A:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb76"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb76-1"><a href="#cb76-1" aria-hidden="true" tabindex="-1"></a>alteration_type <span class="ot">&lt;-</span> <span class="fu">c</span>()</span>
<span id="cb76-2"><a href="#cb76-2" aria-hidden="true" tabindex="-1"></a>purity <span class="ot">&lt;-</span> <span class="fu">c</span>()</span>
<span id="cb76-3"><a href="#cb76-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb76-4"><a href="#cb76-4" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="fu">c</span>(variants_simple<span class="sc">$</span>SAMPLE,</span>
<span id="cb76-5"><a href="#cb76-5" aria-hidden="true" tabindex="-1"></a>            variants_germline<span class="sc">$</span>bcr_patient_barcode)) {</span>
<span id="cb76-6"><a href="#cb76-6" aria-hidden="true" tabindex="-1"></a>  alteration_type <span class="ot">&lt;-</span> <span class="fu">c</span>(alteration_type, sample_anno<span class="sc">$</span>alteration_type[<span class="fu">substr</span>(sample_anno<span class="sc">$</span>tcga_id, <span class="dv">1</span>, <span class="dv">12</span>) <span class="sc">==</span> i])</span>
<span id="cb76-7"><a href="#cb76-7" aria-hidden="true" tabindex="-1"></a>  purity <span class="ot">&lt;-</span> <span class="fu">c</span>(purity, sample_anno<span class="sc">$</span>purity[<span class="fu">substr</span>(sample_anno<span class="sc">$</span>tcga_id, <span class="dv">1</span>, <span class="dv">12</span>) <span class="sc">==</span> i])</span>
<span id="cb76-8"><a href="#cb76-8" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb76-9"><a href="#cb76-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb76-10"><a href="#cb76-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb76-11"><a href="#cb76-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb76-12"><a href="#cb76-12" aria-hidden="true" tabindex="-1"></a><span class="fu">boxplot</span>(<span class="fu">c</span>(variants_simple<span class="sc">$</span>VAF, </span>
<span id="cb76-13"><a href="#cb76-13" aria-hidden="true" tabindex="-1"></a>          variants_germline<span class="sc">$</span>tumorVAF)<span class="sc">~</span> alteration_type,</span>
<span id="cb76-14"><a href="#cb76-14" aria-hidden="true" tabindex="-1"></a>        <span class="at">names =</span> <span class="fu">c</span>(<span class="st">"Mut"</span>, <span class="st">"Mut+CN"</span>),</span>
<span id="cb76-15"><a href="#cb76-15" aria-hidden="true" tabindex="-1"></a>        <span class="at">xlab =</span> <span class="cn">NULL</span>,</span>
<span id="cb76-16"><a href="#cb76-16" aria-hidden="true" tabindex="-1"></a>        <span class="at">ylab =</span> <span class="st">"Variant allele frequency (VAF)"</span>,</span>
<span id="cb76-17"><a href="#cb76-17" aria-hidden="true" tabindex="-1"></a>        <span class="at">ylim =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">1</span>),</span>
<span id="cb76-18"><a href="#cb76-18" aria-hidden="true" tabindex="-1"></a>        <span class="at">col =</span> <span class="fu">c</span>(<span class="st">"salmon"</span>, <span class="st">"darkgreen"</span>),</span>
<span id="cb76-19"><a href="#cb76-19" aria-hidden="true" tabindex="-1"></a>        <span class="at">main =</span> <span class="st">"Pan-cancer"</span>)</span>
<span id="cb76-20"><a href="#cb76-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb76-21"><a href="#cb76-21" aria-hidden="true" tabindex="-1"></a><span class="fu">stripchart</span>(<span class="fu">c</span>(variants_simple<span class="sc">$</span>VAF, </span>
<span id="cb76-22"><a href="#cb76-22" aria-hidden="true" tabindex="-1"></a>          variants_germline<span class="sc">$</span>tumorVAF) <span class="sc">~</span> alteration_type,</span>
<span id="cb76-23"><a href="#cb76-23" aria-hidden="true" tabindex="-1"></a>           <span class="at">add =</span> <span class="cn">TRUE</span>,</span>
<span id="cb76-24"><a href="#cb76-24" aria-hidden="true" tabindex="-1"></a>           <span class="at">vertical =</span> <span class="cn">TRUE</span>,</span>
<span id="cb76-25"><a href="#cb76-25" aria-hidden="true" tabindex="-1"></a>           <span class="at">method =</span> <span class="st">"jitter"</span>,</span>
<span id="cb76-26"><a href="#cb76-26" aria-hidden="true" tabindex="-1"></a>           <span class="at">pch =</span> <span class="dv">16</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="figure_code_files/figure-html/unnamed-chunk-25-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<blockquote class="blockquote">
<p>Variant allele frequency is higher in samples with both mutations and copy number loss compared to mutations alone (two-sided Mann-Whitney U test with continuity correction p=0.002, Figure S2A).</p>
</blockquote>
<div class="cell">
<div class="sourceCode cell-code" id="cb77"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb77-1"><a href="#cb77-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Wilcoxon rank sum = Mann-Whitney U test</span></span>
<span id="cb77-2"><a href="#cb77-2" aria-hidden="true" tabindex="-1"></a>mu_p <span class="ot">&lt;-</span> <span class="fu">round</span>(<span class="fu">wilcox.test</span>(<span class="fu">c</span>(variants_simple<span class="sc">$</span>VAF,</span>
<span id="cb77-3"><a href="#cb77-3" aria-hidden="true" tabindex="-1"></a>                            variants_germline<span class="sc">$</span>tumorVAF) <span class="sc">~</span></span>
<span id="cb77-4"><a href="#cb77-4" aria-hidden="true" tabindex="-1"></a>                            alteration_type,</span>
<span id="cb77-5"><a href="#cb77-5" aria-hidden="true" tabindex="-1"></a>                          <span class="at">alternative =</span> <span class="st">"two.sided"</span>,</span>
<span id="cb77-6"><a href="#cb77-6" aria-hidden="true" tabindex="-1"></a>                          <span class="at">correct =</span> <span class="cn">TRUE</span>)<span class="sc">$</span>p.val, <span class="dv">3</span>)</span>
<span id="cb77-7"><a href="#cb77-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb77-8"><a href="#cb77-8" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="fu">sprintf</span>(<span class="st">"Two-sided Mann-Whitney U test with continuity correction p=%s"</span>,</span>
<span id="cb77-9"><a href="#cb77-9" aria-hidden="true" tabindex="-1"></a>            mu_p), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Two-sided Mann-Whitney U test with continuity correction p=0.002 </code></pre>
</div>
</div>
<p>S2B:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb79"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb79-1"><a href="#cb79-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(purity,</span>
<span id="cb79-2"><a href="#cb79-2" aria-hidden="true" tabindex="-1"></a>     <span class="fu">c</span>(variants_simple<span class="sc">$</span>VAF,</span>
<span id="cb79-3"><a href="#cb79-3" aria-hidden="true" tabindex="-1"></a>       variants_germline<span class="sc">$</span>tumorVAF),</span>
<span id="cb79-4"><a href="#cb79-4" aria-hidden="true" tabindex="-1"></a>     <span class="at">xlab =</span> <span class="st">"Tumor purity"</span>,</span>
<span id="cb79-5"><a href="#cb79-5" aria-hidden="true" tabindex="-1"></a>     <span class="at">ylab =</span> <span class="st">"Variant allele frequency (VAF)"</span>,</span>
<span id="cb79-6"><a href="#cb79-6" aria-hidden="true" tabindex="-1"></a>     <span class="at">xlim =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">1</span>),</span>
<span id="cb79-7"><a href="#cb79-7" aria-hidden="true" tabindex="-1"></a>     <span class="at">ylim =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">1</span>),</span>
<span id="cb79-8"><a href="#cb79-8" aria-hidden="true" tabindex="-1"></a>     <span class="at">col =</span> <span class="fu">ifelse</span>(alteration_type <span class="sc">==</span> <span class="st">"mutation"</span>,</span>
<span id="cb79-9"><a href="#cb79-9" aria-hidden="true" tabindex="-1"></a>                  <span class="st">"salmon"</span>,</span>
<span id="cb79-10"><a href="#cb79-10" aria-hidden="true" tabindex="-1"></a>                  <span class="st">"darkgreen"</span>),</span>
<span id="cb79-11"><a href="#cb79-11" aria-hidden="true" tabindex="-1"></a>     <span class="at">pch =</span> <span class="dv">16</span>,</span>
<span id="cb79-12"><a href="#cb79-12" aria-hidden="true" tabindex="-1"></a>     <span class="at">main =</span> <span class="st">"Pan-cancer"</span>)</span>
<span id="cb79-13"><a href="#cb79-13" aria-hidden="true" tabindex="-1"></a><span class="fu">abline</span>(<span class="at">a =</span> <span class="dv">0</span>, <span class="at">b =</span> <span class="dv">1</span>, <span class="at">lty =</span> <span class="st">"dotted"</span>, <span class="at">col =</span> <span class="st">"grey"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="figure_code_files/figure-html/unnamed-chunk-27-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<blockquote class="blockquote">
<p>Variant allele frequency and tumor purity have a moderate positive correlation (linear regression Wherry adjusted r<sup>2</sup>=0.35, Figure S2B),</p>
</blockquote>
<div class="cell">
<div class="sourceCode cell-code" id="cb80"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb80-1"><a href="#cb80-1" aria-hidden="true" tabindex="-1"></a>lm_sum <span class="ot">&lt;-</span> <span class="fu">round</span>(<span class="fu">summary</span>(<span class="fu">lm</span>(<span class="fu">c</span>(variants_simple<span class="sc">$</span>VAF,</span>
<span id="cb80-2"><a href="#cb80-2" aria-hidden="true" tabindex="-1"></a>                             variants_germline<span class="sc">$</span>tumorVAF) <span class="sc">~</span></span>
<span id="cb80-3"><a href="#cb80-3" aria-hidden="true" tabindex="-1"></a>                             purity))<span class="sc">$</span>adj.r.squared, <span class="dv">2</span>)</span>
<span id="cb80-4"><a href="#cb80-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb80-5"><a href="#cb80-5" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="fu">sprintf</span>(<span class="st">"Linear model adjusted r-squared: %s"</span>, lm_sum), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Linear model adjusted r-squared: 0.35 </code></pre>
</div>
</div>
<blockquote class="blockquote">
<p>The most highly-altered cancer types for BAP1 are KIRC, CHOL, UVM, MESO, LUSC and HNSC – all with &gt;30% altered samples within their tumor types (Figure 2B and Supplemental Table S3A-S3B).</p>
</blockquote>
<div class="cell">
<div class="sourceCode cell-code" id="cb82"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb82-1"><a href="#cb82-1" aria-hidden="true" tabindex="-1"></a>most_altered <span class="ot">&lt;-</span> cohort_anno<span class="sc">$</span>Cohort[cohort_anno<span class="sc">$</span>Altered <span class="sc">/</span> </span>
<span id="cb82-2"><a href="#cb82-2" aria-hidden="true" tabindex="-1"></a>                                        cohort_anno<span class="sc">$</span>Cohort.size <span class="sc">&gt;</span> <span class="fl">0.3</span>]</span>
<span id="cb82-3"><a href="#cb82-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb82-4"><a href="#cb82-4" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="fu">sprintf</span>(<span class="st">"Cancer types with &gt;30 percent altered samples: %s"</span>,</span>
<span id="cb82-5"><a href="#cb82-5" aria-hidden="true" tabindex="-1"></a>            <span class="fu">paste</span>(most_altered, <span class="at">collapse =</span> <span class="st">", "</span>)), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Cancer types with &gt;30 percent altered samples: KIRC, CHOL, MESO, UVM, LUSC, HNSC </code></pre>
</div>
</div>
</section>
<section id="figure-2b" class="level3">
<h3 class="anchored" data-anchor-id="figure-2b">Figure 2B</h3>
<p>Stacked barplot of % altered samples for each cancer type, sorted by highest overall % alterations. Image was finalized in Adobe Illustrator 2021.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb84"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb84-1"><a href="#cb84-1" aria-hidden="true" tabindex="-1"></a>cohort_anno<span class="sc">$</span>cohort <span class="ot">&lt;-</span> <span class="fu">rownames</span>(cohort_anno)</span>
<span id="cb84-2"><a href="#cb84-2" aria-hidden="true" tabindex="-1"></a>keep_cols <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Cohort"</span>, <span class="st">"Mutation.only"</span>, <span class="st">"Copynumber.only"</span>,</span>
<span id="cb84-3"><a href="#cb84-3" aria-hidden="true" tabindex="-1"></a>                              <span class="st">"Mutation.Copynumber"</span>, <span class="st">"Unaltered"</span>, <span class="st">"Cohort.size"</span>)</span>
<span id="cb84-4"><a href="#cb84-4" aria-hidden="true" tabindex="-1"></a>percent_cols <span class="ot">&lt;-</span> keep_cols[<span class="sc">!</span>(keep_cols <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Cohort"</span>, <span class="st">"Cohort.size"</span>))]</span>
<span id="cb84-5"><a href="#cb84-5" aria-hidden="true" tabindex="-1"></a>altered_df <span class="ot">&lt;-</span> cohort_anno[, keep_cols] </span>
<span id="cb84-6"><a href="#cb84-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb84-7"><a href="#cb84-7" aria-hidden="true" tabindex="-1"></a>altered_df[, percent_cols] <span class="ot">&lt;-</span> altered_df[, percent_cols] <span class="sc">/</span> altered_df<span class="sc">$</span>Cohort.size</span>
<span id="cb84-8"><a href="#cb84-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb84-9"><a href="#cb84-9" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(altered_df) <span class="ot">&lt;-</span> altered_df<span class="sc">$</span>Cohort</span>
<span id="cb84-10"><a href="#cb84-10" aria-hidden="true" tabindex="-1"></a>altered_df <span class="ot">&lt;-</span> altered_df[, <span class="sc">-</span><span class="fu">which</span>(<span class="fu">colnames</span>(altered_df) <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"Cohort"</span>, <span class="st">"Cohort.size"</span>))]</span>
<span id="cb84-11"><a href="#cb84-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb84-12"><a href="#cb84-12" aria-hidden="true" tabindex="-1"></a>altered_df <span class="ot">&lt;-</span> altered_df[, <span class="fu">c</span>(<span class="st">"Copynumber.only"</span>, <span class="st">"Mutation.Copynumber"</span>,</span>
<span id="cb84-13"><a href="#cb84-13" aria-hidden="true" tabindex="-1"></a>                             <span class="st">"Mutation.only"</span>, <span class="st">"Unaltered"</span>)]</span>
<span id="cb84-14"><a href="#cb84-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb84-15"><a href="#cb84-15" aria-hidden="true" tabindex="-1"></a>altered_df <span class="ot">&lt;-</span> altered_df <span class="sc">*</span> <span class="dv">100</span></span>
<span id="cb84-16"><a href="#cb84-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb84-17"><a href="#cb84-17" aria-hidden="true" tabindex="-1"></a><span class="fu">barplot</span>(<span class="fu">t</span>(altered_df), <span class="at">col =</span> <span class="fu">c</span>(<span class="st">"lightgreen"</span>, <span class="st">"darkgreen"</span>, <span class="st">"salmon"</span>, <span class="st">"lightgrey"</span>), <span class="at">las =</span> <span class="dv">2</span>, <span class="at">cex.names =</span> <span class="fl">0.7</span>, <span class="at">ylab =</span> <span class="st">"% of cohort altered"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="figure_code_files/figure-html/unnamed-chunk-30-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="figure-s3" class="level3">
<h3 class="anchored" data-anchor-id="figure-s3">Figure S3</h3>
<p>Comparison of BAP1 RNA-level expression by alteration type. Image was finalized in Adobe Illustrator 2021.</p>
<blockquote class="blockquote">
<p>All alteration types, regardless of cancer type, resulted in lower RNA-level expression of <em>BAP1</em> than unaltered samples (two-sided pairwise Mann-Whitney U test with continuity correction p&lt;0.01 for each alteration type, Figure S3 and Supplemental Table S3B).</p>
</blockquote>
<div class="cell">
<div class="sourceCode cell-code" id="cb85"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb85-1"><a href="#cb85-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb85-2"><a href="#cb85-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb85-3"><a href="#cb85-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Using the subset of data for which we have tumor RNA expression data and were able to call altered/unaltered</span></span>
<span id="cb85-4"><a href="#cb85-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb85-5"><a href="#cb85-5" aria-hidden="true" tabindex="-1"></a>sample_subset <span class="ot">&lt;-</span> sample_anno[<span class="sc">!</span><span class="fu">is.na</span>(sample_anno<span class="sc">$</span>alteration_type) <span class="sc">&amp;</span> </span>
<span id="cb85-6"><a href="#cb85-6" aria-hidden="true" tabindex="-1"></a>                               <span class="sc">!</span><span class="fu">is.na</span>(sample_anno<span class="sc">$</span>log2_bap1), ]</span>
<span id="cb85-7"><a href="#cb85-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb85-8"><a href="#cb85-8" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(sample_subset, </span>
<span id="cb85-9"><a href="#cb85-9" aria-hidden="true" tabindex="-1"></a>       <span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">factor</span>(alteration_type, </span>
<span id="cb85-10"><a href="#cb85-10" aria-hidden="true" tabindex="-1"></a>                      <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"unaltered"</span>, <span class="st">"copynumber"</span>, <span class="st">"mutation+copynumber"</span>, <span class="st">"mutation"</span>)),</span>
<span id="cb85-11"><a href="#cb85-11" aria-hidden="true" tabindex="-1"></a>           <span class="at">y =</span> log2_bap1)) <span class="sc">+</span> </span>
<span id="cb85-12"><a href="#cb85-12" aria-hidden="true" tabindex="-1"></a>  ggdist<span class="sc">::</span><span class="fu">stat_halfeye</span>(</span>
<span id="cb85-13"><a href="#cb85-13" aria-hidden="true" tabindex="-1"></a>    <span class="at">adjust =</span> <span class="fl">0.5</span>,</span>
<span id="cb85-14"><a href="#cb85-14" aria-hidden="true" tabindex="-1"></a>    <span class="at">width =</span> <span class="fl">0.6</span>,</span>
<span id="cb85-15"><a href="#cb85-15" aria-hidden="true" tabindex="-1"></a>    <span class="at">.width =</span> <span class="dv">0</span>,</span>
<span id="cb85-16"><a href="#cb85-16" aria-hidden="true" tabindex="-1"></a>    <span class="at">justification =</span> <span class="sc">-</span><span class="fl">0.2</span>,</span>
<span id="cb85-17"><a href="#cb85-17" aria-hidden="true" tabindex="-1"></a>    <span class="at">point_colour =</span> <span class="cn">NA</span>,</span>
<span id="cb85-18"><a href="#cb85-18" aria-hidden="true" tabindex="-1"></a>    <span class="at">show.legend =</span> F</span>
<span id="cb85-19"><a href="#cb85-19" aria-hidden="true" tabindex="-1"></a>) <span class="sc">+</span></span>
<span id="cb85-20"><a href="#cb85-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="at">outlier.shape =</span> <span class="cn">NA</span>, </span>
<span id="cb85-21"><a href="#cb85-21" aria-hidden="true" tabindex="-1"></a>               <span class="at">width =</span> <span class="fl">0.12</span>, </span>
<span id="cb85-22"><a href="#cb85-22" aria-hidden="true" tabindex="-1"></a>               <span class="at">fill =</span> <span class="fu">c</span>(<span class="st">"grey"</span>, <span class="st">"lightgreen"</span>, <span class="st">"darkgreen"</span>, <span class="st">"salmon"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="figure_code_files/figure-html/unnamed-chunk-31-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb86"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb86-1"><a href="#cb86-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Pairwise two-sided Mann-Whitney U test with continuity correction</span></span>
<span id="cb86-2"><a href="#cb86-2" aria-hidden="true" tabindex="-1"></a><span class="fu">pairwise.wilcox.test</span>(sample_subset<span class="sc">$</span>log2_bap1,</span>
<span id="cb86-3"><a href="#cb86-3" aria-hidden="true" tabindex="-1"></a>                     sample_subset<span class="sc">$</span>alteration_type,</span>
<span id="cb86-4"><a href="#cb86-4" aria-hidden="true" tabindex="-1"></a>                     <span class="at">p.adjust.method =</span> <span class="st">"bonferroni"</span>,</span>
<span id="cb86-5"><a href="#cb86-5" aria-hidden="true" tabindex="-1"></a>                     <span class="at">alternative =</span> <span class="st">"two.sided"</span>,</span>
<span id="cb86-6"><a href="#cb86-6" aria-hidden="true" tabindex="-1"></a>                     <span class="at">correct =</span> <span class="cn">TRUE</span>)<span class="sc">$</span>p.value</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                       copynumber     mutation mutation+copynumber
mutation             1.083904e-04           NA                  NA
mutation+copynumber  4.524204e-12 2.796071e-02                  NA
unaltered           4.812718e-129 4.800928e-21        4.835604e-33</code></pre>
</div>
</div>
</section>
</section>
<section id="figure-3---bap1-activity-scores-capture-bap1-driven-changes" class="level2">
<h2 class="anchored" data-anchor-id="figure-3---bap1-activity-scores-capture-bap1-driven-changes">Figure 3 - <em>BAP1</em> activity scores capture <em>BAP1</em>-driven changes</h2>
<blockquote class="blockquote">
<p>Of 16,333 expressed protein coding genes tested in this manner, 3966 (24.3%) were significantly changed at an adjusted p-value less than 0.05 (Figure 3A and Supplemental Table S4A-S4B).</p>
</blockquote>
<div class="cell">
<div class="sourceCode cell-code" id="cb88"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb88-1"><a href="#cb88-1" aria-hidden="true" tabindex="-1"></a>resLFC <span class="ot">&lt;-</span> <span class="fu">read.delim</span>(<span class="st">"tables/table_s4a.txt"</span>)</span>
<span id="cb88-2"><a href="#cb88-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb88-3"><a href="#cb88-3" aria-hidden="true" tabindex="-1"></a>resLFC_sig <span class="ot">&lt;-</span> resLFC[resLFC<span class="sc">$</span>padj <span class="sc">&lt;</span> <span class="fl">0.05</span>, ]</span>
<span id="cb88-4"><a href="#cb88-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb88-5"><a href="#cb88-5" aria-hidden="true" tabindex="-1"></a>num_total <span class="ot">&lt;-</span> <span class="fu">nrow</span>(resLFC)</span>
<span id="cb88-6"><a href="#cb88-6" aria-hidden="true" tabindex="-1"></a>num_sig <span class="ot">&lt;-</span> <span class="fu">nrow</span>(resLFC_sig)</span>
<span id="cb88-7"><a href="#cb88-7" aria-hidden="true" tabindex="-1"></a>pcnt_sig <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="fu">round</span>(num_sig<span class="sc">/</span>num_total <span class="sc">*</span> <span class="dv">100</span>, <span class="dv">1</span>), <span class="st">"%"</span>)</span>
<span id="cb88-8"><a href="#cb88-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb88-9"><a href="#cb88-9" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="fu">sprintf</span>(<span class="st">"%s/%s (%s) genes significant at padj&lt;0.05"</span>,</span>
<span id="cb88-10"><a href="#cb88-10" aria-hidden="true" tabindex="-1"></a>            num_sig, num_total, pcnt_sig))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>3966/16333 (24.3%) genes significant at padj&lt;0.05</code></pre>
</div>
</div>
<section id="figure-3a" class="level3">
<h3 class="anchored" data-anchor-id="figure-3a">Figure 3A</h3>
<p>This is a simple volcano plot of differentially expressed genes between samples with <em>BAP1</em> mutations (Mut or Mut+CN) and unaltered samples in 5 tumor types with a higher representation of mutant samples. These differential expression results are from DESeq2 and used Wald testing with shrunken log2 fold changes using apeglm. P-values in DESeq2 are adjusted using the Benjamini-Hochberg procedure. Image was finalized in Adobe Illustrator 2021.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb90"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb90-1"><a href="#cb90-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(EnhancedVolcano)</span>
<span id="cb90-2"><a href="#cb90-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-3"><a href="#cb90-3" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">EnhancedVolcano</span>(resLFC, <span class="at">lab =</span> resLFC<span class="sc">$</span>symbol, <span class="at">x =</span> <span class="st">"log2FoldChange"</span>, </span>
<span id="cb90-4"><a href="#cb90-4" aria-hidden="true" tabindex="-1"></a>                      <span class="at">y =</span> <span class="st">"padj"</span>, <span class="at">pCutoff =</span> <span class="fl">0.05</span>, <span class="at">FCcutoff =</span> <span class="fl">0.5</span>, </span>
<span id="cb90-5"><a href="#cb90-5" aria-hidden="true" tabindex="-1"></a>                      <span class="at">legendPosition =</span> <span class="st">"right"</span>, </span>
<span id="cb90-6"><a href="#cb90-6" aria-hidden="true" tabindex="-1"></a>                      <span class="at">legendLabels =</span> <span class="fu">c</span>(<span class="st">"NS"</span>, <span class="st">"Log2 FC"</span>, </span>
<span id="cb90-7"><a href="#cb90-7" aria-hidden="true" tabindex="-1"></a>                                       <span class="st">"adj. p-value"</span>, <span class="st">"adj. p-value &amp; Log2 FC"</span>), </span>
<span id="cb90-8"><a href="#cb90-8" aria-hidden="true" tabindex="-1"></a>                      <span class="at">max.overlaps =</span> <span class="dv">30</span>, <span class="at">labSize =</span> <span class="fl">3.0</span>,</span>
<span id="cb90-9"><a href="#cb90-9" aria-hidden="true" tabindex="-1"></a>                      <span class="at">selectLab =</span> resLFC<span class="sc">$</span>symbol[<span class="fu">abs</span>(resLFC<span class="sc">$</span>log2FoldChange) <span class="sc">&gt;</span> <span class="fl">0.5</span> <span class="sc">&amp;</span></span>
<span id="cb90-10"><a href="#cb90-10" aria-hidden="true" tabindex="-1"></a>                                                  resLFC<span class="sc">$</span>padj <span class="sc">&lt;</span> <span class="fl">0.05</span>]))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="figure_code_files/figure-html/unnamed-chunk-34-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="figure-3b" class="level3">
<h3 class="anchored" data-anchor-id="figure-3b">Figure 3B</h3>
<p>This is a heatmap of differentially expressed genes from the above analysis. We first compute <em>BAP1</em> activity scores to include as annotations in the heatmap. Image was finalized in Adobe Illustrator 2021.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb91"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb91-1"><a href="#cb91-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(DESeq2)</span>
<span id="cb91-2"><a href="#cb91-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb91-3"><a href="#cb91-3" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"../results/activity_signature/mut_vs_unaltered_dds.rds"</span>)</span>
<span id="cb91-4"><a href="#cb91-4" aria-hidden="true" tabindex="-1"></a>mutation_anno <span class="ot">&lt;-</span> <span class="fu">colData</span>(dds)</span>
<span id="cb91-5"><a href="#cb91-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb91-6"><a href="#cb91-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(mclust)</span>
<span id="cb91-7"><a href="#cb91-7" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">1</span>)</span>
<span id="cb91-8"><a href="#cb91-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb91-9"><a href="#cb91-9" aria-hidden="true" tabindex="-1"></a>log2_mediancenter <span class="ot">&lt;-</span> <span class="cf">function</span>(counts) {</span>
<span id="cb91-10"><a href="#cb91-10" aria-hidden="true" tabindex="-1"></a>  counts_transformed <span class="ot">&lt;-</span> <span class="fu">log2</span>(counts <span class="sc">+</span> <span class="dv">1</span>)</span>
<span id="cb91-11"><a href="#cb91-11" aria-hidden="true" tabindex="-1"></a>  counts_medians <span class="ot">&lt;-</span> <span class="fu">apply</span>(counts_transformed, <span class="dv">1</span>, median)</span>
<span id="cb91-12"><a href="#cb91-12" aria-hidden="true" tabindex="-1"></a>  counts_transformed <span class="ot">&lt;-</span> counts_transformed <span class="sc">-</span> counts_medians</span>
<span id="cb91-13"><a href="#cb91-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(counts_transformed)</span>
<span id="cb91-14"><a href="#cb91-14" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb91-15"><a href="#cb91-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb91-16"><a href="#cb91-16" aria-hidden="true" tabindex="-1"></a>mut_counts <span class="ot">&lt;-</span> <span class="fu">counts</span>(dds, <span class="at">normalized =</span> <span class="cn">TRUE</span>)</span>
<span id="cb91-17"><a href="#cb91-17" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb91-18"><a href="#cb91-18" aria-hidden="true" tabindex="-1"></a>mut_counts <span class="ot">&lt;-</span> mut_counts[, mutation_anno<span class="sc">$</span>tcga_id]</span>
<span id="cb91-19"><a href="#cb91-19" aria-hidden="true" tabindex="-1"></a>mut_counts <span class="ot">&lt;-</span> mut_counts[<span class="fu">rownames</span>(mut_counts) <span class="sc">%in%</span> </span>
<span id="cb91-20"><a href="#cb91-20" aria-hidden="true" tabindex="-1"></a>                           resLFC_sig<span class="sc">$</span>ensembl_id[resLFC_sig<span class="sc">$</span>padj <span class="sc">&lt;</span> <span class="fl">0.05</span>], ]</span>
<span id="cb91-21"><a href="#cb91-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb91-22"><a href="#cb91-22" aria-hidden="true" tabindex="-1"></a><span class="co"># We use log2 counts for computing BAP1 mutation scores and median-centered log2 counts for visualization in the heatmap</span></span>
<span id="cb91-23"><a href="#cb91-23" aria-hidden="true" tabindex="-1"></a>mut_counts_log2 <span class="ot">&lt;-</span> <span class="fu">log2</span>(mut_counts <span class="sc">+</span> <span class="dv">1</span>)</span>
<span id="cb91-24"><a href="#cb91-24" aria-hidden="true" tabindex="-1"></a>mut_counts_log2mc <span class="ot">&lt;-</span> <span class="fu">log2_mediancenter</span>(mut_counts)</span>
<span id="cb91-25"><a href="#cb91-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb91-26"><a href="#cb91-26" aria-hidden="true" tabindex="-1"></a><span class="co"># Subset of all significant genes in the up direction</span></span>
<span id="cb91-27"><a href="#cb91-27" aria-hidden="true" tabindex="-1"></a>mut_up_genes <span class="ot">&lt;-</span> <span class="fu">rownames</span>(mut_counts_log2)[<span class="fu">rownames</span>(mut_counts_log2) <span class="sc">%in%</span> resLFC_sig<span class="sc">$</span>ensembl_id[resLFC_sig<span class="sc">$</span>padj <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> <span class="fu">sign</span>(resLFC_sig<span class="sc">$</span>log2FoldChange) <span class="sc">==</span> <span class="st">"1"</span>]]</span>
<span id="cb91-28"><a href="#cb91-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb91-29"><a href="#cb91-29" aria-hidden="true" tabindex="-1"></a><span class="co"># Subset of all significant genes in the down direction</span></span>
<span id="cb91-30"><a href="#cb91-30" aria-hidden="true" tabindex="-1"></a>mut_down_genes <span class="ot">&lt;-</span> <span class="fu">rownames</span>(mut_counts_log2)[<span class="fu">rownames</span>(mut_counts_log2) <span class="sc">%in%</span> resLFC_sig<span class="sc">$</span>ensembl_id[resLFC_sig<span class="sc">$</span>padj <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> <span class="fu">sign</span>(resLFC_sig<span class="sc">$</span>log2FoldChange) <span class="sc">==</span> <span class="st">"-1"</span>]]</span>
<span id="cb91-31"><a href="#cb91-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb91-32"><a href="#cb91-32" aria-hidden="true" tabindex="-1"></a>score_up <span class="ot">&lt;-</span> <span class="fu">colSums</span>(mut_counts_log2[mut_up_genes, ])</span>
<span id="cb91-33"><a href="#cb91-33" aria-hidden="true" tabindex="-1"></a>score_down <span class="ot">&lt;-</span> <span class="fu">colSums</span>(mut_counts_log2[mut_down_genes, ])</span>
<span id="cb91-34"><a href="#cb91-34" aria-hidden="true" tabindex="-1"></a>combined_scores <span class="ot">&lt;-</span> score_down <span class="sc">-</span> score_up</span>
<span id="cb91-35"><a href="#cb91-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb91-36"><a href="#cb91-36" aria-hidden="true" tabindex="-1"></a><span class="co"># We use Mclust to find an approximate threshold between two distributions in the data. In rare cases, a low-value outlier can be assigned to group 2, so we specify that anything below the max value of group 1 should be assigned to group 2</span></span>
<span id="cb91-37"><a href="#cb91-37" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">1</span>)</span>
<span id="cb91-38"><a href="#cb91-38" aria-hidden="true" tabindex="-1"></a>BIC <span class="ot">&lt;-</span> <span class="fu">mclustBIC</span>(combined_scores, <span class="at">G =</span> <span class="dv">2</span>)</span>
<span id="cb91-39"><a href="#cb91-39" aria-hidden="true" tabindex="-1"></a>mod <span class="ot">&lt;-</span> <span class="fu">Mclust</span>(combined_scores, <span class="at">x =</span> BIC)</span>
<span id="cb91-40"><a href="#cb91-40" aria-hidden="true" tabindex="-1"></a>mod<span class="sc">$</span>classification[combined_scores <span class="sc">&lt;</span> <span class="fu">max</span>(combined_scores[mod<span class="sc">$</span>classification <span class="sc">==</span> <span class="dv">1</span>])] <span class="ot">&lt;-</span> <span class="dv">1</span></span>
<span id="cb91-41"><a href="#cb91-41" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb91-42"><a href="#cb91-42" aria-hidden="true" tabindex="-1"></a>mutation_anno<span class="sc">$</span>activity_score <span class="ot">&lt;-</span> combined_scores</span>
<span id="cb91-43"><a href="#cb91-43" aria-hidden="true" tabindex="-1"></a>mutation_anno<span class="sc">$</span>activity_class <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(mod<span class="sc">$</span>classification <span class="sc">==</span> <span class="dv">1</span>,</span>
<span id="cb91-44"><a href="#cb91-44" aria-hidden="true" tabindex="-1"></a>                                       <span class="st">"Mutant-like"</span>, <span class="st">"Wildtype-like"</span>)</span>
<span id="cb91-45"><a href="#cb91-45" aria-hidden="true" tabindex="-1"></a>mutation_anno <span class="ot">&lt;-</span> mutation_anno[<span class="fu">order</span>(mutation_anno<span class="sc">$</span>activity_score, </span>
<span id="cb91-46"><a href="#cb91-46" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">decreasing =</span> <span class="cn">FALSE</span>), ]</span>
<span id="cb91-47"><a href="#cb91-47" aria-hidden="true" tabindex="-1"></a>mut_counts_log2mc <span class="ot">&lt;-</span> mut_counts_log2mc[, <span class="fu">rownames</span>(mutation_anno)]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb92"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb92-1"><a href="#cb92-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(circlize)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>========================================
circlize version 0.4.16
CRAN page: https://cran.r-project.org/package=circlize
Github page: https://github.com/jokergoo/circlize
Documentation: https://jokergoo.github.io/circlize_book/book/

If you use it in published research, please cite:
Gu, Z. circlize implements and enhances circular visualization
  in R. Bioinformatics 2014.

This message can be suppressed by:
  suppressPackageStartupMessages(library(circlize))
========================================</code></pre>
</div>
<div class="sourceCode cell-code" id="cb94"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb94-1"><a href="#cb94-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ComplexHeatmap)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: grid</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>========================================
ComplexHeatmap version 2.20.0
Bioconductor page: http://bioconductor.org/packages/ComplexHeatmap/
Github page: https://github.com/jokergoo/ComplexHeatmap
Documentation: http://jokergoo.github.io/ComplexHeatmap-reference

If you use it in published research, please cite either one:
- Gu, Z. Complex Heatmap Visualization. iMeta 2022.
- Gu, Z. Complex heatmaps reveal patterns and correlations in multidimensional 
    genomic data. Bioinformatics 2016.


The new InteractiveComplexHeatmap package can directly export static 
complex heatmaps into an interactive Shiny app with zero effort. Have a try!

This message can be suppressed by:
  suppressPackageStartupMessages(library(ComplexHeatmap))
========================================</code></pre>
</div>
<div class="sourceCode cell-code" id="cb97"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb97-1"><a href="#cb97-1" aria-hidden="true" tabindex="-1"></a>ht_opt<span class="sc">$</span>message <span class="ot">&lt;-</span> <span class="cn">FALSE</span></span>
<span id="cb97-2"><a href="#cb97-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb97-3"><a href="#cb97-3" aria-hidden="true" tabindex="-1"></a>mut_cohort <span class="ot">&lt;-</span> <span class="fu">c</span>()</span>
<span id="cb97-4"><a href="#cb97-4" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (cohort <span class="cf">in</span> mutation_anno<span class="sc">$</span>cohort) {</span>
<span id="cb97-5"><a href="#cb97-5" aria-hidden="true" tabindex="-1"></a>  mut_cohort <span class="ot">&lt;-</span> <span class="fu">c</span>(mut_cohort, <span class="fu">strsplit</span>(cohort, <span class="st">"-"</span>)[[<span class="dv">1</span>]][<span class="dv">2</span>])</span>
<span id="cb97-6"><a href="#cb97-6" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb97-7"><a href="#cb97-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb97-8"><a href="#cb97-8" aria-hidden="true" tabindex="-1"></a>col_fun <span class="ot">&lt;-</span> <span class="fu">colorRamp2</span>(<span class="fu">c</span>(<span class="fu">min</span>(mutation_anno<span class="sc">$</span>activity_score), </span>
<span id="cb97-9"><a href="#cb97-9" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">mean</span>(<span class="fu">max</span>(mutation_anno<span class="sc">$</span>activity_score[mutation_anno<span class="sc">$</span>activity_class <span class="sc">==</span> <span class="st">"Mutant-like"</span>]), </span>
<span id="cb97-10"><a href="#cb97-10" aria-hidden="true" tabindex="-1"></a>                             <span class="fu">min</span>(mutation_anno<span class="sc">$</span>activity_score[mutation_anno<span class="sc">$</span>activity_class <span class="sc">==</span> <span class="st">"Wildtype-like"</span>])), </span>
<span id="cb97-11"><a href="#cb97-11" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">max</span>(mutation_anno<span class="sc">$</span>activity_score)), </span>
<span id="cb97-12"><a href="#cb97-12" aria-hidden="true" tabindex="-1"></a>                      <span class="fu">c</span>(<span class="st">"red"</span>, <span class="st">"white"</span>, <span class="st">"blue"</span>))</span>
<span id="cb97-13"><a href="#cb97-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb97-14"><a href="#cb97-14" aria-hidden="true" tabindex="-1"></a>alt_types <span class="ot">&lt;-</span> <span class="fu">c</span>()</span>
<span id="cb97-15"><a href="#cb97-15" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (sample_id <span class="cf">in</span> mutation_anno<span class="sc">$</span>tcga_id) {</span>
<span id="cb97-16"><a href="#cb97-16" aria-hidden="true" tabindex="-1"></a>  alt_types <span class="ot">&lt;-</span> <span class="fu">c</span>(alt_types, </span>
<span id="cb97-17"><a href="#cb97-17" aria-hidden="true" tabindex="-1"></a>                 sample_anno<span class="sc">$</span>alteration_type[sample_anno<span class="sc">$</span>tcga_id <span class="sc">==</span> sample_id])</span>
<span id="cb97-18"><a href="#cb97-18" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb97-19"><a href="#cb97-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb97-20"><a href="#cb97-20" aria-hidden="true" tabindex="-1"></a>col_fun2 <span class="ot">&lt;-</span> <span class="fu">colorRamp2</span>(<span class="fu">c</span>(<span class="sc">-</span><span class="dv">2</span>, <span class="dv">0</span>, <span class="dv">2</span>), <span class="fu">c</span>(<span class="st">"blue"</span>, <span class="st">"black"</span>, <span class="st">"yellow"</span>))</span>
<span id="cb97-21"><a href="#cb97-21" aria-hidden="true" tabindex="-1"></a>col_fun3 <span class="ot">&lt;-</span> <span class="fu">colorRamp2</span>(<span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>, <span class="dv">0</span>, <span class="dv">1</span>), <span class="fu">c</span>(<span class="st">"lightblue"</span>, <span class="st">"black"</span>, <span class="st">"yellow"</span>))</span>
<span id="cb97-22"><a href="#cb97-22" aria-hidden="true" tabindex="-1"></a>pal_purity <span class="ot">&lt;-</span> <span class="fu">colorRamp2</span>(<span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">1</span>), <span class="fu">c</span>(<span class="st">"white"</span>, <span class="st">"steelblue"</span>))</span>
<span id="cb97-23"><a href="#cb97-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb97-24"><a href="#cb97-24" aria-hidden="true" tabindex="-1"></a>gene_anno <span class="ot">&lt;-</span> <span class="fu">c</span>()</span>
<span id="cb97-25"><a href="#cb97-25" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (gene <span class="cf">in</span> <span class="fu">rownames</span>(mut_counts_log2mc)) {</span>
<span id="cb97-26"><a href="#cb97-26" aria-hidden="true" tabindex="-1"></a>  gene_anno <span class="ot">&lt;-</span> <span class="fu">c</span>(gene_anno,</span>
<span id="cb97-27"><a href="#cb97-27" aria-hidden="true" tabindex="-1"></a>                 resLFC_sig<span class="sc">$</span>log2FoldChange[resLFC_sig<span class="sc">$</span>ensembl_id <span class="sc">==</span> gene])</span>
<span id="cb97-28"><a href="#cb97-28" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb97-29"><a href="#cb97-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb97-30"><a href="#cb97-30" aria-hidden="true" tabindex="-1"></a>ha_row <span class="ot">&lt;-</span> <span class="fu">rowAnnotation</span>(<span class="at">log2FC =</span> <span class="fu">sort</span>(gene_anno),</span>
<span id="cb97-31"><a href="#cb97-31" aria-hidden="true" tabindex="-1"></a>                        <span class="at">col =</span> <span class="fu">list</span>(<span class="at">log2FC =</span> col_fun3))</span>
<span id="cb97-32"><a href="#cb97-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb97-33"><a href="#cb97-33" aria-hidden="true" tabindex="-1"></a>ha <span class="ot">&lt;-</span> <span class="fu">HeatmapAnnotation</span>(<span class="at">project =</span> mut_cohort, </span>
<span id="cb97-34"><a href="#cb97-34" aria-hidden="true" tabindex="-1"></a>                        <span class="at">purity =</span> mutation_anno<span class="sc">$</span>purity,</span>
<span id="cb97-35"><a href="#cb97-35" aria-hidden="true" tabindex="-1"></a>                        <span class="at">alteration_type =</span> alt_types,</span>
<span id="cb97-36"><a href="#cb97-36" aria-hidden="true" tabindex="-1"></a>                        <span class="at">col =</span> <span class="fu">list</span>(<span class="at">project =</span> tcga_col[<span class="fu">names</span>(tcga_col) <span class="sc">%in%</span></span>
<span id="cb97-37"><a href="#cb97-37" aria-hidden="true" tabindex="-1"></a>                                                        mut_cohort],</span>
<span id="cb97-38"><a href="#cb97-38" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">purity =</span> pal_purity,</span>
<span id="cb97-39"><a href="#cb97-39" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">alteration_type =</span> <span class="fu">c</span>(<span class="st">"copynumber"</span> <span class="ot">=</span> <span class="st">"lightgreen"</span>,</span>
<span id="cb97-40"><a href="#cb97-40" aria-hidden="true" tabindex="-1"></a>                                                         <span class="st">"mutation"</span> <span class="ot">=</span> <span class="st">"salmon"</span>,</span>
<span id="cb97-41"><a href="#cb97-41" aria-hidden="true" tabindex="-1"></a>                                                         <span class="st">"mutation+copynumber"</span> <span class="ot">=</span> <span class="st">"darkgreen"</span>, <span class="st">"unaltered"</span> <span class="ot">=</span> <span class="st">"grey"</span>)))</span>
<span id="cb97-42"><a href="#cb97-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb97-43"><a href="#cb97-43" aria-hidden="true" tabindex="-1"></a>ha_b <span class="ot">&lt;-</span> <span class="fu">HeatmapAnnotation</span>(<span class="at">act_sig =</span> mutation_anno<span class="sc">$</span>activity_score, </span>
<span id="cb97-44"><a href="#cb97-44" aria-hidden="true" tabindex="-1"></a>                          <span class="at">act_class =</span> mutation_anno<span class="sc">$</span>activity_class,</span>
<span id="cb97-45"><a href="#cb97-45" aria-hidden="true" tabindex="-1"></a>                          <span class="at">col =</span> <span class="fu">list</span>(<span class="at">act_sig =</span> col_fun,</span>
<span id="cb97-46"><a href="#cb97-46" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">act_class =</span> <span class="fu">c</span>(<span class="st">"Mutant-like"</span> <span class="ot">=</span> <span class="st">"salmon"</span>, </span>
<span id="cb97-47"><a href="#cb97-47" aria-hidden="true" tabindex="-1"></a>                                                   <span class="st">"Wildtype-like"</span> <span class="ot">=</span> <span class="st">"grey"</span>)))</span>
<span id="cb97-48"><a href="#cb97-48" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb97-49"><a href="#cb97-49" aria-hidden="true" tabindex="-1"></a>ht <span class="ot">&lt;-</span> <span class="fu">Heatmap</span>(<span class="fu">as.matrix</span>(mut_counts_log2mc[<span class="fu">order</span>(gene_anno), ]),</span>
<span id="cb97-50"><a href="#cb97-50" aria-hidden="true" tabindex="-1"></a>              <span class="at">top_annotation =</span> ha, <span class="at">bottom_annotation =</span> ha_b,</span>
<span id="cb97-51"><a href="#cb97-51" aria-hidden="true" tabindex="-1"></a>              <span class="at">right_annotation =</span> ha_row,</span>
<span id="cb97-52"><a href="#cb97-52" aria-hidden="true" tabindex="-1"></a>              <span class="at">show_column_names =</span> <span class="cn">FALSE</span>, <span class="at">show_row_names =</span> <span class="cn">FALSE</span>,</span>
<span id="cb97-53"><a href="#cb97-53" aria-hidden="true" tabindex="-1"></a>              <span class="at">cluster_rows =</span> <span class="cn">FALSE</span>, <span class="at">cluster_columns =</span> <span class="cn">FALSE</span>,</span>
<span id="cb97-54"><a href="#cb97-54" aria-hidden="true" tabindex="-1"></a>              <span class="at">col =</span> col_fun2, <span class="at">name =</span> <span class="st">"Expression"</span>)</span>
<span id="cb97-55"><a href="#cb97-55" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb97-56"><a href="#cb97-56" aria-hidden="true" tabindex="-1"></a><span class="fu">draw</span>(ht)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="figure_code_files/figure-html/unnamed-chunk-36-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<blockquote class="blockquote">
<p>Among the five tumor types in this analysis, 57/59 (96.6%) of mutant tumors were classified as mutant-like in this manner.</p>
</blockquote>
<div class="cell">
<div class="sourceCode cell-code" id="cb98"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb98-1"><a href="#cb98-1" aria-hidden="true" tabindex="-1"></a>num_mut <span class="ot">&lt;-</span> <span class="fu">sum</span>(<span class="fu">grepl</span>(<span class="st">"mutation"</span>, mutation_anno<span class="sc">$</span>alteration_type))</span>
<span id="cb98-2"><a href="#cb98-2" aria-hidden="true" tabindex="-1"></a>num_mutlike_mut <span class="ot">&lt;-</span> <span class="fu">sum</span>(<span class="fu">grepl</span>(<span class="st">"mutation"</span>, mutation_anno<span class="sc">$</span>alteration_type) <span class="sc">&amp;</span></span>
<span id="cb98-3"><a href="#cb98-3" aria-hidden="true" tabindex="-1"></a>                         mutation_anno<span class="sc">$</span>activity_class <span class="sc">==</span> <span class="st">"Mutant-like"</span>)</span>
<span id="cb98-4"><a href="#cb98-4" aria-hidden="true" tabindex="-1"></a>pcnt_mutlike_mut <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="fu">round</span>(num_mutlike_mut<span class="sc">/</span>num_mut <span class="sc">*</span> <span class="dv">100</span>, <span class="dv">1</span>), <span class="st">"%"</span>)</span>
<span id="cb98-5"><a href="#cb98-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb98-6"><a href="#cb98-6" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="fu">sprintf</span>(<span class="st">"%s/%s (%s) of mutant tumors classified as mutant-like"</span>,</span>
<span id="cb98-7"><a href="#cb98-7" aria-hidden="true" tabindex="-1"></a>            num_mutlike_mut, num_mut, pcnt_mutlike_mut))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>57/59 (96.6%) of mutant tumors classified as mutant-like</code></pre>
</div>
</div>
<blockquote class="blockquote">
<p>We then applied this mutation signature scoring method to the rest of the TCGA dataset, where 117/233 (50.2%) of mutant tumors were classified as mutant-like, with mutant tumors having lower activity scores and tumors with <em>BAP1</em> mutation plus copy number loss having the lowest scores (Figure 3C).</p>
</blockquote>
<div class="cell">
<div class="sourceCode cell-code" id="cb100"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb100-1"><a href="#cb100-1" aria-hidden="true" tabindex="-1"></a>num_pancan_mut <span class="ot">&lt;-</span> <span class="fu">sum</span>(<span class="fu">grepl</span>(<span class="st">"mutation"</span>, sample_anno<span class="sc">$</span>alteration_type), <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb100-2"><a href="#cb100-2" aria-hidden="true" tabindex="-1"></a>num_pancan_mutlike_mut <span class="ot">&lt;-</span> <span class="fu">sum</span>(<span class="fu">grepl</span>(<span class="st">"mutation"</span>, sample_anno<span class="sc">$</span>alteration_type) <span class="sc">&amp;</span> </span>
<span id="cb100-3"><a href="#cb100-3" aria-hidden="true" tabindex="-1"></a>                                sample_anno<span class="sc">$</span>bap1_activity_class <span class="sc">==</span> <span class="st">"Mutant-like"</span>,</span>
<span id="cb100-4"><a href="#cb100-4" aria-hidden="true" tabindex="-1"></a>                              <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb100-5"><a href="#cb100-5" aria-hidden="true" tabindex="-1"></a>pcnt_pancan_mutlike_mut <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="fu">round</span>(num_pancan_mutlike_mut<span class="sc">/</span>num_pancan_mut <span class="sc">*</span> <span class="dv">100</span>, <span class="dv">1</span>), <span class="st">"%"</span>)</span>
<span id="cb100-6"><a href="#cb100-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb100-7"><a href="#cb100-7" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="fu">sprintf</span>(<span class="st">"%s/%s (%s) of mutant tumors classified as mutant-like"</span>,</span>
<span id="cb100-8"><a href="#cb100-8" aria-hidden="true" tabindex="-1"></a>            num_pancan_mutlike_mut, num_pancan_mut, pcnt_pancan_mutlike_mut))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>117/233 (50.2%) of mutant tumors classified as mutant-like</code></pre>
</div>
</div>
</section>
<section id="figure-3c" class="level3">
<h3 class="anchored" data-anchor-id="figure-3c">Figure 3C</h3>
<p>This is a modified half-eye plot showing <em>BAP1</em> activity scores for each alteration type, with a reference for the threshold for mutant-like classification. Image was finalized in Adobe Illustrator 2021.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb102"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb102-1"><a href="#cb102-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(sample_anno[<span class="sc">!</span><span class="fu">is.na</span>(sample_anno<span class="sc">$</span>alteration_type) <span class="sc">&amp;</span></span>
<span id="cb102-2"><a href="#cb102-2" aria-hidden="true" tabindex="-1"></a>                     <span class="sc">!</span><span class="fu">is.na</span>(sample_anno<span class="sc">$</span>bap1_activity_score), ], </span>
<span id="cb102-3"><a href="#cb102-3" aria-hidden="true" tabindex="-1"></a>       <span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">factor</span>(alteration_type, </span>
<span id="cb102-4"><a href="#cb102-4" aria-hidden="true" tabindex="-1"></a>                      <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"unaltered"</span>, <span class="st">"copynumber"</span>, </span>
<span id="cb102-5"><a href="#cb102-5" aria-hidden="true" tabindex="-1"></a>                                 <span class="st">"mutation+copynumber"</span>, <span class="st">"mutation"</span>)),</span>
<span id="cb102-6"><a href="#cb102-6" aria-hidden="true" tabindex="-1"></a>           <span class="at">y =</span> bap1_activity_score)) <span class="sc">+</span> </span>
<span id="cb102-7"><a href="#cb102-7" aria-hidden="true" tabindex="-1"></a>  ggdist<span class="sc">::</span><span class="fu">stat_halfeye</span>(</span>
<span id="cb102-8"><a href="#cb102-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">adjust =</span> <span class="fl">0.5</span>,</span>
<span id="cb102-9"><a href="#cb102-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">width =</span> <span class="fl">0.6</span>,</span>
<span id="cb102-10"><a href="#cb102-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">.width =</span> <span class="dv">0</span>,</span>
<span id="cb102-11"><a href="#cb102-11" aria-hidden="true" tabindex="-1"></a>    <span class="at">justification =</span> <span class="sc">-</span><span class="fl">0.2</span>,</span>
<span id="cb102-12"><a href="#cb102-12" aria-hidden="true" tabindex="-1"></a>    <span class="at">point_colour =</span> <span class="cn">NA</span>,</span>
<span id="cb102-13"><a href="#cb102-13" aria-hidden="true" tabindex="-1"></a>    <span class="at">show.legend =</span> F</span>
<span id="cb102-14"><a href="#cb102-14" aria-hidden="true" tabindex="-1"></a>) <span class="sc">+</span></span>
<span id="cb102-15"><a href="#cb102-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">"BAP1 alteration type"</span>) <span class="sc">+</span></span>
<span id="cb102-16"><a href="#cb102-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"BAP1 activity score"</span>) <span class="sc">+</span></span>
<span id="cb102-17"><a href="#cb102-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="fu">max</span>(mutation_anno<span class="sc">$</span>activity_score[mutation_anno<span class="sc">$</span>activity_class <span class="sc">==</span> <span class="st">"Mutant-like"</span>]),</span>
<span id="cb102-18"><a href="#cb102-18" aria-hidden="true" tabindex="-1"></a>             <span class="at">linetype =</span> <span class="st">"dotted"</span>) <span class="sc">+</span></span>
<span id="cb102-19"><a href="#cb102-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="at">outlier.shape =</span> <span class="cn">NA</span>, </span>
<span id="cb102-20"><a href="#cb102-20" aria-hidden="true" tabindex="-1"></a>               <span class="at">width =</span> <span class="fl">0.12</span>, </span>
<span id="cb102-21"><a href="#cb102-21" aria-hidden="true" tabindex="-1"></a>               <span class="at">fill =</span> <span class="fu">c</span>(<span class="st">"grey"</span>, <span class="st">"lightgreen"</span>, <span class="st">"salmon"</span>, <span class="st">"red"</span>)) <span class="sc">+</span></span>
<span id="cb102-22"><a href="#cb102-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="figure_code_files/figure-html/unnamed-chunk-39-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb103"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb103-1"><a href="#cb103-1" aria-hidden="true" tabindex="-1"></a><span class="fu">pairwise.wilcox.test</span>(sample_anno<span class="sc">$</span>bap1_activity_score, sample_anno<span class="sc">$</span>alteration_type,</span>
<span id="cb103-2"><a href="#cb103-2" aria-hidden="true" tabindex="-1"></a>                     <span class="at">p.adjust.method =</span> <span class="st">"bonferroni"</span>)<span class="sc">$</span>p.val</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                      copynumber     mutation mutation+copynumber
mutation            9.241534e-18           NA                  NA
mutation+copynumber 8.493848e-47 8.020092e-09                  NA
unaltered           6.048609e-73 5.793543e-36        3.993868e-57</code></pre>
</div>
</div>
</section>
<section id="figure-3d" class="level3">
<h3 class="anchored" data-anchor-id="figure-3d">Figure 3D</h3>
<p>This figure shows RNA expression of <em>BAP1</em> for tumors in the two activity score-based classifications. Image was finalized in Adobe Illustrator 2021.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb105"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb105-1"><a href="#cb105-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(sample_anno[<span class="sc">!</span><span class="fu">is.na</span>(sample_anno<span class="sc">$</span>bap1_activity_class) <span class="sc">&amp;</span></span>
<span id="cb105-2"><a href="#cb105-2" aria-hidden="true" tabindex="-1"></a>                     <span class="sc">!</span><span class="fu">is.na</span>(sample_anno<span class="sc">$</span>log2_bap1), ], </span>
<span id="cb105-3"><a href="#cb105-3" aria-hidden="true" tabindex="-1"></a>       <span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">factor</span>(bap1_activity_class, </span>
<span id="cb105-4"><a href="#cb105-4" aria-hidden="true" tabindex="-1"></a>                      <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"Mutant-like"</span>, <span class="st">"Wildtype-like"</span>)),</span>
<span id="cb105-5"><a href="#cb105-5" aria-hidden="true" tabindex="-1"></a>           <span class="at">y =</span> log2_bap1)) <span class="sc">+</span> </span>
<span id="cb105-6"><a href="#cb105-6" aria-hidden="true" tabindex="-1"></a>  ggdist<span class="sc">::</span><span class="fu">stat_halfeye</span>(</span>
<span id="cb105-7"><a href="#cb105-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">adjust =</span> <span class="fl">0.5</span>,</span>
<span id="cb105-8"><a href="#cb105-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">width =</span> <span class="fl">0.6</span>,</span>
<span id="cb105-9"><a href="#cb105-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">.width =</span> <span class="dv">0</span>,</span>
<span id="cb105-10"><a href="#cb105-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">justification =</span> <span class="sc">-</span><span class="fl">0.2</span>,</span>
<span id="cb105-11"><a href="#cb105-11" aria-hidden="true" tabindex="-1"></a>    <span class="at">point_colour =</span> <span class="cn">NA</span>,</span>
<span id="cb105-12"><a href="#cb105-12" aria-hidden="true" tabindex="-1"></a>    <span class="at">show.legend =</span> F</span>
<span id="cb105-13"><a href="#cb105-13" aria-hidden="true" tabindex="-1"></a>) <span class="sc">+</span></span>
<span id="cb105-14"><a href="#cb105-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">"BAP1 activity class"</span>) <span class="sc">+</span></span>
<span id="cb105-15"><a href="#cb105-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"Log2 BAP1 RNA expression"</span>) <span class="sc">+</span></span>
<span id="cb105-16"><a href="#cb105-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="at">outlier.shape =</span> <span class="cn">NA</span>, </span>
<span id="cb105-17"><a href="#cb105-17" aria-hidden="true" tabindex="-1"></a>               <span class="at">width =</span> <span class="fl">0.12</span>, </span>
<span id="cb105-18"><a href="#cb105-18" aria-hidden="true" tabindex="-1"></a>               <span class="at">fill =</span> <span class="fu">c</span>(<span class="st">"salmon"</span>, <span class="st">"grey"</span>)) <span class="sc">+</span></span>
<span id="cb105-19"><a href="#cb105-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="figure_code_files/figure-html/unnamed-chunk-41-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb106"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb106-1"><a href="#cb106-1" aria-hidden="true" tabindex="-1"></a><span class="fu">wilcox.test</span>(sample_anno<span class="sc">$</span>log2_bap1 <span class="sc">~</span> sample_anno<span class="sc">$</span>bap1_activity_class)<span class="sc">$</span>p.val</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 8.979363e-51</code></pre>
</div>
</div>
</section>
<section id="figure-3e" class="level3">
<h3 class="anchored" data-anchor-id="figure-3e">Figure 3E</h3>
<p>This plot computes <em>BAP1</em> activity scores for human liver organoids with various mutation backgrounds including <em>BAP1</em> knockout (GSE215785) using the geneset derived from the TCGA analysis. Image was finalized in Adobe Illustrator 2021.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb108"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb108-1"><a href="#cb108-1" aria-hidden="true" tabindex="-1"></a>gse215785 <span class="ot">&lt;-</span> <span class="fu">read.delim</span>(<span class="st">"../data/gse215785/GSE215785_processed_raw_data_FLC.csv"</span>,</span>
<span id="cb108-2"><a href="#cb108-2" aria-hidden="true" tabindex="-1"></a>                        <span class="at">sep =</span> <span class="st">","</span>, <span class="at">row.names =</span> <span class="dv">1</span>)</span>
<span id="cb108-3"><a href="#cb108-3" aria-hidden="true" tabindex="-1"></a>gse215785_descriptions <span class="ot">&lt;-</span> <span class="fu">read.delim</span>(<span class="st">"../data/gse215785/sample_descriptions.txt"</span>)</span>
<span id="cb108-4"><a href="#cb108-4" aria-hidden="true" tabindex="-1"></a>gse215785[<span class="fu">is.na</span>(gse215785)] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb108-5"><a href="#cb108-5" aria-hidden="true" tabindex="-1"></a>gse215785_anno <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">sample_id =</span> gse215785_descriptions<span class="sc">$</span>sample_id,</span>
<span id="cb108-6"><a href="#cb108-6" aria-hidden="true" tabindex="-1"></a>                             <span class="at">sample_description =</span> gse215785_descriptions<span class="sc">$</span>sample_description,</span>
<span id="cb108-7"><a href="#cb108-7" aria-hidden="true" tabindex="-1"></a>                             <span class="at">row.names =</span> <span class="fu">colnames</span>(gse215785))</span>
<span id="cb108-8"><a href="#cb108-8" aria-hidden="true" tabindex="-1"></a>gse215785_anno<span class="sc">$</span>sample_type <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">rep</span>(<span class="st">"PRKAR2A"</span>, <span class="dv">3</span>), <span class="fu">rep</span>(<span class="st">"BAP1KO"</span>, <span class="dv">3</span>), </span>
<span id="cb108-9"><a href="#cb108-9" aria-hidden="true" tabindex="-1"></a>                                <span class="fu">rep</span>(<span class="st">"Fusion"</span>, <span class="dv">2</span>), <span class="fu">rep</span>(<span class="st">"DoubleKO"</span>, <span class="dv">3</span>), </span>
<span id="cb108-10"><a href="#cb108-10" aria-hidden="true" tabindex="-1"></a>                                <span class="fu">rep</span>(<span class="st">"DoubleKO_DUC"</span>, <span class="dv">3</span>), <span class="fu">rep</span>(<span class="st">"WT"</span>, <span class="dv">2</span>), </span>
<span id="cb108-11"><a href="#cb108-11" aria-hidden="true" tabindex="-1"></a>                                <span class="fu">rep</span>(<span class="st">"WT_DUC"</span>, <span class="dv">2</span>), <span class="fu">rep</span>(<span class="st">"BAP1KO"</span>, <span class="dv">2</span>), </span>
<span id="cb108-12"><a href="#cb108-12" aria-hidden="true" tabindex="-1"></a>                                <span class="fu">rep</span>(<span class="st">"Fusion"</span>, <span class="dv">3</span>), <span class="st">"PRKAR2A"</span>, </span>
<span id="cb108-13"><a href="#cb108-13" aria-hidden="true" tabindex="-1"></a>                                <span class="fu">rep</span>(<span class="st">"WT"</span>, <span class="dv">2</span>), <span class="fu">rep</span>(<span class="st">"WT_DUC"</span>, <span class="dv">2</span>), </span>
<span id="cb108-14"><a href="#cb108-14" aria-hidden="true" tabindex="-1"></a>                                <span class="fu">rep</span>(<span class="st">"DoubleKO"</span>, <span class="dv">2</span>), <span class="st">"DoubleKO_DUC"</span>)</span>
<span id="cb108-15"><a href="#cb108-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb108-16"><a href="#cb108-16" aria-hidden="true" tabindex="-1"></a>mut_up_gene_symbols <span class="ot">&lt;-</span> <span class="fu">c</span>()</span>
<span id="cb108-17"><a href="#cb108-17" aria-hidden="true" tabindex="-1"></a>mut_down_gene_symbols <span class="ot">&lt;-</span> <span class="fu">c</span>()</span>
<span id="cb108-18"><a href="#cb108-18" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> mut_up_genes) {</span>
<span id="cb108-19"><a href="#cb108-19" aria-hidden="true" tabindex="-1"></a>  mut_up_gene_symbols <span class="ot">&lt;-</span> <span class="fu">c</span>(mut_up_gene_symbols,</span>
<span id="cb108-20"><a href="#cb108-20" aria-hidden="true" tabindex="-1"></a>                           grch38<span class="sc">$</span>symbol[grch38<span class="sc">$</span>ensgene <span class="sc">==</span> i])</span>
<span id="cb108-21"><a href="#cb108-21" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb108-22"><a href="#cb108-22" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> mut_down_genes) {</span>
<span id="cb108-23"><a href="#cb108-23" aria-hidden="true" tabindex="-1"></a>  mut_down_gene_symbols <span class="ot">&lt;-</span> <span class="fu">c</span>(mut_down_gene_symbols,</span>
<span id="cb108-24"><a href="#cb108-24" aria-hidden="true" tabindex="-1"></a>                           grch38<span class="sc">$</span>symbol[grch38<span class="sc">$</span>ensgene <span class="sc">==</span> i])</span>
<span id="cb108-25"><a href="#cb108-25" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb108-26"><a href="#cb108-26" aria-hidden="true" tabindex="-1"></a>mut_up_gene_symbols <span class="ot">&lt;-</span> mut_up_gene_symbols[<span class="sc">!</span><span class="fu">is.na</span>(mut_up_gene_symbols) <span class="sc">&amp;</span></span>
<span id="cb108-27"><a href="#cb108-27" aria-hidden="true" tabindex="-1"></a>                                             mut_up_gene_symbols <span class="sc">!=</span> <span class="st">""</span>]</span>
<span id="cb108-28"><a href="#cb108-28" aria-hidden="true" tabindex="-1"></a>mut_up_gene_symbols <span class="ot">&lt;-</span> mut_up_gene_symbols[<span class="sc">!</span><span class="fu">duplicated</span>(mut_up_gene_symbols)]</span>
<span id="cb108-29"><a href="#cb108-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb108-30"><a href="#cb108-30" aria-hidden="true" tabindex="-1"></a>mut_down_gene_symbols <span class="ot">&lt;-</span> mut_down_gene_symbols[<span class="sc">!</span><span class="fu">is.na</span>(mut_down_gene_symbols) <span class="sc">&amp;</span></span>
<span id="cb108-31"><a href="#cb108-31" aria-hidden="true" tabindex="-1"></a>                                             mut_down_gene_symbols <span class="sc">!=</span> <span class="st">""</span>]</span>
<span id="cb108-32"><a href="#cb108-32" aria-hidden="true" tabindex="-1"></a>mut_down_gene_symbols <span class="ot">&lt;-</span> mut_down_gene_symbols[<span class="sc">!</span><span class="fu">duplicated</span>(mut_down_gene_symbols)]</span>
<span id="cb108-33"><a href="#cb108-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb108-34"><a href="#cb108-34" aria-hidden="true" tabindex="-1"></a>mut_up_gse215785 <span class="ot">&lt;-</span> mut_up_gene_symbols[mut_up_gene_symbols <span class="sc">%in%</span> </span>
<span id="cb108-35"><a href="#cb108-35" aria-hidden="true" tabindex="-1"></a>                                          <span class="fu">rownames</span>(gse215785)]</span>
<span id="cb108-36"><a href="#cb108-36" aria-hidden="true" tabindex="-1"></a>mut_down_gse215785 <span class="ot">&lt;-</span> mut_down_gene_symbols[mut_down_gene_symbols <span class="sc">%in%</span> </span>
<span id="cb108-37"><a href="#cb108-37" aria-hidden="true" tabindex="-1"></a>                                          <span class="fu">rownames</span>(gse215785)]</span>
<span id="cb108-38"><a href="#cb108-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb108-39"><a href="#cb108-39" aria-hidden="true" tabindex="-1"></a>dds_gse215785 <span class="ot">&lt;-</span> <span class="fu">suppressMessages</span>(<span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> gse215785,</span>
<span id="cb108-40"><a href="#cb108-40" aria-hidden="true" tabindex="-1"></a>                                        <span class="at">colData =</span> gse215785_anno,</span>
<span id="cb108-41"><a href="#cb108-41" aria-hidden="true" tabindex="-1"></a>                                        <span class="at">design =</span> <span class="sc">~</span> <span class="dv">1</span>))</span>
<span id="cb108-42"><a href="#cb108-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb108-43"><a href="#cb108-43" aria-hidden="true" tabindex="-1"></a>dds_gse215785 <span class="ot">&lt;-</span> <span class="fu">estimateSizeFactors</span>(dds_gse215785)</span>
<span id="cb108-44"><a href="#cb108-44" aria-hidden="true" tabindex="-1"></a>gse215785_counts <span class="ot">&lt;-</span> <span class="fu">counts</span>(dds_gse215785, <span class="at">normalized =</span> <span class="cn">TRUE</span>)</span>
<span id="cb108-45"><a href="#cb108-45" aria-hidden="true" tabindex="-1"></a>gse215785_counts_log2 <span class="ot">&lt;-</span> <span class="fu">log2</span>(gse215785_counts <span class="sc">+</span> <span class="dv">1</span>)</span>
<span id="cb108-46"><a href="#cb108-46" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb108-47"><a href="#cb108-47" aria-hidden="true" tabindex="-1"></a>score_up <span class="ot">&lt;-</span> <span class="fu">colSums</span>(gse215785_counts_log2[mut_up_gse215785, ], <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb108-48"><a href="#cb108-48" aria-hidden="true" tabindex="-1"></a>score_down <span class="ot">&lt;-</span> <span class="fu">colSums</span>(gse215785_counts_log2[mut_down_gse215785, ], <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb108-49"><a href="#cb108-49" aria-hidden="true" tabindex="-1"></a>act_scores <span class="ot">&lt;-</span> score_down <span class="sc">-</span> score_up</span>
<span id="cb108-50"><a href="#cb108-50" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb108-51"><a href="#cb108-51" aria-hidden="true" tabindex="-1"></a>gse215785_anno<span class="sc">$</span>activity_score <span class="ot">&lt;-</span> act_scores</span>
<span id="cb108-52"><a href="#cb108-52" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb108-53"><a href="#cb108-53" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove ductal media samples, which aren't relevant for this analysis</span></span>
<span id="cb108-54"><a href="#cb108-54" aria-hidden="true" tabindex="-1"></a>gse215785_anno <span class="ot">&lt;-</span> gse215785_anno[<span class="sc">!</span><span class="fu">grepl</span>(<span class="st">"DUC"</span>, gse215785_anno<span class="sc">$</span>sample_type),]</span>
<span id="cb108-55"><a href="#cb108-55" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb108-56"><a href="#cb108-56" aria-hidden="true" tabindex="-1"></a><span class="fu">boxplot</span>(gse215785_anno<span class="sc">$</span>activity_score <span class="sc">~</span> gse215785_anno<span class="sc">$</span>sample_type, </span>
<span id="cb108-57"><a href="#cb108-57" aria-hidden="true" tabindex="-1"></a>        <span class="at">outline =</span> <span class="cn">FALSE</span>, <span class="at">xlab =</span> <span class="st">"Sample type"</span>, <span class="at">ylab =</span> <span class="st">"BAP1 mutation score"</span>)</span>
<span id="cb108-58"><a href="#cb108-58" aria-hidden="true" tabindex="-1"></a><span class="fu">stripchart</span>(gse215785_anno<span class="sc">$</span>activity_score <span class="sc">~</span> gse215785_anno<span class="sc">$</span>sample_type,</span>
<span id="cb108-59"><a href="#cb108-59" aria-hidden="true" tabindex="-1"></a>           <span class="at">add =</span> <span class="cn">TRUE</span>, <span class="at">vertical =</span> <span class="cn">TRUE</span>, <span class="at">method =</span> <span class="st">"jitter"</span>, <span class="at">pch =</span> <span class="dv">16</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="figure_code_files/figure-html/unnamed-chunk-43-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<blockquote class="blockquote">
<p>Both single- and double-knockout organoids had significantly lower <em>BAP1</em> activity scores relative to wildtype (pairwise t-test with Bonferroni correction p=8.3e-11 and p=1.5e-12 respectively, Figure 3E), but no difference was observed in the samples with the classic <em>DNAJB1-PRKACA</em> fusion or <em>PRKACA</em> knockout which is the known major driver of that subset of liver cancer.</p>
</blockquote>
<div class="cell">
<div class="sourceCode cell-code" id="cb109"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb109-1"><a href="#cb109-1" aria-hidden="true" tabindex="-1"></a><span class="fu">pairwise.t.test</span>(gse215785_anno<span class="sc">$</span>activity_score, gse215785_anno<span class="sc">$</span>sample_type,</span>
<span id="cb109-2"><a href="#cb109-2" aria-hidden="true" tabindex="-1"></a>                <span class="at">p.adjust.method =</span> <span class="st">"bonferroni"</span>)<span class="sc">$</span>p.val</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>               BAP1KO     DoubleKO Fusion PRKAR2A
DoubleKO 3.591952e-03           NA     NA      NA
Fusion   8.138684e-11 1.146121e-12     NA      NA
PRKAR2A  4.389205e-10 5.525827e-12      1      NA
WT       8.343321e-11 1.456590e-12      1       1</code></pre>
</div>
</div>
</section>
<section id="figure-3f" class="level3">
<h3 class="anchored" data-anchor-id="figure-3f">Figure 3F</h3>
<p>This plot computes <em>BAP1</em> activity scores for human liver organoids including <em>BAP1</em> knockout (GSE129457) using the geneset derived from the TCGA analysis. Image was finalized in Adobe Illustrator 2021.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb111"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb111-1"><a href="#cb111-1" aria-hidden="true" tabindex="-1"></a>gse129457_dir <span class="ot">&lt;-</span> <span class="st">"../data/gse129457/"</span></span>
<span id="cb111-2"><a href="#cb111-2" aria-hidden="true" tabindex="-1"></a>gse129457_files <span class="ot">&lt;-</span> <span class="fu">dir</span>(gse129457_dir)</span>
<span id="cb111-3"><a href="#cb111-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb111-4"><a href="#cb111-4" aria-hidden="true" tabindex="-1"></a>gse129457_anno <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">sample =</span> <span class="fu">gsub</span>(<span class="st">"_coutb.csv.gz"</span>, <span class="st">""</span>,</span>
<span id="cb111-5"><a href="#cb111-5" aria-hidden="true" tabindex="-1"></a>                                           gse129457_files))</span>
<span id="cb111-6"><a href="#cb111-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb111-7"><a href="#cb111-7" aria-hidden="true" tabindex="-1"></a>gse129457_anno<span class="sc">$</span>sample_type <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">rep</span>(<span class="st">"WT"</span>, <span class="dv">3</span>), <span class="fu">rep</span>(<span class="st">"BAP1mut"</span>, <span class="dv">3</span>), <span class="fu">rep</span>(<span class="st">"BAP1mut+24hrDox"</span>, <span class="dv">2</span>))</span>
<span id="cb111-8"><a href="#cb111-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb111-9"><a href="#cb111-9" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(gse129457_anno) <span class="ot">&lt;-</span> gse129457_anno<span class="sc">$</span>sample</span>
<span id="cb111-10"><a href="#cb111-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb111-11"><a href="#cb111-11" aria-hidden="true" tabindex="-1"></a>read_counts <span class="ot">&lt;-</span> <span class="cf">function</span>(counts_file) {</span>
<span id="cb111-12"><a href="#cb111-12" aria-hidden="true" tabindex="-1"></a>  counts_file <span class="ot">&lt;-</span> <span class="fu">file.path</span>(gse129457_dir, counts_file)</span>
<span id="cb111-13"><a href="#cb111-13" aria-hidden="true" tabindex="-1"></a>  sample_counts <span class="ot">&lt;-</span> <span class="fu">read.delim</span>(counts_file, <span class="at">sep =</span> <span class="st">";"</span>)</span>
<span id="cb111-14"><a href="#cb111-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(sample_counts)</span>
<span id="cb111-15"><a href="#cb111-15" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb111-16"><a href="#cb111-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb111-17"><a href="#cb111-17" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (counts_file <span class="cf">in</span> gse129457_files) {</span>
<span id="cb111-18"><a href="#cb111-18" aria-hidden="true" tabindex="-1"></a>  temp_counts <span class="ot">&lt;-</span> <span class="fu">read_counts</span>(counts_file)</span>
<span id="cb111-19"><a href="#cb111-19" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">exists</span>(<span class="st">"gse129457_genes"</span>)) {</span>
<span id="cb111-20"><a href="#cb111-20" aria-hidden="true" tabindex="-1"></a>    gse129457_genes <span class="ot">&lt;-</span> <span class="fu">intersect</span>(gse129457_genes, temp_counts<span class="sc">$</span>GENEID)</span>
<span id="cb111-21"><a href="#cb111-21" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb111-22"><a href="#cb111-22" aria-hidden="true" tabindex="-1"></a>  <span class="cf">else</span> {</span>
<span id="cb111-23"><a href="#cb111-23" aria-hidden="true" tabindex="-1"></a>    gse129457_genes <span class="ot">&lt;-</span> temp_counts<span class="sc">$</span>GENEID</span>
<span id="cb111-24"><a href="#cb111-24" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb111-25"><a href="#cb111-25" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb111-26"><a href="#cb111-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb111-27"><a href="#cb111-27" aria-hidden="true" tabindex="-1"></a>gse129457_genes <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"</span><span class="sc">\\</span><span class="st">_.*"</span>, <span class="st">""</span>, gse129457_genes)</span>
<span id="cb111-28"><a href="#cb111-28" aria-hidden="true" tabindex="-1"></a>mut_up_gse129457 <span class="ot">&lt;-</span> mut_up_gene_symbols[mut_up_gene_symbols <span class="sc">%in%</span> </span>
<span id="cb111-29"><a href="#cb111-29" aria-hidden="true" tabindex="-1"></a>                                          gse129457_genes]</span>
<span id="cb111-30"><a href="#cb111-30" aria-hidden="true" tabindex="-1"></a>mut_down_gse129457 <span class="ot">&lt;-</span> mut_down_gene_symbols[mut_down_gene_symbols <span class="sc">%in%</span> </span>
<span id="cb111-31"><a href="#cb111-31" aria-hidden="true" tabindex="-1"></a>                                          gse129457_genes]</span>
<span id="cb111-32"><a href="#cb111-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb111-33"><a href="#cb111-33" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (counts_file <span class="cf">in</span> gse129457_files) {</span>
<span id="cb111-34"><a href="#cb111-34" aria-hidden="true" tabindex="-1"></a>  temp_counts <span class="ot">&lt;-</span> <span class="fu">read_counts</span>(counts_file)</span>
<span id="cb111-35"><a href="#cb111-35" aria-hidden="true" tabindex="-1"></a>  temp_counts<span class="sc">$</span>GENEID <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"</span><span class="sc">\\</span><span class="st">_.*"</span>, <span class="st">""</span>, temp_counts<span class="sc">$</span>GENEID)</span>
<span id="cb111-36"><a href="#cb111-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rownames</span>(temp_counts) <span class="ot">&lt;-</span> temp_counts<span class="sc">$</span>GENEID</span>
<span id="cb111-37"><a href="#cb111-37" aria-hidden="true" tabindex="-1"></a>  temp_counts <span class="ot">&lt;-</span> temp_counts[<span class="fu">c</span>(mut_up_gse129457, mut_down_gse129457), ]</span>
<span id="cb111-38"><a href="#cb111-38" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">exists</span>(<span class="st">"gse129457"</span>)) {</span>
<span id="cb111-39"><a href="#cb111-39" aria-hidden="true" tabindex="-1"></a>    gse129457 <span class="ot">&lt;-</span> <span class="fu">cbind</span>(gse129457, temp_counts[, <span class="dv">2</span>])</span>
<span id="cb111-40"><a href="#cb111-40" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb111-41"><a href="#cb111-41" aria-hidden="true" tabindex="-1"></a>  <span class="cf">else</span> {</span>
<span id="cb111-42"><a href="#cb111-42" aria-hidden="true" tabindex="-1"></a>    gse129457 <span class="ot">&lt;-</span> temp_counts</span>
<span id="cb111-43"><a href="#cb111-43" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb111-44"><a href="#cb111-44" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb111-45"><a href="#cb111-45" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb111-46"><a href="#cb111-46" aria-hidden="true" tabindex="-1"></a>gse129457 <span class="ot">&lt;-</span> gse129457[, <span class="dv">2</span><span class="sc">:</span><span class="fu">ncol</span>(gse129457)]</span>
<span id="cb111-47"><a href="#cb111-47" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(gse129457) <span class="ot">&lt;-</span> gse129457_anno<span class="sc">$</span>sample</span>
<span id="cb111-48"><a href="#cb111-48" aria-hidden="true" tabindex="-1"></a>dds_gse129457 <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> gse129457,</span>
<span id="cb111-49"><a href="#cb111-49" aria-hidden="true" tabindex="-1"></a>                                        <span class="at">colData =</span> gse129457_anno,</span>
<span id="cb111-50"><a href="#cb111-50" aria-hidden="true" tabindex="-1"></a>                                        <span class="at">design =</span> <span class="sc">~</span> <span class="dv">1</span>)</span>
<span id="cb111-51"><a href="#cb111-51" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb111-52"><a href="#cb111-52" aria-hidden="true" tabindex="-1"></a>dds_gse129457 <span class="ot">&lt;-</span> <span class="fu">estimateSizeFactors</span>(dds_gse129457)</span>
<span id="cb111-53"><a href="#cb111-53" aria-hidden="true" tabindex="-1"></a>gse129457_counts <span class="ot">&lt;-</span> <span class="fu">counts</span>(dds_gse129457, <span class="at">normalized =</span> <span class="cn">TRUE</span>)</span>
<span id="cb111-54"><a href="#cb111-54" aria-hidden="true" tabindex="-1"></a>gse129457_counts_log2 <span class="ot">&lt;-</span> <span class="fu">log2</span>(gse129457_counts <span class="sc">+</span> <span class="dv">1</span>)</span>
<span id="cb111-55"><a href="#cb111-55" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb111-56"><a href="#cb111-56" aria-hidden="true" tabindex="-1"></a>score_up <span class="ot">&lt;-</span> <span class="fu">colSums</span>(gse129457_counts_log2[mut_up_gse129457, ], <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb111-57"><a href="#cb111-57" aria-hidden="true" tabindex="-1"></a>score_down <span class="ot">&lt;-</span> <span class="fu">colSums</span>(gse129457_counts_log2[mut_down_gse129457, ], <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb111-58"><a href="#cb111-58" aria-hidden="true" tabindex="-1"></a>act_scores <span class="ot">&lt;-</span> score_down <span class="sc">-</span> score_up</span>
<span id="cb111-59"><a href="#cb111-59" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb111-60"><a href="#cb111-60" aria-hidden="true" tabindex="-1"></a>gse129457_anno<span class="sc">$</span>activity_score <span class="ot">&lt;-</span> act_scores</span>
<span id="cb111-61"><a href="#cb111-61" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb111-62"><a href="#cb111-62" aria-hidden="true" tabindex="-1"></a><span class="fu">boxplot</span>(gse129457_anno<span class="sc">$</span>activity_score <span class="sc">~</span> gse129457_anno<span class="sc">$</span>sample_type, </span>
<span id="cb111-63"><a href="#cb111-63" aria-hidden="true" tabindex="-1"></a>        <span class="at">outline =</span> <span class="cn">FALSE</span>, <span class="at">xlab =</span> <span class="st">"Sample type"</span>, <span class="at">ylab =</span> <span class="st">"BAP1 mutation score"</span>)</span>
<span id="cb111-64"><a href="#cb111-64" aria-hidden="true" tabindex="-1"></a><span class="fu">stripchart</span>(gse129457_anno<span class="sc">$</span>activity_score <span class="sc">~</span> gse129457_anno<span class="sc">$</span>sample_type,</span>
<span id="cb111-65"><a href="#cb111-65" aria-hidden="true" tabindex="-1"></a>           <span class="at">add =</span> <span class="cn">TRUE</span>, <span class="at">vertical =</span> <span class="cn">TRUE</span>, <span class="at">method =</span> <span class="st">"jitter"</span>, <span class="at">pch =</span> <span class="dv">16</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="figure_code_files/figure-html/unnamed-chunk-45-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<blockquote class="blockquote">
<p>Consistent with GSE215785 data in Figure 3E, we also observed reduced <em>BAP1</em> activity scores among <em>BAP1</em> knockout organoids relative to wildtype (pairwise t-test p=0.015, Figure 3F).</p>
</blockquote>
<div class="cell">
<div class="sourceCode cell-code" id="cb112"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb112-1"><a href="#cb112-1" aria-hidden="true" tabindex="-1"></a><span class="fu">pairwise.t.test</span>(gse129457_anno<span class="sc">$</span>activity_score, gse129457_anno<span class="sc">$</span>sample_type,</span>
<span id="cb112-2"><a href="#cb112-2" aria-hidden="true" tabindex="-1"></a>                <span class="at">p.adjust.method =</span> <span class="st">"bonferroni"</span>)<span class="sc">$</span>p.val</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                  BAP1mut BAP1mut+24hrDox
BAP1mut+24hrDox 0.5465809              NA
WT              0.0149440       0.1246203</code></pre>
</div>
</div>
</section>
</section>
<section id="figure-4---bap1-mutant-like-lihc-tumor-analysis" class="level2">
<h2 class="anchored" data-anchor-id="figure-4---bap1-mutant-like-lihc-tumor-analysis">Figure 4 - <em>BAP1</em> mutant-like LIHC tumor analysis</h2>
<blockquote class="blockquote">
<p>Consistent with these previous data, 17/31 (54.8%) of <em>BAP1</em>-altered tumors were one ofthese two molecular subtypes; among the unaltered tumors, 77/327 (23.5%) were one of these subtypes. Blast-like and CHOL-like tumors have <em>BAP1</em> activity scores lower than the other LIHC tumors (pairwise two-sided Mann-Whitney U test with continuity correction and Bonferroni adjusted p=2.47e-3 and p=2.38e-9 respectively, Figures 4A-B and Supplemental Table S3B).</p>
</blockquote>
<div class="cell">
<div class="sourceCode cell-code" id="cb114"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb114-1"><a href="#cb114-1" aria-hidden="true" tabindex="-1"></a>lihc_anno <span class="ot">&lt;-</span> sample_anno[sample_anno<span class="sc">$</span>cohort <span class="sc">==</span> <span class="st">"TCGA-LIHC"</span> <span class="sc">&amp;</span></span>
<span id="cb114-2"><a href="#cb114-2" aria-hidden="true" tabindex="-1"></a>                               <span class="sc">!</span><span class="fu">is.na</span>(sample_anno<span class="sc">$</span>log2_bap1) <span class="sc">&amp;</span></span>
<span id="cb114-3"><a href="#cb114-3" aria-hidden="true" tabindex="-1"></a>                               <span class="sc">!</span><span class="fu">is.na</span>(sample_anno<span class="sc">$</span>purity) <span class="sc">&amp;</span></span>
<span id="cb114-4"><a href="#cb114-4" aria-hidden="true" tabindex="-1"></a>                               <span class="sc">!</span><span class="fu">is.na</span>(sample_anno<span class="sc">$</span>alteration_type) <span class="sc">&amp;</span></span>
<span id="cb114-5"><a href="#cb114-5" aria-hidden="true" tabindex="-1"></a>                               <span class="sc">!</span><span class="fu">grepl</span>(<span class="st">"NA"</span>, sample_anno<span class="sc">$</span>subtype), ]</span>
<span id="cb114-6"><a href="#cb114-6" aria-hidden="true" tabindex="-1"></a>lihc_anno <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(lihc_anno)</span>
<span id="cb114-7"><a href="#cb114-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb114-8"><a href="#cb114-8" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(lihc_anno<span class="sc">$</span>altered_status, lihc_anno<span class="sc">$</span>subtype)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>           
            LIHC_Blast.Like LIHC_CHOL.Like LIHC_Liver.Like
  altered                 6             11              14
  unaltered              57             20             250</code></pre>
</div>
</div>
<section id="figure-4a" class="level3">
<h3 class="anchored" data-anchor-id="figure-4a">Figure 4A</h3>
<p>This is a heatmap of differentially expressed genes between mutant-like and wildtype-like tumors in LIHC. Image was finalized in Adobe Illustrator 2021.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb116"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb116-1"><a href="#cb116-1" aria-hidden="true" tabindex="-1"></a>lihc_dds <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"../results/activity_signature/de_dds/lihc_mutlike_vs_nonmutlike.rds"</span>)</span>
<span id="cb116-2"><a href="#cb116-2" aria-hidden="true" tabindex="-1"></a>lihc_resLFC <span class="ot">&lt;-</span> <span class="fu">read.delim</span>(<span class="st">"../results/activity_signature/de_results/lihc_de_genes_shrink_mut_class_MutlikevsWTlike.txt"</span>)</span>
<span id="cb116-3"><a href="#cb116-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb116-4"><a href="#cb116-4" aria-hidden="true" tabindex="-1"></a>col_fun <span class="ot">&lt;-</span> <span class="fu">colorRamp2</span>(<span class="fu">c</span>(<span class="fu">min</span>(lihc_anno<span class="sc">$</span>bap1_activity_score), </span>
<span id="cb116-5"><a href="#cb116-5" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">mean</span>(<span class="fu">max</span>(lihc_anno<span class="sc">$</span>bap1_activity_score[lihc_anno<span class="sc">$</span>bap1_activity_class <span class="sc">==</span> <span class="st">"Mutant-like"</span>]), </span>
<span id="cb116-6"><a href="#cb116-6" aria-hidden="true" tabindex="-1"></a>                             <span class="fu">min</span>(lihc_anno<span class="sc">$</span>bap1_activity_score[lihc_anno<span class="sc">$</span>bap1_activity_class <span class="sc">==</span> <span class="st">"Wildtype-like"</span>])), </span>
<span id="cb116-7"><a href="#cb116-7" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">max</span>(lihc_anno<span class="sc">$</span>bap1_activity_score)), </span>
<span id="cb116-8"><a href="#cb116-8" aria-hidden="true" tabindex="-1"></a>                      <span class="fu">c</span>(<span class="st">"red"</span>, <span class="st">"white"</span>, <span class="st">"blue"</span>))</span>
<span id="cb116-9"><a href="#cb116-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb116-10"><a href="#cb116-10" aria-hidden="true" tabindex="-1"></a>col_fun2 <span class="ot">&lt;-</span> <span class="fu">colorRamp2</span>(<span class="fu">c</span>(<span class="sc">-</span><span class="dv">2</span>, <span class="dv">0</span>, <span class="dv">2</span>), <span class="fu">c</span>(<span class="st">"blue"</span>, <span class="st">"black"</span>, <span class="st">"yellow"</span>))</span>
<span id="cb116-11"><a href="#cb116-11" aria-hidden="true" tabindex="-1"></a>col_fun3 <span class="ot">&lt;-</span> <span class="fu">colorRamp2</span>(<span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>, <span class="dv">0</span>, <span class="dv">1</span>), <span class="fu">c</span>(<span class="st">"lightblue"</span>, <span class="st">"black"</span>, <span class="st">"yellow"</span>))</span>
<span id="cb116-12"><a href="#cb116-12" aria-hidden="true" tabindex="-1"></a>pal_purity <span class="ot">&lt;-</span> <span class="fu">colorRamp2</span>(<span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">1</span>), <span class="fu">c</span>(<span class="st">"white"</span>, <span class="st">"steelblue"</span>))</span>
<span id="cb116-13"><a href="#cb116-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb116-14"><a href="#cb116-14" aria-hidden="true" tabindex="-1"></a>lihc_counts <span class="ot">&lt;-</span> <span class="fu">counts</span>(lihc_dds, <span class="at">normalized =</span> <span class="cn">TRUE</span>)</span>
<span id="cb116-15"><a href="#cb116-15" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb116-16"><a href="#cb116-16" aria-hidden="true" tabindex="-1"></a>lihc_counts <span class="ot">&lt;-</span> lihc_counts[, lihc_anno<span class="sc">$</span>tcga_id]</span>
<span id="cb116-17"><a href="#cb116-17" aria-hidden="true" tabindex="-1"></a>lihc_counts <span class="ot">&lt;-</span> lihc_counts[<span class="fu">rownames</span>(lihc_counts) <span class="sc">%in%</span> </span>
<span id="cb116-18"><a href="#cb116-18" aria-hidden="true" tabindex="-1"></a>                           <span class="fu">rownames</span>(lihc_resLFC)[lihc_resLFC<span class="sc">$</span>padj <span class="sc">&lt;</span> <span class="fl">0.05</span>], ]</span>
<span id="cb116-19"><a href="#cb116-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb116-20"><a href="#cb116-20" aria-hidden="true" tabindex="-1"></a>lihc_counts_log2mc <span class="ot">&lt;-</span> <span class="fu">log2_mediancenter</span>(lihc_counts)</span>
<span id="cb116-21"><a href="#cb116-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb116-22"><a href="#cb116-22" aria-hidden="true" tabindex="-1"></a>gene_anno <span class="ot">&lt;-</span> <span class="fu">c</span>()</span>
<span id="cb116-23"><a href="#cb116-23" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (gene <span class="cf">in</span> <span class="fu">rownames</span>(lihc_counts_log2mc)) {</span>
<span id="cb116-24"><a href="#cb116-24" aria-hidden="true" tabindex="-1"></a>  gene_anno <span class="ot">&lt;-</span> <span class="fu">c</span>(gene_anno,</span>
<span id="cb116-25"><a href="#cb116-25" aria-hidden="true" tabindex="-1"></a>                 lihc_resLFC<span class="sc">$</span>log2FoldChange[<span class="fu">rownames</span>(lihc_resLFC) <span class="sc">==</span> gene])</span>
<span id="cb116-26"><a href="#cb116-26" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb116-27"><a href="#cb116-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb116-28"><a href="#cb116-28" aria-hidden="true" tabindex="-1"></a>lihc_anno <span class="ot">&lt;-</span> lihc_anno[<span class="fu">order</span>(lihc_anno<span class="sc">$</span>bap1_activity_score, <span class="at">decreasing =</span> <span class="cn">FALSE</span>), ]</span>
<span id="cb116-29"><a href="#cb116-29" aria-hidden="true" tabindex="-1"></a>lihc_counts_log2mc <span class="ot">&lt;-</span> lihc_counts_log2mc[, lihc_anno<span class="sc">$</span>tcga_id]</span>
<span id="cb116-30"><a href="#cb116-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb116-31"><a href="#cb116-31" aria-hidden="true" tabindex="-1"></a>ha_row <span class="ot">&lt;-</span> <span class="fu">rowAnnotation</span>(<span class="at">log2FC =</span> <span class="fu">sort</span>(gene_anno),</span>
<span id="cb116-32"><a href="#cb116-32" aria-hidden="true" tabindex="-1"></a>                        <span class="at">col =</span> <span class="fu">list</span>(<span class="at">log2FC =</span> col_fun3))</span>
<span id="cb116-33"><a href="#cb116-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb116-34"><a href="#cb116-34" aria-hidden="true" tabindex="-1"></a>ha <span class="ot">&lt;-</span> <span class="fu">HeatmapAnnotation</span>(<span class="at">subtype =</span> lihc_anno<span class="sc">$</span>subtype,</span>
<span id="cb116-35"><a href="#cb116-35" aria-hidden="true" tabindex="-1"></a>                        <span class="at">purity =</span> lihc_anno<span class="sc">$</span>purity,</span>
<span id="cb116-36"><a href="#cb116-36" aria-hidden="true" tabindex="-1"></a>                        <span class="at">alteration_type =</span> lihc_anno<span class="sc">$</span>alteration_type,</span>
<span id="cb116-37"><a href="#cb116-37" aria-hidden="true" tabindex="-1"></a>                        <span class="at">col =</span> <span class="fu">list</span>(<span class="at">subtype =</span> <span class="fu">c</span>(<span class="st">"LIHC_Liver.Like"</span> <span class="ot">=</span> <span class="st">"darkgrey"</span>,</span>
<span id="cb116-38"><a href="#cb116-38" aria-hidden="true" tabindex="-1"></a>                                               <span class="st">"LIHC_Blast.Like"</span> <span class="ot">=</span> <span class="st">"salmon"</span>,</span>
<span id="cb116-39"><a href="#cb116-39" aria-hidden="true" tabindex="-1"></a>                                               <span class="st">"LIHC_CHOL.Like"</span> <span class="ot">=</span> <span class="st">"darkgreen"</span>),</span>
<span id="cb116-40"><a href="#cb116-40" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">purity =</span> pal_purity,</span>
<span id="cb116-41"><a href="#cb116-41" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">alteration_type =</span> <span class="fu">c</span>(<span class="st">"copynumber"</span> <span class="ot">=</span> <span class="st">"lightgreen"</span>,</span>
<span id="cb116-42"><a href="#cb116-42" aria-hidden="true" tabindex="-1"></a>                                                         <span class="st">"mutation"</span> <span class="ot">=</span> <span class="st">"salmon"</span>,</span>
<span id="cb116-43"><a href="#cb116-43" aria-hidden="true" tabindex="-1"></a>                                                         <span class="st">"mutation+copynumber"</span> <span class="ot">=</span> <span class="st">"darkgreen"</span>, <span class="st">"unaltered"</span> <span class="ot">=</span> <span class="st">"grey"</span>)))</span>
<span id="cb116-44"><a href="#cb116-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb116-45"><a href="#cb116-45" aria-hidden="true" tabindex="-1"></a>ha_b <span class="ot">&lt;-</span> <span class="fu">HeatmapAnnotation</span>(<span class="at">mut_sig =</span> lihc_anno<span class="sc">$</span>bap1_activity_score, </span>
<span id="cb116-46"><a href="#cb116-46" aria-hidden="true" tabindex="-1"></a>                          <span class="at">mut_class =</span> lihc_anno<span class="sc">$</span>bap1_activity_class,</span>
<span id="cb116-47"><a href="#cb116-47" aria-hidden="true" tabindex="-1"></a>                          <span class="at">col =</span> <span class="fu">list</span>(<span class="at">mut_sig =</span> col_fun,</span>
<span id="cb116-48"><a href="#cb116-48" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">mut_class =</span> <span class="fu">c</span>(<span class="st">"Mutant-like"</span> <span class="ot">=</span> <span class="st">"salmon"</span>, </span>
<span id="cb116-49"><a href="#cb116-49" aria-hidden="true" tabindex="-1"></a>                                                   <span class="st">"Wildtype-like"</span> <span class="ot">=</span> <span class="st">"grey"</span>)))</span>
<span id="cb116-50"><a href="#cb116-50" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb116-51"><a href="#cb116-51" aria-hidden="true" tabindex="-1"></a>ht <span class="ot">&lt;-</span> <span class="fu">Heatmap</span>(<span class="fu">as.matrix</span>(lihc_counts_log2mc[<span class="fu">order</span>(gene_anno), ]),</span>
<span id="cb116-52"><a href="#cb116-52" aria-hidden="true" tabindex="-1"></a>              <span class="at">top_annotation =</span> ha, <span class="at">bottom_annotation =</span> ha_b,</span>
<span id="cb116-53"><a href="#cb116-53" aria-hidden="true" tabindex="-1"></a>              <span class="at">right_annotation =</span> ha_row,</span>
<span id="cb116-54"><a href="#cb116-54" aria-hidden="true" tabindex="-1"></a>              <span class="at">show_column_names =</span> <span class="cn">FALSE</span>, <span class="at">show_row_names =</span> <span class="cn">FALSE</span>,</span>
<span id="cb116-55"><a href="#cb116-55" aria-hidden="true" tabindex="-1"></a>              <span class="at">cluster_rows =</span> <span class="cn">FALSE</span>, <span class="at">cluster_columns =</span> <span class="cn">FALSE</span>,</span>
<span id="cb116-56"><a href="#cb116-56" aria-hidden="true" tabindex="-1"></a>              <span class="at">col =</span> col_fun2, <span class="at">name =</span> <span class="st">"Expression"</span>)</span>
<span id="cb116-57"><a href="#cb116-57" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb116-58"><a href="#cb116-58" aria-hidden="true" tabindex="-1"></a><span class="fu">draw</span>(ht)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="figure_code_files/figure-html/unnamed-chunk-48-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="figure-4b" class="level3">
<h3 class="anchored" data-anchor-id="figure-4b">Figure 4B</h3>
<p>This is a boxplot of <em>BAP1</em> activity scores, compared across true CHOL tumors and LIHC tumor subtypes. Image was finalized in Adobe Illustrator 2021.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb117"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb117-1"><a href="#cb117-1" aria-hidden="true" tabindex="-1"></a>chol_dds <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"../results/activity_signature/de_dds/chol_mutlike_vs_nonmutlike.rds"</span>)</span>
<span id="cb117-2"><a href="#cb117-2" aria-hidden="true" tabindex="-1"></a>chol_coldata <span class="ot">&lt;-</span> <span class="fu">colData</span>(chol_dds)</span>
<span id="cb117-3"><a href="#cb117-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb117-4"><a href="#cb117-4" aria-hidden="true" tabindex="-1"></a>lihc_coldata <span class="ot">&lt;-</span> <span class="fu">colData</span>(lihc_dds)</span>
<span id="cb117-5"><a href="#cb117-5" aria-hidden="true" tabindex="-1"></a>lihc_coldata <span class="ot">&lt;-</span> lihc_coldata[, <span class="fu">intersect</span>(<span class="fu">colnames</span>(lihc_coldata), <span class="fu">colnames</span>(chol_coldata))]</span>
<span id="cb117-6"><a href="#cb117-6" aria-hidden="true" tabindex="-1"></a>chol_coldata <span class="ot">&lt;-</span> chol_coldata[, <span class="fu">intersect</span>(<span class="fu">colnames</span>(lihc_coldata), <span class="fu">colnames</span>(chol_coldata))]</span>
<span id="cb117-7"><a href="#cb117-7" aria-hidden="true" tabindex="-1"></a>combined_coldata <span class="ot">&lt;-</span> <span class="fu">rbind</span>(chol_coldata, lihc_coldata)</span>
<span id="cb117-8"><a href="#cb117-8" aria-hidden="true" tabindex="-1"></a>combined_coldata<span class="sc">$</span>subtype <span class="ot">&lt;-</span> <span class="fu">as.character</span>(combined_coldata<span class="sc">$</span>subtype)</span>
<span id="cb117-9"><a href="#cb117-9" aria-hidden="true" tabindex="-1"></a>combined_coldata<span class="sc">$</span>subtype[combined_coldata<span class="sc">$</span>cohort <span class="sc">==</span> <span class="st">"TCGA-CHOL"</span>] <span class="ot">&lt;-</span> <span class="st">"CHOL"</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb118"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb118-1"><a href="#cb118-1" aria-hidden="true" tabindex="-1"></a><span class="fu">boxplot</span>(combined_coldata<span class="sc">$</span>bap1_activity_score <span class="sc">~</span> <span class="fu">factor</span>(combined_coldata<span class="sc">$</span>subtype,</span>
<span id="cb118-2"><a href="#cb118-2" aria-hidden="true" tabindex="-1"></a>                                             <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"CHOL"</span>,</span>
<span id="cb118-3"><a href="#cb118-3" aria-hidden="true" tabindex="-1"></a>                                                        <span class="st">"LIHC_CHOL.Like"</span>,</span>
<span id="cb118-4"><a href="#cb118-4" aria-hidden="true" tabindex="-1"></a>                                                        <span class="st">"LIHC_Blast.Like"</span>,</span>
<span id="cb118-5"><a href="#cb118-5" aria-hidden="true" tabindex="-1"></a>                                                        <span class="st">"LIHC_Liver.Like"</span>)),</span>
<span id="cb118-6"><a href="#cb118-6" aria-hidden="true" tabindex="-1"></a>        <span class="at">ylab =</span> <span class="st">"BAP1 activity score"</span>,</span>
<span id="cb118-7"><a href="#cb118-7" aria-hidden="true" tabindex="-1"></a>        <span class="at">xlab =</span> <span class="cn">NULL</span>,</span>
<span id="cb118-8"><a href="#cb118-8" aria-hidden="true" tabindex="-1"></a>        <span class="at">col =</span> <span class="fu">c</span>(<span class="st">"blue"</span>, <span class="st">"darkgreen"</span>, <span class="st">"coral"</span>, <span class="st">"gray29"</span>),</span>
<span id="cb118-9"><a href="#cb118-9" aria-hidden="true" tabindex="-1"></a>        <span class="at">outline =</span> <span class="cn">FALSE</span>)</span>
<span id="cb118-10"><a href="#cb118-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb118-11"><a href="#cb118-11" aria-hidden="true" tabindex="-1"></a><span class="fu">stripchart</span>(combined_coldata<span class="sc">$</span>bap1_activity_score <span class="sc">~</span> <span class="fu">factor</span>(combined_coldata<span class="sc">$</span>subtype,</span>
<span id="cb118-12"><a href="#cb118-12" aria-hidden="true" tabindex="-1"></a>                                             <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"CHOL"</span>,</span>
<span id="cb118-13"><a href="#cb118-13" aria-hidden="true" tabindex="-1"></a>                                                        <span class="st">"LIHC_CHOL.Like"</span>,</span>
<span id="cb118-14"><a href="#cb118-14" aria-hidden="true" tabindex="-1"></a>                                                        <span class="st">"LIHC_Blast.Like"</span>,</span>
<span id="cb118-15"><a href="#cb118-15" aria-hidden="true" tabindex="-1"></a>                                                        <span class="st">"LIHC_Liver.Like"</span>)),</span>
<span id="cb118-16"><a href="#cb118-16" aria-hidden="true" tabindex="-1"></a>           <span class="at">vertical =</span> <span class="cn">TRUE</span>,</span>
<span id="cb118-17"><a href="#cb118-17" aria-hidden="true" tabindex="-1"></a>           <span class="at">add =</span> <span class="cn">TRUE</span>,</span>
<span id="cb118-18"><a href="#cb118-18" aria-hidden="true" tabindex="-1"></a>           <span class="at">method =</span> <span class="st">"jitter"</span>,</span>
<span id="cb118-19"><a href="#cb118-19" aria-hidden="true" tabindex="-1"></a>           <span class="at">pch =</span> <span class="dv">16</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="figure_code_files/figure-html/unnamed-chunk-50-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb119"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb119-1"><a href="#cb119-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Pairwise two-sided Mann-Whitney U test with continuity correction and Bonferroni adjustment</span></span>
<span id="cb119-2"><a href="#cb119-2" aria-hidden="true" tabindex="-1"></a><span class="fu">pairwise.wilcox.test</span>(combined_coldata<span class="sc">$</span>bap1_activity_score,</span>
<span id="cb119-3"><a href="#cb119-3" aria-hidden="true" tabindex="-1"></a>                     combined_coldata<span class="sc">$</span>subtype,</span>
<span id="cb119-4"><a href="#cb119-4" aria-hidden="true" tabindex="-1"></a>                     <span class="at">p.adjust.method =</span> <span class="st">"bonferroni"</span>,</span>
<span id="cb119-5"><a href="#cb119-5" aria-hidden="true" tabindex="-1"></a>                     <span class="at">alternative =</span> <span class="st">"two.sided"</span>,</span>
<span id="cb119-6"><a href="#cb119-6" aria-hidden="true" tabindex="-1"></a>                     <span class="at">correct =</span> <span class="cn">TRUE</span>)<span class="sc">$</span>p.value</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                        CHOL LIHC_Blast.Like LIHC_CHOL.Like
LIHC_Blast.Like 2.089848e-12              NA             NA
LIHC_CHOL.Like  7.347360e-02    0.0001213492             NA
LIHC_Liver.Like 8.316584e-17    0.0024737122   2.381085e-09</code></pre>
</div>
</div>
<blockquote class="blockquote">
<p>We used true CHOL samples as a comparison and observed that <em>BAP1</em> mutant-like LIHC tumors were enriched for bile duct markers relative to wildtype-like samples (pairwise two-sided Mann-Whitney U test with continuity correction and Bonferroni adjusted p=1.67e-4, Figure 4C).</p>
</blockquote>
</section>
<section id="figure-4c" class="level3">
<h3 class="anchored" data-anchor-id="figure-4c">Figure 4C</h3>
<p>This is a boxplot of MSigDB C8-derived bileduct signature scores for true CHOL tumors and LIHC <em>BAP1</em> activity tumor subtypes. Image was finalized in Adobe Illustrator 2021.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb121"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb121-1"><a href="#cb121-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(msigdbr)</span>
<span id="cb121-2"><a href="#cb121-2" aria-hidden="true" tabindex="-1"></a>c8 <span class="ot">&lt;-</span> <span class="fu">msigdbr</span>(<span class="at">species =</span> <span class="st">"human"</span>, <span class="at">category =</span> <span class="st">"C8"</span>)</span>
<span id="cb121-3"><a href="#cb121-3" aria-hidden="true" tabindex="-1"></a>bileduct_1 <span class="ot">&lt;-</span> <span class="fu">unique</span>(c8<span class="sc">$</span>ensembl_gene[<span class="fu">grepl</span>(<span class="st">"BILE_DUCT_CELLS_1"</span>, c8<span class="sc">$</span>gs_name)])</span>
<span id="cb121-4"><a href="#cb121-4" aria-hidden="true" tabindex="-1"></a>bileduct_2 <span class="ot">&lt;-</span> <span class="fu">unique</span>(c8<span class="sc">$</span>ensembl_gene[<span class="fu">grepl</span>(<span class="st">"BILE_DUCT_CELLS_2"</span>, c8<span class="sc">$</span>gs_name)])</span>
<span id="cb121-5"><a href="#cb121-5" aria-hidden="true" tabindex="-1"></a>bileduct_3 <span class="ot">&lt;-</span> <span class="fu">unique</span>(c8<span class="sc">$</span>ensembl_gene[<span class="fu">grepl</span>(<span class="st">"BILE_DUCT_CELLS_3"</span>, c8<span class="sc">$</span>gs_name)])</span>
<span id="cb121-6"><a href="#cb121-6" aria-hidden="true" tabindex="-1"></a>bileduct_4 <span class="ot">&lt;-</span> <span class="fu">unique</span>(c8<span class="sc">$</span>ensembl_gene[<span class="fu">grepl</span>(<span class="st">"BILE_DUCT_CELLS_4"</span>, c8<span class="sc">$</span>gs_name)])</span>
<span id="cb121-7"><a href="#cb121-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb121-8"><a href="#cb121-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine unique genes from each bileduct gene set for an overall signature</span></span>
<span id="cb121-9"><a href="#cb121-9" aria-hidden="true" tabindex="-1"></a>bileduct_core_genes <span class="ot">&lt;-</span> <span class="fu">intersect</span>(bileduct_1, bileduct_2)</span>
<span id="cb121-10"><a href="#cb121-10" aria-hidden="true" tabindex="-1"></a>bileduct_core_genes <span class="ot">&lt;-</span> <span class="fu">intersect</span>(bileduct_core_genes, bileduct_3)</span>
<span id="cb121-11"><a href="#cb121-11" aria-hidden="true" tabindex="-1"></a>bileduct_core_genes <span class="ot">&lt;-</span> <span class="fu">intersect</span>(bileduct_core_genes, bileduct_4)</span>
<span id="cb121-12"><a href="#cb121-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb121-13"><a href="#cb121-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Subset to bileduct core genes and combine datasets</span></span>
<span id="cb121-14"><a href="#cb121-14" aria-hidden="true" tabindex="-1"></a>mat_chol <span class="ot">&lt;-</span> <span class="fu">counts</span>(chol_dds, <span class="at">normalized =</span> <span class="cn">TRUE</span>)</span>
<span id="cb121-15"><a href="#cb121-15" aria-hidden="true" tabindex="-1"></a>mat_lihc <span class="ot">&lt;-</span> <span class="fu">counts</span>(lihc_dds, <span class="at">normalized =</span> <span class="cn">TRUE</span>)</span>
<span id="cb121-16"><a href="#cb121-16" aria-hidden="true" tabindex="-1"></a>mat_chol <span class="ot">&lt;-</span> mat_chol[bileduct_core_genes[bileduct_core_genes <span class="sc">%in%</span> </span>
<span id="cb121-17"><a href="#cb121-17" aria-hidden="true" tabindex="-1"></a>                                           <span class="fu">rownames</span>(mat_chol)], ]</span>
<span id="cb121-18"><a href="#cb121-18" aria-hidden="true" tabindex="-1"></a>mat_lihc <span class="ot">&lt;-</span> mat_lihc[bileduct_core_genes[bileduct_core_genes <span class="sc">%in%</span> </span>
<span id="cb121-19"><a href="#cb121-19" aria-hidden="true" tabindex="-1"></a>                                           <span class="fu">rownames</span>(mat_lihc)], ]</span>
<span id="cb121-20"><a href="#cb121-20" aria-hidden="true" tabindex="-1"></a>bileduct_mat <span class="ot">&lt;-</span> <span class="fu">cbind</span>(mat_chol, mat_lihc)</span>
<span id="cb121-21"><a href="#cb121-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb121-22"><a href="#cb121-22" aria-hidden="true" tabindex="-1"></a><span class="co"># Compute bileduct z-scores</span></span>
<span id="cb121-23"><a href="#cb121-23" aria-hidden="true" tabindex="-1"></a>bileduct_log2 <span class="ot">&lt;-</span> <span class="fu">log2</span>(bileduct_mat <span class="sc">+</span> <span class="dv">1</span>)</span>
<span id="cb121-24"><a href="#cb121-24" aria-hidden="true" tabindex="-1"></a>bileduct_meds <span class="ot">&lt;-</span> <span class="fu">apply</span>(bileduct_log2, <span class="dv">1</span>, median)</span>
<span id="cb121-25"><a href="#cb121-25" aria-hidden="true" tabindex="-1"></a>bileduct_medcenter <span class="ot">&lt;-</span> bileduct_log2 <span class="sc">-</span> bileduct_meds</span>
<span id="cb121-26"><a href="#cb121-26" aria-hidden="true" tabindex="-1"></a>bileduct_scores <span class="ot">&lt;-</span> <span class="fu">apply</span>(bileduct_medcenter, <span class="dv">2</span>, sum)</span>
<span id="cb121-27"><a href="#cb121-27" aria-hidden="true" tabindex="-1"></a>bileduct_scores <span class="ot">&lt;-</span> (bileduct_scores <span class="sc">-</span> <span class="fu">mean</span>(bileduct_scores)) <span class="sc">/</span> <span class="fu">sd</span>(bileduct_scores)</span>
<span id="cb121-28"><a href="#cb121-28" aria-hidden="true" tabindex="-1"></a>combined_coldata<span class="sc">$</span>bileduct_scores <span class="ot">&lt;-</span> bileduct_scores</span>
<span id="cb121-29"><a href="#cb121-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb121-30"><a href="#cb121-30" aria-hidden="true" tabindex="-1"></a>combined_coldata<span class="sc">$</span>groups <span class="ot">&lt;-</span> combined_coldata<span class="sc">$</span>bap1_activity_class</span>
<span id="cb121-31"><a href="#cb121-31" aria-hidden="true" tabindex="-1"></a>combined_coldata<span class="sc">$</span>groups <span class="ot">&lt;-</span> <span class="fu">as.character</span>(combined_coldata<span class="sc">$</span>groups)</span>
<span id="cb121-32"><a href="#cb121-32" aria-hidden="true" tabindex="-1"></a>combined_coldata<span class="sc">$</span>groups[combined_coldata<span class="sc">$</span>cohort <span class="sc">==</span> <span class="st">"TCGA-CHOL"</span>] <span class="ot">&lt;-</span> <span class="st">"CHOL"</span></span>
<span id="cb121-33"><a href="#cb121-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb121-34"><a href="#cb121-34" aria-hidden="true" tabindex="-1"></a><span class="fu">boxplot</span>(combined_coldata<span class="sc">$</span>bileduct_scores <span class="sc">~</span> <span class="fu">factor</span>(combined_coldata<span class="sc">$</span>groups,</span>
<span id="cb121-35"><a href="#cb121-35" aria-hidden="true" tabindex="-1"></a>                                                  <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"CHOL"</span>, </span>
<span id="cb121-36"><a href="#cb121-36" aria-hidden="true" tabindex="-1"></a>                                                             <span class="st">"Mutant-like"</span>,</span>
<span id="cb121-37"><a href="#cb121-37" aria-hidden="true" tabindex="-1"></a>                                                             <span class="st">"Wildtype-like"</span>)),</span>
<span id="cb121-38"><a href="#cb121-38" aria-hidden="true" tabindex="-1"></a>        <span class="at">col =</span> <span class="fu">c</span>(<span class="st">"blue"</span>, <span class="st">"salmon"</span>, <span class="st">"lightgrey"</span>),</span>
<span id="cb121-39"><a href="#cb121-39" aria-hidden="true" tabindex="-1"></a>        <span class="at">outline =</span> <span class="cn">FALSE</span>, <span class="at">xlab =</span> <span class="cn">NULL</span>, <span class="at">las =</span> <span class="dv">2</span>, <span class="at">cex.names =</span> <span class="fl">0.7</span>,</span>
<span id="cb121-40"><a href="#cb121-40" aria-hidden="true" tabindex="-1"></a>        <span class="at">ylab =</span> <span class="st">"Bile duct signature score"</span>,</span>
<span id="cb121-41"><a href="#cb121-41" aria-hidden="true" tabindex="-1"></a>        <span class="at">names =</span> <span class="fu">c</span>(<span class="st">"CHOL"</span>, <span class="st">"LIHC Mutant-like"</span>, <span class="st">"LIHC Wildtype-like"</span>))</span>
<span id="cb121-42"><a href="#cb121-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb121-43"><a href="#cb121-43" aria-hidden="true" tabindex="-1"></a><span class="fu">stripchart</span>(combined_coldata<span class="sc">$</span>bileduct_scores <span class="sc">~</span> <span class="fu">factor</span>(combined_coldata<span class="sc">$</span>groups,</span>
<span id="cb121-44"><a href="#cb121-44" aria-hidden="true" tabindex="-1"></a>                                                  <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"CHOL"</span>, <span class="st">"Mutant-like"</span>, <span class="st">"Wildtype-like"</span>)),</span>
<span id="cb121-45"><a href="#cb121-45" aria-hidden="true" tabindex="-1"></a>           <span class="at">add =</span> <span class="cn">TRUE</span>, <span class="at">vertical =</span> <span class="cn">TRUE</span>, <span class="at">method =</span> <span class="st">"jitter"</span>, <span class="at">pch =</span> <span class="dv">16</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="figure_code_files/figure-html/unnamed-chunk-52-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb122"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb122-1"><a href="#cb122-1" aria-hidden="true" tabindex="-1"></a><span class="fu">pairwise.wilcox.test</span>(combined_coldata<span class="sc">$</span>bileduct_scores,</span>
<span id="cb122-2"><a href="#cb122-2" aria-hidden="true" tabindex="-1"></a>                     combined_coldata<span class="sc">$</span>groups, <span class="at">p.adjust.method =</span> <span class="st">"bonferroni"</span>)<span class="sc">$</span>p.val</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                      CHOL  Mutant-like
Mutant-like   1.626741e-07           NA
Wildtype-like 2.544001e-18 0.0001671127</code></pre>
</div>
</div>
<blockquote class="blockquote">
<p>Further, bile duct signature scores in LIHC tumors had a moderate negative correlation to <em>BAP1</em> activity scores (Spearman correlation=-0.52, Figure 4D) […]</p>
</blockquote>
</section>
<section id="figure-4d" class="level3">
<h3 class="anchored" data-anchor-id="figure-4d">Figure 4D</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb124"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb124-1"><a href="#cb124-1" aria-hidden="true" tabindex="-1"></a>lihc_coldata <span class="ot">&lt;-</span> combined_coldata[combined_coldata<span class="sc">$</span>cohort <span class="sc">==</span> <span class="st">"TCGA-LIHC"</span>, ]</span>
<span id="cb124-2"><a href="#cb124-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb124-3"><a href="#cb124-3" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(lihc_coldata<span class="sc">$</span>bap1_activity_score, </span>
<span id="cb124-4"><a href="#cb124-4" aria-hidden="true" tabindex="-1"></a>     lihc_coldata<span class="sc">$</span>bileduct_scores,</span>
<span id="cb124-5"><a href="#cb124-5" aria-hidden="true" tabindex="-1"></a>     <span class="at">xlab =</span> <span class="st">"BAP1 activity score"</span>,</span>
<span id="cb124-6"><a href="#cb124-6" aria-hidden="true" tabindex="-1"></a>     <span class="at">ylab =</span> <span class="st">"Bile duct signature score"</span>,</span>
<span id="cb124-7"><a href="#cb124-7" aria-hidden="true" tabindex="-1"></a>     <span class="at">pch =</span> <span class="dv">16</span>,</span>
<span id="cb124-8"><a href="#cb124-8" aria-hidden="true" tabindex="-1"></a>     <span class="at">col =</span> <span class="fu">ifelse</span>(lihc_coldata<span class="sc">$</span>bap1_activity_class <span class="sc">==</span> <span class="st">"Mutant-like"</span>,</span>
<span id="cb124-9"><a href="#cb124-9" aria-hidden="true" tabindex="-1"></a>                    <span class="st">"salmon"</span>, <span class="st">"steelblue"</span>))</span>
<span id="cb124-10"><a href="#cb124-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb124-11"><a href="#cb124-11" aria-hidden="true" tabindex="-1"></a><span class="fu">abline</span>(<span class="at">v =</span> <span class="fu">max</span>(combined_scores[mod<span class="sc">$</span>classification <span class="sc">==</span> <span class="dv">1</span>]), </span>
<span id="cb124-12"><a href="#cb124-12" aria-hidden="true" tabindex="-1"></a>       <span class="at">lty =</span> <span class="st">"dotted"</span>, <span class="at">col =</span> <span class="st">"grey"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="figure_code_files/figure-html/unnamed-chunk-54-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb125"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb125-1"><a href="#cb125-1" aria-hidden="true" tabindex="-1"></a>lihc_adj_rsq <span class="ot">&lt;-</span> <span class="fu">summary</span>(<span class="fu">lm</span>(lihc_coldata<span class="sc">$</span>bap1_activity_score <span class="sc">~</span> </span>
<span id="cb125-2"><a href="#cb125-2" aria-hidden="true" tabindex="-1"></a>                             lihc_coldata<span class="sc">$</span>bileduct_scores))<span class="sc">$</span>adj.r</span>
<span id="cb125-3"><a href="#cb125-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-4"><a href="#cb125-4" aria-hidden="true" tabindex="-1"></a>lihc_adj_rsq <span class="ot">&lt;-</span> <span class="fu">round</span>(lihc_adj_rsq, <span class="dv">2</span>)</span>
<span id="cb125-5"><a href="#cb125-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-6"><a href="#cb125-6" aria-hidden="true" tabindex="-1"></a>lihc_cor <span class="ot">&lt;-</span> <span class="fu">cor</span>(lihc_coldata<span class="sc">$</span>bap1_activity_score, </span>
<span id="cb125-7"><a href="#cb125-7" aria-hidden="true" tabindex="-1"></a>                lihc_coldata<span class="sc">$</span>bileduct_scores, <span class="at">method =</span> <span class="st">"spearman"</span>)</span>
<span id="cb125-8"><a href="#cb125-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb125-9"><a href="#cb125-9" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="fu">sprintf</span>(<span class="st">"Linear regression of BAP1 and bileduct signature scores with Wherry adjustment: %s"</span>, lihc_adj_rsq), <span class="st">"</span><span class="sc">\n</span><span class="st">"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Linear regression of BAP1 and bileduct signature scores with Wherry adjustment: 0.23 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb127"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb127-1"><a href="#cb127-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="fu">sprintf</span>(<span class="st">"Spearman correlation: %s"</span>, <span class="fu">round</span>(lihc_cor, <span class="dv">2</span>)))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Spearman correlation: -0.52</code></pre>
</div>
</div>
<p>Here we further explore specific bileduct genesets in the C8 collection and create a boxplot separated by BAP1 altered and unaltered status. We also include cholangiocarcinoma samples to compare.</p>
<blockquote class="blockquote">
<p>We used CHOL samples as a comparison and observed that LIHC samples with BAP1 alterations were enriched for bile duct markers relative to unaltered samples (Figure 4D and Supplemental Table S5A, pairwise two-sided Mann-Whitney U test with continuity correction and Bonferroni adjusted p=3.7e-4).</p>
</blockquote>
</section>
<section id="figure-4e" class="level3">
<h3 class="anchored" data-anchor-id="figure-4e">Figure 4E</h3>
<p>This is a plot of positively and negatively enriched terms from gene set enrichment analysis using <code>fgsea</code> and the MSigDB C8 gene sets. Because of the length of the signature names, this plot won’t display well by default in the Quarto preview. Image was finalized in Adobe Illustrator 2021.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb129"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb129-1"><a href="#cb129-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Pre-filtered list of enriched gene sets with padj &lt; 0.05</span></span>
<span id="cb129-2"><a href="#cb129-2" aria-hidden="true" tabindex="-1"></a>lihc_c8 <span class="ot">&lt;-</span> <span class="fu">read.delim</span>(<span class="st">"../results/activity_signature/de_results/lihc_gsea_c8.txt"</span>)</span>
<span id="cb129-3"><a href="#cb129-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb129-4"><a href="#cb129-4" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="fu">rbind</span>(<span class="fu">head</span>(lihc_c8, <span class="at">n =</span> <span class="dv">10</span>), </span>
<span id="cb129-5"><a href="#cb129-5" aria-hidden="true" tabindex="-1"></a>             <span class="fu">tail</span>(lihc_c8, <span class="at">n =</span> <span class="dv">10</span>)), </span>
<span id="cb129-6"><a href="#cb129-6" aria-hidden="true" tabindex="-1"></a>       <span class="fu">aes</span>(<span class="fu">reorder</span>(pathway, NES))) <span class="sc">+</span> </span>
<span id="cb129-7"><a href="#cb129-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>, </span>
<span id="cb129-8"><a href="#cb129-8" aria-hidden="true" tabindex="-1"></a>           <span class="fu">aes</span>(<span class="at">fill =</span> Enrichment, <span class="at">y =</span> NES)) <span class="sc">+</span> </span>
<span id="cb129-9"><a href="#cb129-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>() <span class="sc">+</span> </span>
<span id="cb129-10"><a href="#cb129-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">setNames</span>(<span class="fu">c</span>(<span class="st">"firebrick2"</span>, <span class="st">"dodgerblue2"</span>),</span>
<span id="cb129-11"><a href="#cb129-11" aria-hidden="true" tabindex="-1"></a>                                      <span class="fu">c</span>(<span class="st">"Up-regulated"</span>, <span class="st">"Down-regulated"</span>))) <span class="sc">+</span> </span>
<span id="cb129-12"><a href="#cb129-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="dv">0</span>) <span class="sc">+</span> </span>
<span id="cb129-13"><a href="#cb129-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x=</span><span class="st">"Pathway"</span>, <span class="at">y=</span><span class="st">"Normalized Enrichment Score"</span>) <span class="sc">+</span> <span class="fu">theme_minimal</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="figure_code_files/figure-html/unnamed-chunk-56-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="figure-5---association-of-early-development-with-lower-bap1-activity-scores" class="level2">
<h2 class="anchored" data-anchor-id="figure-5---association-of-early-development-with-lower-bap1-activity-scores">Figure 5 - Association of early development with lower <em>BAP1</em> activity scores</h2>
<blockquote class="blockquote">
<p>Principal component analysis of single-cell data demonstrates separation of cells both by embryonic timepoint and by putative cell type along the first two principal components (Figure 5A).</p>
</blockquote>
<section id="figure-5a" class="level3">
<h3 class="anchored" data-anchor-id="figure-5a">Figure 5A</h3>
<p>This is PCA plot showing annotations by putative cell type (shape) and embryonic timepoint (color). Colors were customized and image was finalized in Adobe Illustrator 2021.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb130"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb130-1"><a href="#cb130-1" aria-hidden="true" tabindex="-1"></a>sc_counts <span class="ot">&lt;-</span> <span class="fu">read.delim</span>(<span class="st">"../data/gse90047/Single-cell_RNA-seq_Read_Count.txt"</span>)</span>
<span id="cb130-2"><a href="#cb130-2" aria-hidden="true" tabindex="-1"></a>sc_anno <span class="ot">&lt;-</span> <span class="fu">read.delim</span>(<span class="st">"../data/gse90047/GSE90047_singlecell_anno.txt"</span>)</span>
<span id="cb130-3"><a href="#cb130-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb130-4"><a href="#cb130-4" aria-hidden="true" tabindex="-1"></a>sc_genes <span class="ot">&lt;-</span> sc_counts[, <span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>]</span>
<span id="cb130-5"><a href="#cb130-5" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(sc_counts) <span class="ot">&lt;-</span> sc_counts<span class="sc">$</span>ID</span>
<span id="cb130-6"><a href="#cb130-6" aria-hidden="true" tabindex="-1"></a>sc_counts <span class="ot">&lt;-</span> sc_counts[, <span class="sc">-</span><span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>)]</span>
<span id="cb130-7"><a href="#cb130-7" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(sc_anno) <span class="ot">&lt;-</span> sc_anno<span class="sc">$</span>Sample_title</span>
<span id="cb130-8"><a href="#cb130-8" aria-hidden="true" tabindex="-1"></a>sc_counts <span class="ot">&lt;-</span> sc_counts[, <span class="fu">rownames</span>(sc_anno)]</span>
<span id="cb130-9"><a href="#cb130-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb130-10"><a href="#cb130-10" aria-hidden="true" tabindex="-1"></a>sc_anno<span class="sc">$</span>Sample_characteristics_ch1 <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"embryonic day: "</span>, <span class="st">""</span>,</span>
<span id="cb130-11"><a href="#cb130-11" aria-hidden="true" tabindex="-1"></a>                                           sc_anno<span class="sc">$</span>Sample_characteristics_ch1)</span>
<span id="cb130-12"><a href="#cb130-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb130-13"><a href="#cb130-13" aria-hidden="true" tabindex="-1"></a>sc_anno<span class="sc">$</span>Sample_characteristics_ch1<span class="fl">.4</span> <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"putative cell type: "</span>, <span class="st">""</span>,</span>
<span id="cb130-14"><a href="#cb130-14" aria-hidden="true" tabindex="-1"></a>                                             sc_anno<span class="sc">$</span>Sample_characteristics_ch1<span class="fl">.4</span>)</span>
<span id="cb130-15"><a href="#cb130-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb130-16"><a href="#cb130-16" aria-hidden="true" tabindex="-1"></a>sc_dds <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> sc_counts,</span>
<span id="cb130-17"><a href="#cb130-17" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">colData =</span> sc_anno,</span>
<span id="cb130-18"><a href="#cb130-18" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">design =</span> <span class="sc">~</span> <span class="dv">1</span>)</span>
<span id="cb130-19"><a href="#cb130-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb130-20"><a href="#cb130-20" aria-hidden="true" tabindex="-1"></a>sc_vsd <span class="ot">&lt;-</span> <span class="fu">vst</span>(sc_dds, <span class="at">blind =</span> <span class="cn">FALSE</span>)</span>
<span id="cb130-21"><a href="#cb130-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb130-22"><a href="#cb130-22" aria-hidden="true" tabindex="-1"></a>mut_up_gene_symbols <span class="ot">&lt;-</span> <span class="fu">c</span>()</span>
<span id="cb130-23"><a href="#cb130-23" aria-hidden="true" tabindex="-1"></a>mut_down_gene_symbols <span class="ot">&lt;-</span> <span class="fu">c</span>()</span>
<span id="cb130-24"><a href="#cb130-24" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> mut_up_genes) {</span>
<span id="cb130-25"><a href="#cb130-25" aria-hidden="true" tabindex="-1"></a>  mut_up_gene_symbols <span class="ot">&lt;-</span> <span class="fu">c</span>(mut_up_gene_symbols,</span>
<span id="cb130-26"><a href="#cb130-26" aria-hidden="true" tabindex="-1"></a>                           grch38<span class="sc">$</span>symbol[grch38<span class="sc">$</span>ensgene <span class="sc">==</span> i])</span>
<span id="cb130-27"><a href="#cb130-27" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb130-28"><a href="#cb130-28" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> mut_down_genes) {</span>
<span id="cb130-29"><a href="#cb130-29" aria-hidden="true" tabindex="-1"></a>  mut_down_gene_symbols <span class="ot">&lt;-</span> <span class="fu">c</span>(mut_down_gene_symbols,</span>
<span id="cb130-30"><a href="#cb130-30" aria-hidden="true" tabindex="-1"></a>                           grch38<span class="sc">$</span>symbol[grch38<span class="sc">$</span>ensgene <span class="sc">==</span> i])</span>
<span id="cb130-31"><a href="#cb130-31" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb130-32"><a href="#cb130-32" aria-hidden="true" tabindex="-1"></a>mut_up_gene_symbols <span class="ot">&lt;-</span> mut_up_gene_symbols[<span class="sc">!</span><span class="fu">is.na</span>(mut_up_gene_symbols) <span class="sc">&amp;</span></span>
<span id="cb130-33"><a href="#cb130-33" aria-hidden="true" tabindex="-1"></a>                                             mut_up_gene_symbols <span class="sc">!=</span> <span class="st">""</span>]</span>
<span id="cb130-34"><a href="#cb130-34" aria-hidden="true" tabindex="-1"></a>mut_up_gene_symbols <span class="ot">&lt;-</span> mut_up_gene_symbols[<span class="sc">!</span><span class="fu">duplicated</span>(mut_up_gene_symbols)]</span>
<span id="cb130-35"><a href="#cb130-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb130-36"><a href="#cb130-36" aria-hidden="true" tabindex="-1"></a>mut_down_gene_symbols <span class="ot">&lt;-</span> mut_down_gene_symbols[<span class="sc">!</span><span class="fu">is.na</span>(mut_down_gene_symbols) <span class="sc">&amp;</span></span>
<span id="cb130-37"><a href="#cb130-37" aria-hidden="true" tabindex="-1"></a>                                             mut_down_gene_symbols <span class="sc">!=</span> <span class="st">""</span>]</span>
<span id="cb130-38"><a href="#cb130-38" aria-hidden="true" tabindex="-1"></a>mut_down_gene_symbols <span class="ot">&lt;-</span> mut_down_gene_symbols[<span class="sc">!</span><span class="fu">duplicated</span>(mut_down_gene_symbols)]</span>
<span id="cb130-39"><a href="#cb130-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb130-40"><a href="#cb130-40" aria-hidden="true" tabindex="-1"></a>mut_up_mouse <span class="ot">&lt;-</span> grcm38<span class="sc">$</span>ensgene[<span class="fu">toupper</span>(grcm38<span class="sc">$</span>symbol) <span class="sc">%in%</span> mut_up_gene_symbols]</span>
<span id="cb130-41"><a href="#cb130-41" aria-hidden="true" tabindex="-1"></a>mut_down_mouse <span class="ot">&lt;-</span> grcm38<span class="sc">$</span>ensgene[<span class="fu">toupper</span>(grcm38<span class="sc">$</span>symbol) <span class="sc">%in%</span> mut_down_gene_symbols]</span>
<span id="cb130-42"><a href="#cb130-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb130-43"><a href="#cb130-43" aria-hidden="true" tabindex="-1"></a>mut_up_mouse <span class="ot">&lt;-</span> mut_up_mouse[<span class="sc">!</span><span class="fu">duplicated</span>(mut_up_mouse) <span class="sc">&amp;</span></span>
<span id="cb130-44"><a href="#cb130-44" aria-hidden="true" tabindex="-1"></a>                               mut_up_mouse <span class="sc">%in%</span> <span class="fu">rownames</span>(<span class="fu">assay</span>(sc_dds))]</span>
<span id="cb130-45"><a href="#cb130-45" aria-hidden="true" tabindex="-1"></a>mut_down_mouse <span class="ot">&lt;-</span> mut_down_mouse[<span class="sc">!</span><span class="fu">duplicated</span>(mut_down_mouse) <span class="sc">&amp;</span></span>
<span id="cb130-46"><a href="#cb130-46" aria-hidden="true" tabindex="-1"></a>                                   mut_down_mouse <span class="sc">%in%</span> <span class="fu">rownames</span>(<span class="fu">assay</span>(sc_dds))]</span>
<span id="cb130-47"><a href="#cb130-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb130-48"><a href="#cb130-48" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb130-49"><a href="#cb130-49" aria-hidden="true" tabindex="-1"></a>mouse_sc_log2 <span class="ot">&lt;-</span> <span class="fu">log2</span>(<span class="fu">assay</span>(sc_dds, <span class="at">normalized =</span> <span class="cn">TRUE</span>) <span class="sc">+</span> <span class="dv">1</span>)</span>
<span id="cb130-50"><a href="#cb130-50" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb130-51"><a href="#cb130-51" aria-hidden="true" tabindex="-1"></a>mouse_score_up <span class="ot">&lt;-</span> <span class="fu">colSums</span>(mouse_sc_log2[mut_up_mouse, ], <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb130-52"><a href="#cb130-52" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb130-53"><a href="#cb130-53" aria-hidden="true" tabindex="-1"></a>mouse_score_down <span class="ot">&lt;-</span> <span class="fu">colSums</span>(mouse_sc_log2[mut_down_mouse, ], <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb130-54"><a href="#cb130-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb130-55"><a href="#cb130-55" aria-hidden="true" tabindex="-1"></a>mouse_act_scores <span class="ot">&lt;-</span> mouse_score_down <span class="sc">-</span> mouse_score_up</span>
<span id="cb130-56"><a href="#cb130-56" aria-hidden="true" tabindex="-1"></a>sc_vsd<span class="sc">$</span>sc_act_scores <span class="ot">&lt;-</span> mouse_act_scores</span>
<span id="cb130-57"><a href="#cb130-57" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb130-58"><a href="#cb130-58" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb130-59"><a href="#cb130-59" aria-hidden="true" tabindex="-1"></a>sc_counts_norm <span class="ot">&lt;-</span> <span class="fu">assay</span>(sc_dds, <span class="at">normalized =</span> <span class="cn">TRUE</span>)</span>
<span id="cb130-60"><a href="#cb130-60" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb130-61"><a href="#cb130-61" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> sc_dds<span class="sc">$</span>Sample_title) {</span>
<span id="cb130-62"><a href="#cb130-62" aria-hidden="true" tabindex="-1"></a>  sc_vsd<span class="sc">$</span>log2_bap1[sc_dds<span class="sc">$</span>Sample_title <span class="sc">==</span> i] <span class="ot">&lt;-</span></span>
<span id="cb130-63"><a href="#cb130-63" aria-hidden="true" tabindex="-1"></a>    <span class="fu">log2</span>(sc_counts_norm[<span class="st">"ENSMUSG00000021901"</span>, i] <span class="sc">+</span> <span class="dv">1</span>)</span>
<span id="cb130-64"><a href="#cb130-64" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb130-65"><a href="#cb130-65" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb130-66"><a href="#cb130-66" aria-hidden="true" tabindex="-1"></a>sc_plot <span class="ot">&lt;-</span> <span class="fu">plotPCA</span>(sc_vsd, <span class="at">intgroup =</span> <span class="fu">c</span>(<span class="st">"Sample_characteristics_ch1"</span>,</span>
<span id="cb130-67"><a href="#cb130-67" aria-hidden="true" tabindex="-1"></a>                                        <span class="st">"Sample_characteristics_ch1.4"</span>,</span>
<span id="cb130-68"><a href="#cb130-68" aria-hidden="true" tabindex="-1"></a>                                        <span class="st">"log2_bap1"</span>,</span>
<span id="cb130-69"><a href="#cb130-69" aria-hidden="true" tabindex="-1"></a>                                        <span class="st">"sc_act_scores"</span>),</span>
<span id="cb130-70"><a href="#cb130-70" aria-hidden="true" tabindex="-1"></a>                   <span class="at">returnData =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>using ntop=500 top features by variance</code></pre>
</div>
<div class="sourceCode cell-code" id="cb132"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb132-1"><a href="#cb132-1" aria-hidden="true" tabindex="-1"></a>percentVar <span class="ot">&lt;-</span> <span class="fu">round</span>(<span class="dv">100</span> <span class="sc">*</span> <span class="fu">attr</span>(sc_plot, <span class="st">"percentVar"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb133"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb133-1"><a href="#cb133-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(sc_plot, <span class="fu">aes</span>(PC1, PC2, <span class="at">color =</span> Sample_characteristics_ch1, </span>
<span id="cb133-2"><a href="#cb133-2" aria-hidden="true" tabindex="-1"></a>                    <span class="at">shape =</span> Sample_characteristics_ch1<span class="fl">.4</span>)) <span class="sc">+</span></span>
<span id="cb133-3"><a href="#cb133-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size=</span><span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb133-4"><a href="#cb133-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="fu">paste0</span>(<span class="st">"PC1: "</span>,percentVar[<span class="dv">1</span>],<span class="st">"% variance"</span>)) <span class="sc">+</span></span>
<span id="cb133-5"><a href="#cb133-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="fu">paste0</span>(<span class="st">"PC2: "</span>,percentVar[<span class="dv">2</span>],<span class="st">"% variance"</span>)) <span class="sc">+</span> </span>
<span id="cb133-6"><a href="#cb133-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_fixed</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="figure_code_files/figure-html/unnamed-chunk-58-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<blockquote class="blockquote">
<p>When <em>BAP1</em> activity score values were overlayed on top of these data, there was a clear trend of lower scores (more mutant-like) in cells that are earlier in development; as the embryonic timepoint increases towards a more differentiated phenotype, scores increase along a pronounced gradient (Figure 5B).</p>
</blockquote>
</section>
<section id="figure-5b" class="level3">
<h3 class="anchored" data-anchor-id="figure-5b">Figure 5B</h3>
<p>This is a variation of the same data from Figure 5A. Here, colors are continuous values representing <em>BAP1</em> activity scores. Image was finalized in Adobe Illustrator 2021.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb134"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb134-1"><a href="#cb134-1" aria-hidden="true" tabindex="-1"></a>sc_plot<span class="sc">$</span>act_floored <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(sc_plot<span class="sc">$</span>sc_act_scores <span class="sc">&lt;</span> <span class="sc">-</span><span class="dv">1750</span>, <span class="sc">-</span><span class="dv">1750</span>,</span>
<span id="cb134-2"><a href="#cb134-2" aria-hidden="true" tabindex="-1"></a>                              sc_plot<span class="sc">$</span>sc_act_scores)</span>
<span id="cb134-3"><a href="#cb134-3" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(sc_plot, <span class="fu">aes</span>(PC1, PC2, <span class="at">color =</span> act_floored, </span>
<span id="cb134-4"><a href="#cb134-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">shape =</span> Sample_characteristics_ch1<span class="fl">.4</span>)) <span class="sc">+</span></span>
<span id="cb134-5"><a href="#cb134-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size=</span><span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb134-6"><a href="#cb134-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="fu">paste0</span>(<span class="st">"PC1: "</span>,percentVar[<span class="dv">1</span>],<span class="st">"% variance"</span>)) <span class="sc">+</span></span>
<span id="cb134-7"><a href="#cb134-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="fu">paste0</span>(<span class="st">"PC2: "</span>,percentVar[<span class="dv">2</span>],<span class="st">"% variance"</span>)) <span class="sc">+</span> </span>
<span id="cb134-8"><a href="#cb134-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_fixed</span>() <span class="sc">+</span> <span class="fu">scale_color_viridis_c</span>(<span class="at">begin =</span> <span class="dv">0</span>, <span class="at">end =</span> <span class="dv">1</span>, <span class="at">direction =</span> <span class="sc">-</span><span class="dv">1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="figure_code_files/figure-html/unnamed-chunk-59-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<blockquote class="blockquote">
<p>This trend holds in bulk RNA-seq data from the same study, showing a high correlation between embryonic timepoint and <em>BAP1</em> activity score (adjusted r<sup>2</sup>=0.81, Figure 5C).</p>
</blockquote>
</section>
<section id="figure-5c" class="level3">
<h3 class="anchored" data-anchor-id="figure-5c">Figure 5C</h3>
<p>This plot uses bulk RNA-seq data from the same project (GSE90047) to show <em>BAP1</em> activity scores across embryonic timepoints. Image was finalized in Adobe Illustrator 2021.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb135"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb135-1"><a href="#cb135-1" aria-hidden="true" tabindex="-1"></a>mouse_bulk <span class="ot">&lt;-</span> <span class="fu">read.delim</span>(<span class="st">"../data/gse90047/Bulk-cell_RNA-seq_Read_Count.txt"</span>)</span>
<span id="cb135-2"><a href="#cb135-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb135-3"><a href="#cb135-3" aria-hidden="true" tabindex="-1"></a>mut_up_gene_symbols <span class="ot">&lt;-</span> <span class="fu">c</span>()</span>
<span id="cb135-4"><a href="#cb135-4" aria-hidden="true" tabindex="-1"></a>mut_down_gene_symbols <span class="ot">&lt;-</span> <span class="fu">c</span>()</span>
<span id="cb135-5"><a href="#cb135-5" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> mut_up_genes) {</span>
<span id="cb135-6"><a href="#cb135-6" aria-hidden="true" tabindex="-1"></a>  mut_up_gene_symbols <span class="ot">&lt;-</span> <span class="fu">c</span>(mut_up_gene_symbols,</span>
<span id="cb135-7"><a href="#cb135-7" aria-hidden="true" tabindex="-1"></a>                           grch38<span class="sc">$</span>symbol[grch38<span class="sc">$</span>ensgene <span class="sc">==</span> i])</span>
<span id="cb135-8"><a href="#cb135-8" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb135-9"><a href="#cb135-9" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> mut_down_genes) {</span>
<span id="cb135-10"><a href="#cb135-10" aria-hidden="true" tabindex="-1"></a>  mut_down_gene_symbols <span class="ot">&lt;-</span> <span class="fu">c</span>(mut_down_gene_symbols,</span>
<span id="cb135-11"><a href="#cb135-11" aria-hidden="true" tabindex="-1"></a>                           grch38<span class="sc">$</span>symbol[grch38<span class="sc">$</span>ensgene <span class="sc">==</span> i])</span>
<span id="cb135-12"><a href="#cb135-12" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb135-13"><a href="#cb135-13" aria-hidden="true" tabindex="-1"></a>mut_up_gene_symbols <span class="ot">&lt;-</span> mut_up_gene_symbols[<span class="sc">!</span><span class="fu">is.na</span>(mut_up_gene_symbols) <span class="sc">&amp;</span></span>
<span id="cb135-14"><a href="#cb135-14" aria-hidden="true" tabindex="-1"></a>                                             mut_up_gene_symbols <span class="sc">!=</span> <span class="st">""</span>]</span>
<span id="cb135-15"><a href="#cb135-15" aria-hidden="true" tabindex="-1"></a>mut_up_gene_symbols <span class="ot">&lt;-</span> mut_up_gene_symbols[<span class="sc">!</span><span class="fu">duplicated</span>(mut_up_gene_symbols)]</span>
<span id="cb135-16"><a href="#cb135-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb135-17"><a href="#cb135-17" aria-hidden="true" tabindex="-1"></a>mut_down_gene_symbols <span class="ot">&lt;-</span> mut_down_gene_symbols[<span class="sc">!</span><span class="fu">is.na</span>(mut_down_gene_symbols) <span class="sc">&amp;</span></span>
<span id="cb135-18"><a href="#cb135-18" aria-hidden="true" tabindex="-1"></a>                                             mut_down_gene_symbols <span class="sc">!=</span> <span class="st">""</span>]</span>
<span id="cb135-19"><a href="#cb135-19" aria-hidden="true" tabindex="-1"></a>mut_down_gene_symbols <span class="ot">&lt;-</span> mut_down_gene_symbols[<span class="sc">!</span><span class="fu">duplicated</span>(mut_down_gene_symbols)]</span>
<span id="cb135-20"><a href="#cb135-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb135-21"><a href="#cb135-21" aria-hidden="true" tabindex="-1"></a>mouse_bulk<span class="sc">$</span>Symbol <span class="ot">&lt;-</span> <span class="fu">toupper</span>(mouse_bulk<span class="sc">$</span>Symbol)</span>
<span id="cb135-22"><a href="#cb135-22" aria-hidden="true" tabindex="-1"></a>mouse_bulk <span class="ot">&lt;-</span> mouse_bulk[<span class="sc">!</span><span class="fu">duplicated</span>(mouse_bulk<span class="sc">$</span>Symbol), ]</span>
<span id="cb135-23"><a href="#cb135-23" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(mouse_bulk) <span class="ot">&lt;-</span> mouse_bulk<span class="sc">$</span>Symbol</span>
<span id="cb135-24"><a href="#cb135-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb135-25"><a href="#cb135-25" aria-hidden="true" tabindex="-1"></a>mouse_bulk <span class="ot">&lt;-</span> mouse_bulk[, <span class="sc">-</span><span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>)]</span>
<span id="cb135-26"><a href="#cb135-26" aria-hidden="true" tabindex="-1"></a>mouse_anno <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">sample =</span> <span class="fu">colnames</span>(mouse_bulk),</span>
<span id="cb135-27"><a href="#cb135-27" aria-hidden="true" tabindex="-1"></a>                         <span class="at">row.names =</span> <span class="fu">colnames</span>(mouse_bulk))</span>
<span id="cb135-28"><a href="#cb135-28" aria-hidden="true" tabindex="-1"></a>mouse_dds <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> mouse_bulk,</span>
<span id="cb135-29"><a href="#cb135-29" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">colData =</span> mouse_anno,</span>
<span id="cb135-30"><a href="#cb135-30" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">design =</span> <span class="sc">~</span><span class="dv">1</span>)</span>
<span id="cb135-31"><a href="#cb135-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb135-32"><a href="#cb135-32" aria-hidden="true" tabindex="-1"></a>mouse_dds <span class="ot">&lt;-</span> <span class="fu">estimateSizeFactors</span>(mouse_dds)</span>
<span id="cb135-33"><a href="#cb135-33" aria-hidden="true" tabindex="-1"></a>mouse_counts <span class="ot">&lt;-</span> <span class="fu">counts</span>(mouse_dds, <span class="at">normalized =</span> <span class="cn">TRUE</span>)</span>
<span id="cb135-34"><a href="#cb135-34" aria-hidden="true" tabindex="-1"></a>mouse_counts_log2 <span class="ot">&lt;-</span> <span class="fu">log2</span>(mouse_counts <span class="sc">+</span> <span class="dv">1</span>)</span>
<span id="cb135-35"><a href="#cb135-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb135-36"><a href="#cb135-36" aria-hidden="true" tabindex="-1"></a>mouse_score_up <span class="ot">&lt;-</span> <span class="fu">colSums</span>(mouse_counts_log2[mut_up_gene_symbols[mut_up_gene_symbols <span class="sc">%in%</span> <span class="fu">rownames</span>(mouse_counts_log2)], ], <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb135-37"><a href="#cb135-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb135-38"><a href="#cb135-38" aria-hidden="true" tabindex="-1"></a>mouse_score_down <span class="ot">&lt;-</span> <span class="fu">colSums</span>(mouse_counts_log2[mut_down_gene_symbols[mut_down_gene_symbols <span class="sc">%in%</span> <span class="fu">rownames</span>(mouse_counts_log2)], ], <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb135-39"><a href="#cb135-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb135-40"><a href="#cb135-40" aria-hidden="true" tabindex="-1"></a>mouse_act_scores <span class="ot">&lt;-</span> mouse_score_down <span class="sc">-</span> mouse_score_up</span>
<span id="cb135-41"><a href="#cb135-41" aria-hidden="true" tabindex="-1"></a>mouse_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">bap1_activity_score =</span> mouse_act_scores, </span>
<span id="cb135-42"><a href="#cb135-42" aria-hidden="true" tabindex="-1"></a>                       <span class="at">sample =</span> <span class="fu">colnames</span>(mouse_counts))</span>
<span id="cb135-43"><a href="#cb135-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb135-44"><a href="#cb135-44" aria-hidden="true" tabindex="-1"></a>mouse_timepoint <span class="ot">&lt;-</span> <span class="fu">c</span>()</span>
<span id="cb135-45"><a href="#cb135-45" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> mouse_df<span class="sc">$</span>sample) {</span>
<span id="cb135-46"><a href="#cb135-46" aria-hidden="true" tabindex="-1"></a>  mouse_timepoint <span class="ot">&lt;-</span> <span class="fu">c</span>(mouse_timepoint, <span class="fu">as.numeric</span>(<span class="fu">substr</span>(i, <span class="dv">2</span>, <span class="dv">5</span>)))</span>
<span id="cb135-47"><a href="#cb135-47" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb135-48"><a href="#cb135-48" aria-hidden="true" tabindex="-1"></a>mouse_df<span class="sc">$</span>timepoint <span class="ot">&lt;-</span> mouse_timepoint</span>
<span id="cb135-49"><a href="#cb135-49" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb135-50"><a href="#cb135-50" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(mouse_df<span class="sc">$</span>timepoint, mouse_df<span class="sc">$</span>bap1_activity_score, <span class="at">pch =</span> <span class="dv">16</span>, <span class="at">xlab =</span> <span class="st">"Embryonic timepoint (day)"</span>, <span class="at">ylab =</span> <span class="st">"BAP1 activity score"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="figure_code_files/figure-html/unnamed-chunk-60-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb136"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb136-1"><a href="#cb136-1" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(<span class="fu">lm</span>(mouse_df<span class="sc">$</span>timepoint <span class="sc">~</span> mouse_df<span class="sc">$</span>bap1_activity_score))<span class="sc">$</span>adj.r</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0.805378</code></pre>
</div>
</div>
</section>
</section>
<section id="supplemental-figure-s4" class="level2">
<h2 class="anchored" data-anchor-id="supplemental-figure-s4">Supplemental Figure S4</h2>
<p>This is a plot showing no strong relationship between <em>BAP1</em> expression and <em>BAP1</em> activity score at a pan-cancer level. Image was finalized in Adobe Illustrator 2021.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb138"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb138-1"><a href="#cb138-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(sample_anno<span class="sc">$</span>log2_bap1, sample_anno<span class="sc">$</span>bap1_activity_score, <span class="at">pch =</span> <span class="dv">16</span>,</span>
<span id="cb138-2"><a href="#cb138-2" aria-hidden="true" tabindex="-1"></a>     <span class="at">xlab =</span> <span class="st">"log2 BAP1 expression"</span>, <span class="at">ylab =</span> <span class="st">"BAP1 activity score"</span>, </span>
<span id="cb138-3"><a href="#cb138-3" aria-hidden="true" tabindex="-1"></a>     <span class="at">main =</span> <span class="st">"Pan-cancer"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="figure_code_files/figure-html/unnamed-chunk-62-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb139"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb139-1"><a href="#cb139-1" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(<span class="fu">lm</span>(sample_anno<span class="sc">$</span>log2_bap1 <span class="sc">~</span> sample_anno<span class="sc">$</span>bap1_activity_score))<span class="sc">$</span>adj.r</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0.03479131</code></pre>
</div>
</div>
</section>
<section id="supplemental-figure-s5" class="level2">
<h2 class="anchored" data-anchor-id="supplemental-figure-s5">Supplemental Figure S5</h2>
<p>This is a meta-plot showing <em>BAP1</em> activity scores for each alteration type and tumor type. Because the image is so large (6x6 boxplots), it will not display properly in the Quarto preview. Image was finalized in Adobe Illustrator 2021.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb141"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb141-1"><a href="#cb141-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dev.new</span>(<span class="at">height =</span> <span class="dv">8</span>, <span class="at">width =</span> <span class="dv">11</span>)</span>
<span id="cb141-2"><a href="#cb141-2" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow =</span> <span class="fu">c</span>(<span class="dv">6</span>, <span class="dv">6</span>), <span class="at">mar =</span> <span class="fu">c</span>(<span class="dv">3</span>, <span class="dv">3</span>, <span class="dv">3</span>, <span class="dv">3</span>))</span>
<span id="cb141-3"><a href="#cb141-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb141-4"><a href="#cb141-4" aria-hidden="true" tabindex="-1"></a>cohorts <span class="ot">&lt;-</span> <span class="fu">sort</span>(<span class="fu">unique</span>(sample_anno<span class="sc">$</span>cohort))</span>
<span id="cb141-5"><a href="#cb141-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb141-6"><a href="#cb141-6" aria-hidden="true" tabindex="-1"></a>score_plots <span class="ot">&lt;-</span> <span class="cf">function</span>(cohort) {</span>
<span id="cb141-7"><a href="#cb141-7" aria-hidden="true" tabindex="-1"></a>  alt_types <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"unaltered"</span>, <span class="st">"copynumber"</span>, <span class="st">"mutation+copynumber"</span>, <span class="st">"mutation"</span>)</span>
<span id="cb141-8"><a href="#cb141-8" aria-hidden="true" tabindex="-1"></a>  alt_types <span class="ot">&lt;-</span> alt_types[alt_types <span class="sc">%in%</span> sample_anno<span class="sc">$</span>alteration_type[sample_anno<span class="sc">$</span>cohort <span class="sc">==</span> cohort]]</span>
<span id="cb141-9"><a href="#cb141-9" aria-hidden="true" tabindex="-1"></a>  alt_names <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(alt_types <span class="sc">==</span> <span class="st">"unaltered"</span>, <span class="st">"Un"</span>, </span>
<span id="cb141-10"><a href="#cb141-10" aria-hidden="true" tabindex="-1"></a>                      <span class="fu">ifelse</span>(alt_types <span class="sc">==</span> <span class="st">"copynumber"</span>, <span class="st">"CN"</span>,</span>
<span id="cb141-11"><a href="#cb141-11" aria-hidden="true" tabindex="-1"></a>                             <span class="fu">ifelse</span>(alt_types <span class="sc">==</span> <span class="st">"mutation+copynumber"</span>, <span class="st">"Mut+CN"</span>, <span class="st">"Mut"</span>)))</span>
<span id="cb141-12"><a href="#cb141-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">boxplot</span>(sample_anno<span class="sc">$</span>bap1_activity_score[sample_anno<span class="sc">$</span>cohort <span class="sc">==</span> cohort] <span class="sc">~</span> </span>
<span id="cb141-13"><a href="#cb141-13" aria-hidden="true" tabindex="-1"></a>          <span class="fu">factor</span>(sample_anno<span class="sc">$</span>alteration_type[sample_anno<span class="sc">$</span>cohort <span class="sc">==</span> cohort],</span>
<span id="cb141-14"><a href="#cb141-14" aria-hidden="true" tabindex="-1"></a>                 <span class="at">levels =</span> alt_types),</span>
<span id="cb141-15"><a href="#cb141-15" aria-hidden="true" tabindex="-1"></a>        <span class="at">outline =</span> <span class="cn">FALSE</span>, <span class="at">main =</span> <span class="fu">gsub</span>(<span class="st">"TCGA-"</span>, <span class="st">""</span>, cohort), <span class="at">names =</span> alt_names, </span>
<span id="cb141-16"><a href="#cb141-16" aria-hidden="true" tabindex="-1"></a>        <span class="at">xlab =</span> <span class="cn">NULL</span>, <span class="at">ylab =</span> <span class="st">"BAP1 activity score"</span>)</span>
<span id="cb141-17"><a href="#cb141-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stripchart</span>(sample_anno<span class="sc">$</span>bap1_activity_score[sample_anno<span class="sc">$</span>cohort <span class="sc">==</span> cohort] <span class="sc">~</span> </span>
<span id="cb141-18"><a href="#cb141-18" aria-hidden="true" tabindex="-1"></a>          <span class="fu">factor</span>(sample_anno<span class="sc">$</span>alteration_type[sample_anno<span class="sc">$</span>cohort <span class="sc">==</span> cohort],</span>
<span id="cb141-19"><a href="#cb141-19" aria-hidden="true" tabindex="-1"></a>                 <span class="at">levels =</span> alt_types),</span>
<span id="cb141-20"><a href="#cb141-20" aria-hidden="true" tabindex="-1"></a>          <span class="at">add =</span> <span class="cn">TRUE</span>, <span class="at">vertical =</span> <span class="cn">TRUE</span>, <span class="at">method =</span> <span class="st">"jitter"</span>, <span class="at">pch =</span> <span class="dv">16</span>)</span>
<span id="cb141-21"><a href="#cb141-21" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb141-22"><a href="#cb141-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb141-23"><a href="#cb141-23" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (cohort <span class="cf">in</span> cohorts) {</span>
<span id="cb141-24"><a href="#cb141-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">score_plots</span>(cohort)</span>
<span id="cb141-25"><a href="#cb141-25" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb142"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb142-1"><a href="#cb142-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Run on all cohorts and manually statistics to plot</span></span>
<span id="cb142-2"><a href="#cb142-2" aria-hidden="true" tabindex="-1"></a>score_pvals <span class="ot">&lt;-</span> <span class="cf">function</span>(cohort) {</span>
<span id="cb142-3"><a href="#cb142-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pairwise.t.test</span>(sample_anno<span class="sc">$</span>bap1_activity_score[sample_anno<span class="sc">$</span>cohort <span class="sc">==</span> cohort <span class="sc">&amp;</span> </span>
<span id="cb142-4"><a href="#cb142-4" aria-hidden="true" tabindex="-1"></a>                                                    <span class="sc">!</span><span class="fu">is.na</span>(sample_anno<span class="sc">$</span>bap1_activity_score)], </span>
<span id="cb142-5"><a href="#cb142-5" aria-hidden="true" tabindex="-1"></a>          sample_anno<span class="sc">$</span>alteration_type[sample_anno<span class="sc">$</span>cohort <span class="sc">==</span> cohort <span class="sc">&amp;</span> </span>
<span id="cb142-6"><a href="#cb142-6" aria-hidden="true" tabindex="-1"></a>                                        <span class="sc">!</span><span class="fu">is.na</span>(sample_anno<span class="sc">$</span>bap1_activity_score)], </span>
<span id="cb142-7"><a href="#cb142-7" aria-hidden="true" tabindex="-1"></a>          <span class="at">p.adjust.method =</span> <span class="st">"bonferroni"</span>)<span class="sc">$</span>p.val</span>
<span id="cb142-8"><a href="#cb142-8" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="end-of-document" class="level1">
<h1>End of document</h1>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->




</body></html>